---
title: "ASAH1 rGBM Manuscript ELISA Figure"
author: "Justin Sing, University of Toronto"
date: "`r format(Sys.time(), '%a %B %d %X %Z %Y')`"
header-includes:
- \usepackage{pdflscape}
- \newcommand{\blandscape}{\begin{landscape}}
- \newcommand{\elandscape}{\end{landscape}}
- \usepackage{fancyhdr}
- \usepackage[fit, breakall]{truncate}
  \usepackage{tcolorbox}
  \newtcolorbox{myquote}{colback=red!5!white, colframe=red!75!black}
  \renewenvironment{quote}{\begin{myquote}}{\end{myquote}}
output:
  pdf_document: 
    keep_tex: no
    latex_engine: xelatex
    toc: yes
    toc_depth: 4
  html_notebook: default
editor_options:
  chunk_output_type: console
---

\pagestyle{fancy}
<!-- \fancyhead[LO,CE]{ASAH1 rGBM Manuscript} -->
\fancyfoot[CO,CE]{Justin Sing, University of Toronto}
\fancyfoot[LE,RO]{\thepage}

<!-- # Required Libraries -->
```{r message=FALSE, warning=FALSE, include=FALSE}
#' @title Try load package, install from correct package manager if not installed
#' @param package (character) string name of package to load
#' @param s (character) source to install package from it not installed
installIfNeeded <- function(package, s = "CRAN") {
  # s: "CRAN" or "BIO"
  if (s == "CRAN") {
    if (! requireNamespace(package, quietly=TRUE)) {
      install.packages(package)
      library(package, character.only = TRUE, quietly = TRUE)
    } else {
      library(package, character.only = TRUE, quietly = TRUE)
    }
  } else if (s == "BIO") {
    if (! requireNamespace("BiocManager", quietly=TRUE)) {
      install.packages("BiocManager")
    }
    if (! requireNamespace(package, quietly=TRUE)) {
      BiocManager::install(package)
      library(package, character.only = TRUE, quietly = TRUE)
    } else {
      library(package, character.only = TRUE, quietly = TRUE)
    }
  } else {
    stop(sprintf("Unknown source \"%s\".", s))
  }
}

# Data I/O and Wrangling
installIfNeeded("data.table")
installIfNeeded("dplyr")
# Visualization
installIfNeeded("ggplot2")
installIfNeeded("grid")
installIfNeeded("gridExtra")
installIfNeeded("ggpubr")
# Table Visualization
installIfNeeded("gt")
# Statistics
installIfNeeded("rstatix")
installIfNeeded("broman")
installIfNeeded("pwr")
# Survival Analysis
installIfNeeded("survival")
installIfNeeded("survminer")
# Machine Learning 
installIfNeeded("caret")
installIfNeeded("plotROC")
installIfNeeded("mixOmics", "BIO")
# parallilization
installIfNeeded("future")
```

<!-- # Helper Functions -->
```{r message=FALSE, warning=FALSE, include=FALSE}
cs <- function(.x) {    # center/scale vector
  print(.x)
  out <- .x - mean(.x)  # center
  out / sd(out)         # scale
}

median_centre_data <- function(train_df){
  values_to = "measure_col"
  measure_col="measure_col"
  train_df %>%
  tibble::rowid_to_column(var = "sample_id_col") %>%
  tidyr::pivot_longer(cols = c(ASAH1, MMP.9, SYNEMIN, GPNMB), names_to = "feature_id_col", values_to = "measure_col") ->
  train_long

  normalized_df = train_long %>% group_by(sample_id_col) %>% 
      mutate(median_run = median(measure_col, na.rm = TRUE)) %>% 
      ungroup()
  old_measure_col = paste("preNorm", measure_col, sep = "_")
  normalized_df = normalized_df %>% mutate(median_global = median(!!(sym(measure_col)), 
    na.rm = TRUE), `:=`(!!(old_measure_col), !!(sym(measure_col)))) %>% 
    mutate(diff_norm = median_global - median_run) %>% mutate(`:=`(!!(sym(measure_col)), 
    !!(sym(measure_col)) + diff_norm))
    
  normalized_df = switch("default", all = normalized_df, 
      default = normalized_df, minimal = normalized_df %>% 
        dplyr::select(c(sample_id_col, feature_id_col, 
          measure_col, old_measure_col)))
  
  # ggplot(normalized_df, aes(x=as.factor(sample_id_col), y=measure_col, col=sample_id_col)) +
  #   geom_boxplot()
  
  normalized_df %>%
    dplyr::select(sample_id_col, class, feature_id_col, measure_col) %>%
    tidyr::pivot_wider(id_cols = c("sample_id_col", "class"), names_from = "feature_id_col", values_from = "measure_col") %>%
    tibble::column_to_rownames("sample_id_col") -> train_df
  
  return(train_df)
}


#' @title Master Trainer and Prediction
#' @description Top Function to perform predictions for testing logistic regression, logistic regression + elastic regularization, random forest and k-Nearest Neightbours
#' 
#' @param train_df (data.frame) A dataframe that contains labels and features to run through prediction models. Has to have the following structure:
#' \item{class}{(character) sample/observation label. i.e. ctrl, iGBM, or rGBM sample.}
#' \item{feature_1}{(numeric} a specific feature measurement such as protein ASAH1)}
#' \item{feature_n}{(numeric) another specific feature.}
#' @param pred_col_prob (character) A character corresponding to the class of probabilities you want to extract for. i.e. "GBM"
#' @param multiclass (logical) Run predictions using multiclass modelling and prediction if more than 2 classes.
#' 
#' @return
#' 
#' returns a list object that contains trained model objects, lift data and roc data.
#' 
#' \item(model){(glmnet) Logistic Regression with Elastic Net Regularization Model }
#' \item(model_rf){(rf) Random Forest Model}
#' \item(model_logit){(glm) Logistic Regression Model}
#' \item(model_boostedlogit){(knn) k-Nearest Neighbours Models}
#' \item(master_lift){(data.table) lift data for all models}
#' \item(roc_data){(data.table) roc data for all models}
run_predictions <- function(train_df, pred_col_prob="GBM", multiclass=FALSE){
  
  ##*****************************************************
  ##  Model Training  ----
  
  if (!multiclass){
    # specify the cross-validation method
    ctrl <- trainControl(method = "LOOCV", search = "grid", classProbs = T, savePredictions = "final", summaryFunction = twoClassSummary, verboseIter = T)
    
    # fit models and use LOOCV to evaluate performance
    model <- train( class ~ ., data = train_df, na.action = na.omit, method = "glmnet", family="binomial", trControl = ctrl, metric = "ROC")
    model_rf <- train( class ~ ., data = train_df, na.action = na.omit, method = "rf", verbose=1, trControl = ctrl, metric = "ROC")
    model_logit <- train( class ~ . , data = train_df, na.action = na.omit, method = "glm", family="binomial", trControl = ctrl, metric = "ROC")
    model_boostedlogit <- train( class ~ . , data = train_df, na.action = na.omit, method = "knn", trControl = ctrl, metric = "ROC")
    
  } else{
    # specify the cross-validation method
    ctrl <- trainControl(method = "LOOCV", number = 10, search = "grid", classProbs = T, savePredictions = "final", verboseIter = T)
    
    # fit models and use LOOCV to evaluate performance
    model <- train( class ~ . , data = train_df, na.action = na.omit, method = "glmnet", family="multinomial", trControl = ctrl, metric = "ROC")
    model_rf <- train( class ~ ., data = train_df, na.action = na.omit, method = "rf", verbose=1, trControl = ctrl, metric = "ROC")
    model_logit <- train( class ~ . , data = train_df, na.action = na.omit, method = "multinom", family="binomial", trControl = ctrl, metric = "ROC")
    model_boostedlogit <- train( class ~ . , data = train_df, na.action = na.omit, method = "knn", trControl = ctrl, metric = "ROC")
  }
  
  ##*****************************************************
  ##  Generate Lift Plot Data ----
  
  if(!multiclass)  {
    for_lift = data.frame(Class = model$pred[model$pred$alpha==model$bestTune$alpha & model$pred$lambda==model$bestTune$lambda ,]$obs,  
                          glmnet = model$pred[model$pred$alpha==model$bestTune$alpha & model$pred$lambda==model$bestTune$lambda , pred_col_prob])
    for_lift_rf = data.frame(Class = model_rf$pred[model_rf$pred$mtry==model_rf$bestTune$mtry,]$obs,  
                             rf = model_rf$pred[model_rf$pred$mtry==model_rf$bestTune$mtry ,pred_col_prob])
    if (!multiclass){
      for_lift_logit = data.frame(Class = model_logit$pred$obs,  
                                  logit = model_logit$pred[,pred_col_prob])
    } else {
      for_lift_logit = data.frame(Class = model_logit$pred[model_logit$pred$decay==model_logit$bestTune$decay,]$obs,  
                                  logit = model_logit$pred[model_logit$pred$decay==model_logit$bestTune$decay,pred_col_prob])
    }
    for_lift_boostedlogit = data.frame(Class = model_boostedlogit$pred[model_boostedlogit$pred$k==model_boostedlogit$bestTune$k,]$obs,  
                                       knn = model_boostedlogit$pred[model_boostedlogit$pred$k==model_boostedlogit$bestTune$k ,pred_col_prob])
    
    lift_obj = lift(Class ~ glmnet, data = for_lift, class = pred_col_prob)
    lift_obj_rf = lift(Class ~ rf, data = for_lift_rf, class = pred_col_prob)
    lift_obj_logit = lift(Class ~ logit, data = for_lift_logit, class = pred_col_prob)
    lift_obj_boostedlogit = lift(Class ~ knn, data = for_lift_boostedlogit, class = pred_col_prob)
    
    master_lift <- rbindlist(list(lift_obj$data, lift_obj_rf$data, lift_obj_logit$data, lift_obj_boostedlogit$data))
    master_lift$liftModelVar <- factor(master_lift$liftModelVar, levels = c("logit", "glmnet", "knn", "rf"))
    
  } else {
    master_lift <- data.frame()
  }
  
  ##*****************************************************
  ## Generate ROC Prediction Data ----
  
  if (!multiclass){
    roc_data <- rbindlist(list(Logit=model_logit$pred, 
                               `Elastic Logit`=model$pred[model$pred$alpha==model$bestTune$alpha & model$pred$lambda==model$bestTune$lambda, ],
                               KNN=model_boostedlogit$pred[model_boostedlogit$pred$k==model_boostedlogit$bestTune$k, ], 
                               RF=model_rf$pred[model_rf$pred$mtry==model_rf$bestTune$mtry, ]), 
                          idcol = "Model", use.names = TRUE, fill = TRUE)
    roc_data$Model <- factor(roc_data$Model, levels=c("Logit", "Elastic Logit", "KNN", "RF"))
  } else {
    roc_data <- rbindlist(list(Logit=model_logit$pred[model_logit$pred$decay==model_logit$bestTune$decay, ], 
                               `Elastic Logit`=model$pred[model$pred$alpha==model$bestTune$alpha & model$pred$lambda==model$bestTune$lambda, ], 
                               KNN=model_boostedlogit$pred[model_boostedlogit$pred$k==model_boostedlogit$bestTune$k, ], 
                               RF=model_rf$pred[model_rf$pred$mtry==model_rf$bestTune$mtry, ]), 
                          idcol = "Model", use.names = TRUE, fill = TRUE)
    roc_data$Model <- factor(roc_data$Model, levels=c("Logit", "Elastic Logit", "KNN", "RF"))
  }
  
  ##*****************************************************
  ##  Return a list object with each model object, ----
  ##  lift data and roc data
  
  return(list(model=model, model_rf=model_rf, model_logit=model_logit, model_boostedlogit=model_boostedlogit, master_lift=master_lift, roc_data=roc_data))
}

#' @title Permutation Test Statistics
#' @description Perform permutation test to compare two groups
#' 
#' @param data (data.frame) if paired, dataframe must have comparison groups in two different columns with the associated log2 conc, else comparison groups must be in a single column with the grouping name. 
#' @param gene (char) character of gene being tested
#' @param response_var (char) column containing response data
#' @param group1 (char) character of group 1. i.e. control
#' @param group2 (char) character of group 2. i.e. treatment
#' @param method (char) character of either 'paired.perm.test' or 'perm.test'
#' @param grouping_column (Char) character of column containing group labels
#' 
#' @return returns a data.frame of data and test p-value
do_perm_test <- function(data, gene, response_var, group1, group2, method, grouping_column){
  data <- as.data.table(data)
  results <- data.frame(gene=gene, .y.=response_var, group1=group1, group2=group2, method=method, p=NA )
  if (method=="paired.perm.test"){
    message(sprintf("INFO: Performing %s", method))
    results$p <- broman::paired.perm.test(data[[group2]] - data[[group1]])
  } else if (method=="perm.test"){
    message(sprintf("INFO: Performing %s", method))
    results$p <- broman::perm.test(data[data[[grouping_column]]==group1][[response_var]], data[data[[grouping_column]]==group2][[response_var]], n.perm = 20000, var.equal = FALSE)
  }
  message(sprintf("INFO: Returning results"))
  return( results )
}

#' @title Subset a data frame based on groups
#' @description Subset a data frame based on groups
#' 
#' @param data (data.frame) if paired, dataframe must have comparison groups in two different columns with the associated log2 conc, else comparison groups must be in a single column with the grouping name. 
#' @param filter_groups (char) vector of character of groups to filter for
#' @param method (char) character of either 'paired.perm.test' or 'perm.test'
#' @param grouping_column (Char) character of column containing groups labels
#' 
#' @return returns a subset of data
filter_data <- function(data, filter_groups, method, grouping_column){
  data %>%
    dplyr::filter(!!rlang::sym(grouping_column) %in% filter_groups) %>%
    dplyr::select(Number.x, !!rlang::sym(grouping_column), gene, log2_conc) -> data_sub
  if (method=="paired.perm.test"){
    data_sub %>% 
      dplyr::group_by(Number.x) %>% 
      dplyr::add_count() %>%
      dplyr::ungroup() %>% 
      dplyr::filter(n==8) %>%
      dplyr::select(-n) -> data_sub
  }
  data_sub
}

#' Calculate the ratio of iGBM to rGBM and rGBM to iGBM for a given data frame
#'
#' @param data_long2 A data frame with columns "gene", "Number.x", "patient_state", "Age at surgery (years)", "Sex", "Tumor localization", "Type of resection", "MGMT promotor methylation", "p53 accumulation", "Ki67-Li", "EGFR vIII", "Treatment", "Time-to-reccurence (days)", "Latency (days)", and "conc".
#' @return A data frame with the calculated ratios.
get_ratio_data <- function(data_long2){
  ## Data wrangling ----
  ### iGBM over rGBM ratio
  data_long2 %>% 
    dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
    dplyr::group_by(gene, Number.x) %>%
    dplyr::group_map(function(.d, ...){
      .d <- as.data.table(.d)
      .d %>% 
        dplyr::group_by(gene, Number.x, patient_state) %>% 
        dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                         agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
      # print(.d)
      if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
        .d_initial <- .d[manifestation_type=="Initial manifestation"]
        .d_second <- .d[manifestation_type=="Second manifestation"]
        tmp$ratio <- .d_initial$conc / .d_second$conc
      } else if (nrow(.d)==2) {
        .d_initial <- .d[collection_type=="E1"]
        .d_second <- .d[collection_type=="E2"]
        tmp$ratio <- .d_initial$conc / .d_second$conc
      } else {
        tmp$ratio <- NA #.d$conc
      }
      return(tmp)
    }, .keep = TRUE) %>%
    rbindlist() %>%
    dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBMoverrGBM", "pre-surgery/post-surgery"))) %>%
    dplyr::filter(sample_comparison=="iGBMoverrGBM" ) %>%
    dplyr::mutate(log2_conc = log2(ratio)) %>%
    # dplyr::group_by(gene, Number.x) %>%
    # dplyr::add_count() %>%
    # dplyr::filter(n==2) %>%
    dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("iGBMoverrGBM"))) -> pdata
  
  ### rGBM over iGBM
  data_long2 %>% 
    dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
    dplyr::filter(!is.na(manifestation_type) | patient_state!=0) %>%
    dplyr::group_by(gene, Number.x) %>%
    dplyr::group_map(function(.d, ...){
      .d <- as.data.table(.d)
      .d %>% 
        dplyr::group_by(gene, Number.x, patient_state) %>% 
        dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                         agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
      # print(.d)
      if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
        .d_initial <- .d[manifestation_type=="Initial manifestation"]
        .d_second <- .d[manifestation_type=="Second manifestation"]
        tmp$ratio <- .d_second$conc / .d_initial$conc
      } else if (nrow(.d)==2) {
        .d_initial <- .d[collection_type=="E1"]
        .d_second <- .d[collection_type=="E2"]
        tmp$ratio <- .d_second$conc / .d_initial$conc
      } else {
        tmp$ratio <- NA #.d$conc
      }
      return(tmp)
    }, .keep = TRUE) %>%
    rbindlist() %>%
    dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "rGBMoveriGBM", "post-surgery/pre-surgery"))) %>%
    dplyr::filter(sample_comparison=="rGBMoveriGBM" ) %>%
    dplyr::filter(grepl(";", agg_lab_id)) %>%
    dplyr::mutate(log2_conc = log2(ratio)) %>%
    # dplyr::group_by(gene, Number.x) %>%
    # dplyr::add_count() %>%
    # dplyr::filter(n==2) %>%
    dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("rGBMoveriGBM"))) -> pdata2
  
  ### Prepare final table with ratios
  pdata %>%
    dplyr::select(Number.x, gene, log2_conc) %>%
    tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
    dplyr::mutate(class="iGBM") -> pdata
  
  pdata2 %>%
    dplyr::select(Number.x, gene, log2_conc) %>%
    tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
    dplyr::mutate(class="rGBM") -> pdata2
  
  
  pdata <- rbindlist(list(pdata, pdata2))

  ### Pivot Table to wide
  pdata %>%
    dplyr::select(Number.x, class, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide
  
  ## Remove non-paired patients
  data_wide %>%
    dplyr::group_by(Number.x) %>%
    dplyr::add_count() %>%
    dplyr::filter(n==2) %>%
    dplyr::ungroup() %>%
    dplyr::select(-n) -> data_wide
  
  # Prepare final table for analysis
  train_df <- data_wide
  train_df <- na.omit(train_df)  
  train_df <- train_df[ !is.infinite(train_df$ASAH1) , ]
  # Make Valid Column Names 
  colnames(train_df) <- make.names(colnames(train_df))
  train_df$class <- factor(train_df$class, levels=c("iGBM", "rGBM"))
  colnames(train_df) <- make.names(colnames(train_df))
  train_df <- as.data.table(train_df)
  train_df$group=0
  
  return(train_df)
}

#' Generate train and test datasets for leave-one-patient-out cross-validation
#'
#' This function takes in a dataset and returns a list containing train and test datasets
#' for leave-one-patient-out cross-validation. The test dataset consists of a single patient
#' with the "rGBM" class. If the `test_patient_id` parameter is not `NULL`, the test dataset
#' is subsetted using the provided patient ID. If the `test_patient_id` parameter is `NULL`,
#' a random patient with the "rGBM" class is chosen for the test dataset. The train dataset
#' consists of all other patients. If the `unique_random_train_patient_per_class` parameter is 
#' `TRUE`, the train dataset is further subsetted to include a unique patient for each "iGBM"
#' and "rGBM" class. The `test` parameter can be used to set the seed for the random number 
#' generator when sampling patients for the test and train datasets. This is useful for testing 
#' purposes.
#'
#' @param data A data frame containing the "Number.x" and "class" columns.
#' @param test_patient_id An optional patient ID to use for the test dataset. If not provided,
#' a random patient with the "rGBM" class is chosen.
#' @param unique_random_train_patient_per_class A logical value indicating whether to include
#' a unique patient for each "iGBM" and "rGBM" class in the train dataset.
#' @param test A logical value indicating whether to set the seed for the random number
#' generator for testing purposes.
#' @return A list containing the train and test datasets.
#' @export
#' @examples
#' get_train_test_for_loocv(data)
get_train_test_for_loocv <- function(data, test_patient_id=NULL, unique_random_train_patient_per_class = TRUE, test = FALSE){
  # Set seed for testing
  if (test){
    set.seed(2023)
  }
  
  if (is.null(test_patient_id)){
    # Sample single sample rGBM for Test data
    test_patients <- sample(unique(data$Number.x), 1)
    test_df <- subset(data, Number.x %in% test_patients & class == "rGBM")
  } else {
    # Subset for test data using provided test_patient_id
    test_df <- subset(data, Number.x %in% test_patient_id & class == "rGBM")
    test_patients <- test_patient_id
  }
  
  # Subset data to remove sample in test data
  data <- subset(data, !(Number.x %in% test_patients))
  
  if (unique_random_train_patient_per_class) {
    # Sample a unique patient for each iGBM and rGBM class
    igbm_train_patients <- sample(unique(data$Number.x[data$class == "iGBM"]), length(unique(data$Number.x))/2)
    rgbm_train_patients <- setdiff(data$Number.x, igbm_train_patients)
    
    # Filter data to obtain training data with a unique patient for each iGBM and rGBM class
    train_df <- subset(data, (Number.x %in% igbm_train_patients & class == "iGBM") | 
                         (Number.x %in% rgbm_train_patients & class == "rGBM"))
  } else {
    train_df <- data
  }
  
  return(list(train_df = train_df, test_df = test_df))
}


#' Calculate leave-one-patient-out cross-validation performance for splsda model
#'
#' This function takes in a dataset and a vector of patient IDs and returns a proportion
#' of correct predictions using a leave-one-patient-out cross-validation approach. A splsda
#' model is fit using the train data and used to make predictions on the test data. The
#' `unique_random_train_patient_per_class` parameter can be used to specify whether to 
#' include a unique patient for each "iGBM" and "rGBM" class in the train dataset. The
#' function returns a proportion of correct predictions.
#'
#' @param ratio_data A data frame containing the ratio data and class labels.
#' @param patient_ids A vector of patient IDs to use for the leave-one-patient-out
#' cross-validation.
#' @param unique_random_train_patient_per_class A logical value indicating whether to include
#' a unique patient for each "iGBM" and "rGBM" class in the train dataset.
#' @return A proportion of correct predictions.
#' @examples
#' splsda_loocv_perf(ratio_data, patient_ids)
splsda_loocv_perf <- function(ratio_data, patient_ids, unique_random_train_patient_per_class=TRUE){
  # Vector to store accuracy of LOOCV
  acc_vec <- numeric(length(patient_ids))
  i <- 1
  for (test_patient_id in patient_ids){
    # Split data into train and testing
    train_test <- get_train_test_for_loocv(ratio_data, test_patient_id=test_patient_id, unique_random_train_patient_per_class=unique_random_train_patient_per_class)
    train_df <- train_test$train_df
    test_df <- train_test$test_df
    # Perform splsda
    # MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2, -7, -8, -9, -10)], train_df$class, keepX = c(4,4)) 
    MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2)], train_df$class, keepX = c(4,4)) 
    # Make prediction
    # pred_test <- predict(MyResult.splsda, test_df[, c(-1, -2, -7, -8, -9, -10)], test_df$class)
    pred_test <- predict(MyResult.splsda, test_df[, c(-1, -2)], test_df$class)
    # evaluate the prediction accuracy for the first two components
    predict.comp2 <- pred_test$class$mahalanobis.dist[,2]
    # compute confusion matrix
    conf_mat <- table(factor(predict.comp2, levels = levels(train_df$class)), test_df$class)
    # print(conf_mat)
    # compute accuracy
    acc <- (conf_mat[1,1] + conf_mat[2,2])/(conf_mat[1,1] + conf_mat[2,2] + conf_mat[1,2] + conf_mat[2,1])
    # message(sprintf("INFO: Test Patient %s - predicted class = %s - model accuracy = %s", test_patient_id, predict.comp2, acc))
    # Store accuracy
    acc_vec[i] <- acc
    # update index
    i <- i + 1
  }
  # Compute proportion of correct predictions
  mean(acc_vec==1)
}

splsda <- function(train_df, test_df){
    
    # Perform splsda
    MyResult.splsda <- mixOmics::splsda(train_df, train_df$class, keepX = c(4,4)) 
    # Make prediction
    pred_test <- predict(MyResult.splsda, test_df, test_df$class)
    # evaluate the prediction accuracy for the first two components
    predict.comp2 <- pred_test$class$mahalanobis.dist[,2]
    # compute confusion matrix
    conf_mat <- table(factor(predict.comp2, levels = levels(train_df$class)), test_df$class)
    # print(conf_mat)
    # compute accuracy
    acc <- (conf_mat[1,1] + conf_mat[2,2])/(conf_mat[1,1] + conf_mat[2,2] + conf_mat[1,2] + conf_mat[2,1])
    
    return(list(model=MyResult.splsda, acc=acc))
}
```

<!-- # Data File Paths -->
```{r message=FALSE, warning=FALSE, include=FALSE}
patient_clinical_data <- "../../data/elisa_plasma_proteomics/Patients_information_FINAL.csv"
patient_longitudinal_mapping_file <- "../../data/elisa_plasma_proteomics/Patients_for_ELISA_longitudinal_mapping.csv"
elisa_data_file <- "../../data/elisa_plasma_proteomics/ELISA_data.csv"
```

<!-- # Load Data -->
```{r echo=FALSE, warning=FALSE}
clinical_data <- fread(patient_clinical_data)
patient_longitudinal_mapping_data <- fread(patient_longitudinal_mapping_file)

patient_meta_data <- merge(clinical_data, patient_longitudinal_mapping_data, by.x="LAB_ID", by.y = "LAB_ID", all = TRUE)
# patient_meta_data[ LAB_ID != LAB_ID.y]$LAB_ID <-  patient_meta_data[ LAB_ID != LAB_ID.y]$LAB_ID.y

elisa_data <- fread(elisa_data_file)

data_long <- merge(elisa_data, patient_meta_data, by = "LAB_ID")
# Remove patient 11, for some reason they are kept
data_long %>%
  dplyr::filter(Number.x!=11) -> data_long

patient_meta_data %>%
  dplyr::filter(!is.na(manifestation_type) & ELISA=="X") %>%
  dplyr::group_by(Number.x) %>%
  dplyr::add_count() %>%
  dplyr::select(`Unique Patient Number`=Number.x, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`) %>% 
  unique() %>% dplyr::ungroup() -> recurrent_patients_meta

# message(sprintf("There are %s patients with intial and recurrent ELISA measurements", nrow(recurrent_patients_meta)))

patient_meta_data %>%
  dplyr::filter(!is.na(manifestation_type)) %>%
  dplyr::select(Number.x, Sex, `Age at surgery (years)`, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, `Mass Spectrometry`, Lipidomics, `RT-qPCR`, IHC, ELISA) %>%
  dplyr::group_by(Number.x) %>%
  dplyr::add_count() %>%
  unique() %>%
  dplyr::ungroup()-> meta_dt

patient_id_meta <- unique(data_long[, .(LAB_ID, Number.x, collection_type, manifestation_type, patient_state, `Time-to-reccurence (days)`)])
```

\pagebreak

<!-- # Distribution of ELISA Measurements -->
```{r echo=FALSE, fig.height=11, fig.width=21, message=FALSE, warning=FALSE}
# iGBM vs rGBM ----
## Build gg dataframe for plotting ----
data_long %>%
  dplyr::filter(!is.na(manifestation_type) | patient_state==0) %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(Manifestation = plyr::mapvalues(manifestation_type, from=c("Initial manifestation", "Second manifestation"), to=c("iGBM", "rGBM"))) %>%
  dplyr::mutate(Manifestation = ifelse(is.na(Manifestation), "Ctrl", Manifestation)) %>%
  dplyr::mutate(Manifestation = factor(Manifestation, levels = c("Ctrl", "iGBM", "rGBM"))) -> pdata
pdata <- as.data.table(pdata)

## Perform Statistical-Test ----

comparisons <- matrix(c("Ctrl", "Ctrl", "iGBM", "iGBM", "rGBM", "rGBM"), ncol=2)
methods <- c("perm.test", "perm.test", "paired.perm.test")
# methods <- c("perm.test", "perm.test", "perm.test")

stats_frame <- parallel::mclapply(seq_len(nrow(comparisons)), function(iterator){
  comp <- comparisons[iterator, ]
  method <- methods[iterator]
  data_sub <- filter_data(pdata, comp, method, "Manifestation" )
  
  uni_genes <- data_sub %>% dplyr::pull(gene) %>% unique()
  
  gene_stat_frame <- parallel::mclapply(split(data_sub, data_sub$gene), function(.d){
    if (method=="paired.perm.test"){
      .d %>%
        dplyr::select(-gene) %>%
        tidyr::pivot_wider(id_cols = "Number.x", names_from = "Manifestation", values_from = "log2_conc") -> .d_wide
    } else {
      .d -> .d_wide
    }
    message(sprintf("INFO: Computing %s for %s for comparison %s", method, unique(.d$gene), paste(comp, collapse=", ") ))
    do_perm_test(data = .d_wide, gene = .d$gene, response_var = "log2_conc", group1 = comp[1], group2 = comp[2], method = method, grouping_column = "Manifestation")
  }, mc.cores = 4) %>%
    rbindlist() %>%
    unique()
}, mc.cores=3) %>% rbindlist()

stat_test <- stats_frame %>%
  dplyr::group_by(group1, group2) %>%
  adjust_pvalue(method = "fdr") %>%
  add_significance(p.col="p", output.col = "p.signif",
                   cutpoints = c(0, 1e-04, 0.001, 0.01, 0.05, 1),
                   symbols = c("****", "***", "**", "*", "ns")) %>%
  dplyr::arrange(gene) %>%
  add_xy_position(data=pdata, formula= log2_conc ~ Manifestation) 

stat_test %>%
  mutate(y.position = c(6.64, 6.64, 6.64, 8.70004, 8.70004, 8.70004, 11.49704, 11.49704, 11.49704, 13.62804, 13.62804, 13.62804)) -> stat_test

stat_test_igm_v_rgm <- stat_test

## Build the plot ----
colfunc <- colorRampPalette(c("red","yellow","springgreen","royalblue"))
ttr_categorizer <- function(data) {
  data <- as.numeric(data)
  if (is.na(data)){
    return(NA)
  }
  if (data >= 100 & data < 200){
    return("100-200")
  } else if (data >= 200 & data < 300){
    return("200-300")
  } else if (data >= 300 & data < 400){
    return("300-400")
  } else if (data >= 400 & data < 500){
    return("400-500")
  } else if (data >= 500 & data < 600){
    return("500-600")
  }
}
# ttr_categorizer <- function(data) {
#   data <- as.numeric(data)
#   if (is.na(data)){
#     return(NA)
#   }
#   if (data >= 100 & data < 150){
#     return("100-150")
#   } else if (data >= 150 & data < 200){
#     return("150-200")
#   } else if (data >= 200 & data < 250){
#     return("200-250")
#   } else if (data >= 250 & data < 300){
#     return("250-300")
#   } else if (data >= 300 & data < 350){
#     return("300-350")
#   } else if (data >= 350 & data < 400){
#     return("350-400")
#   } else if (data >= 400 & data < 450){
#     return("400-450")
#   } else if (data >= 450 & data < 500){
#     return("450-500")
#   } else if (data >= 500 & data < 550){
#     return("500-550")
#   } else if (data >= 550 & data < 600){
#     return("550-600")
#   }
# }
# ttr_categorizer <- function(data, split_value) {
#   if (data <= split_value) {
#     return(0)
#   } else {
#     return(1)
#   }
# }
# ttr_categorizer <- function(data) {
#   if (data >= 103 & data < 190){
#     return("early")
#   } else if (data >= 190 & data < 243){
#     return("middle")
#   } else if (data >= 243 & data <= 521){
#     return("late")
#   }
# }
pdata$`ttr_group` <- sapply(as.numeric(pdata$`Time-to-reccurence (days)`), ttr_categorizer)
ggpaired(data = pdata, id = "Number.x", x = "Manifestation", y = "log2_conc", fill = "Manifestation", palette = 'grey', line.color = "grey", line.size = 0.4) +
  # stat_compare_means(paired = TRUE, label.x = 1.5, geom = "label") +
  # stat_compare_means(comparisons = my_comparisons, group.by="gene", label.y = c(4, 7, 7), label.x = c(1, 2, 3)) +
  scale_fill_manual(values = RColorBrewer::brewer.pal(length(unique(pdata$Manifestation)), "Dark2"), labels = c("Ctrl", "iGBM", "rGBM")) +
  # geom_jitter(aes(label=Number.x)) +
  # ggtitle("Initial vs Recurrence") +
  labs(x="", y="log2( Concentration )") +
  facet_wrap(~gene, scales = "free_y") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="vertical",) +
  guides(fill=guide_legend(title="", ncol=3),
         col='none') -> p1_org

p1_org + 
  stat_pvalue_manual(stat_test, label = "{p.signif} p = {p}", step.group.by = "gene", step.increase = 0.2, hide.ns = T, size=4, bracket.size=1) +
  geom_point(data=data.frame(Manifestation=c("iGBM", "iGBM", "iGBM", "iGBM"), gene=c("ASAH1", "SYNEMIN", "GPNMB", "MMP-9" ), y=c(8, 18, 9, 12)),
             aes(x=Manifestation, y=y, group=gene), alpha=0) -> p1_org

# Exclude outlier patient number.x 54 
ggpaired(data = pdata[Number.x!=54], id = "Number.x", x = "Manifestation", y = "log2_conc", fill = "Manifestation", palette = 'grey', line.color = "grey", line.size = 0.4) +
  # stat_compare_means(paired = TRUE, label.x = 1.5, geom = "label") +
  # stat_compare_means(comparisons = my_comparisons, group.by="gene", label.y = c(4, 7, 7), label.x = c(1, 2, 3)) +
  scale_fill_manual(values = RColorBrewer::brewer.pal(length(unique(pdata$Manifestation)), "Dark2"), labels = c("Ctrl", "iGBM", "rGBM")) +
  # geom_jitter(aes(label=Number.x)) +
  # ggtitle("Initial vs Recurrence") +
  labs(x="", y="log2( Concentration )") +
  facet_wrap(~gene, scales = "free_y") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="vertical",) +
  guides(fill=guide_legend(title="", ncol=3),
         col='none') -> p1

p1 + 
  stat_pvalue_manual(stat_test, label = "{p.signif} p = {p}", step.group.by = "gene", step.increase = 0.2, hide.ns = T, size=4, bracket.size=1) +
  geom_point(data=data.frame(Manifestation=c("iGBM", "iGBM", "iGBM", "iGBM"), gene=c("ASAH1", "SYNEMIN", "GPNMB", "MMP-9" ), y=c(7, 15, 8, 11)),
             aes(x=Manifestation, y=y, group=gene), alpha=0) -> p1
# p1

## Average Conc per ttr
pdata %>%
  dplyr::filter(Manifestation!="Ctrl") %>%
  dplyr::select(gene, Number.x, Manifestation, log2_conc, ttr_group) %>%
  dplyr::mutate(Manifestation = plyr::mapvalues(Manifestation, from=c("iGBM", "rGBM"), to=c("initial\nGBM", "recurrent\nGBM"))) %>%
  dplyr::group_by(gene, Manifestation, ttr_group) %>%
  dplyr::mutate(avg_log2_conc_per_mani_ttr = median(log2_conc, na.rm = T)) -> pdata_ttr

p1_avg_ttr <- ggplot(data = pdata_ttr, aes(x=Manifestation, y=log2_conc)) +
  geom_boxplot() +
  geom_line(aes(x=Manifestation, y=avg_log2_conc_per_mani_ttr, col=ttr_group, group=ttr_group), inherit.aes = FALSE) +
  labs(x="", y="log2( Concentration )") +
  facet_wrap(~gene, scales = "free_y") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="vertical",) +
  guides(fill=guide_legend(title="GBM\nOccurence", ncol=1),
         col=guide_legend(title="Avg. log2 ( concentration )\nTime to Recurrence"))

## Build the table ----
stat_test %>%
  dplyr::mutate(method = plyr::mapvalues(method, from = c("perm.test", "paired.perm.test"), to = c("Permutation Test", "Paired Permutation Test"))) %>%
  dplyr::select(Gene=gene, group1, group2, `Statistical Test`=method, p) %>%
  dplyr::mutate(Comparison = sprintf("%s vs %s", group1, group2)) %>%
  tidyr::pivot_wider(id_cols = c("Comparison", "Statistical Test"), names_from = c(Gene), values_from = "p") %>%
  gt() %>%
  tab_header(title = md("**Statistical Comparisons Between Population Groups per Genes**")) %>%
  cols_label(
    Comparison = md("**Comparison**"),
    `Statistical Test` = md("**Statistical Test**"),
    ASAH1 = md("**ASAH1**"),
    GPNMB = md("**GPNMB**"),
    `MMP-9` = md("**MMP-9**"),
    SYNEMIN = md("**SYNEMIN**"),
  ) %>% 
  tab_footnote(
    footnote = md("Permutation test using t-test statisics for non-paired samples, or paired permutation test using pair t-test statistics for paired samples."),
    locations = cells_column_labels("Statistical Test")
  ) %>% 
  tab_footnote(
    footnote = md("Values in cells represent the non-adjusted p-value from the corresponding statiscal test."),
    locations = cells_column_labels(c("ASAH1", "SYNEMIN", "GPNMB", "MMP-9"))
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(ASAH1),
      rows = ASAH1 < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(GPNMB),
      rows = GPNMB < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(`MMP-9`),
      rows = `MMP-9` < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(SYNEMIN),
      rows = SYNEMIN < 0.05
    )
  ) -> stat_comp_table

## Time to Recurrence ----
train_df_Sub <- pdata %>%
  dplyr::filter( !(Number.x %in% c("32", "33"))) %>%
  dplyr::filter(Manifestation!="Ctrl") %>%
  dplyr::select(Number.x, Time.to.reccurence..days.=`Time-to-reccurence (days)`, class=Manifestation, gene, log2_conc) %>%
  # dplyr::mutate(Time.to.reccurence..days. = ifelse(class=="iGBM", 0, Time.to.reccurence..days.)) %>%
  # dplyr::mutate(class = ifelse(class=="iGBM", 0, 1) ) %>%
  dplyr::rowwise() %>%
  # dplyr::mutate(ttr_group = ttr_categorizer(as.numeric(Time.to.reccurence..days.))) %>%
  # dplyr::mutate(Time.to.reccurence..days. = ttr_categorizer(as.numeric(Time.to.reccurence..days.), 253.5)) %>%
  # dplyr::select(-Time.to.reccurence..days.) %>%
  dplyr::mutate(gene=make.names(gene)) %>%
  tidyr::pivot_wider(id_cols =c("Number.x", "Time.to.reccurence..days.", "class"), names_from = "gene", values_from = "log2_conc") %>%
  # tidyr::pivot_wider(id_cols = c("Number.x", "Time.to.reccurence..days."), names_from = c("class"), values_from = c("ASAH1", "SYNEMIN", "GPNMB", "MMP.9")) %>%
  # dplyr::mutate(ASAH1 = (ASAH1_rGBM+1) /( ASAH1_iGBM+1),
  #               SYNEMIN = (SYNEMIN_rGBM+1)/(SYNEMIN_iGBM+1),
  #               GPNMB = (GPNMB_rGBM+1)/(GPNMB_iGBM+1),
  #               MMP.9 = (MMP.9_rGBM+1)/(MMP.9_iGBM+1)) %>%
  # dplyr::select(Number.x, Time.to.reccurence..days., ASAH1, SYNEMIN, GPNMB, MMP.9 ) %>% 
  # unique() %>%
  dplyr::arrange(Time.to.reccurence..days.)
train_df_Sub <- na.omit(train_df_Sub)
train_df_Sub$Number.x <- as.factor(train_df_Sub$Number.x)




train_df_Sub %>%
  dplyr::group_by(Number.x, Time.to.reccurence..days.) %>%
  dplyr::mutate(ASAH1_delta = ASAH1[class=="rGBM"] - ASAH1[class=="iGBM"],
                SYNEMIN_delta = SYNEMIN[class=="rGBM"] - SYNEMIN[class=="iGBM"],
                GPNMB_delta = GPNMB[class=="rGBM"] - GPNMB[class=="iGBM"],
                MMP.9_delta = MMP.9[class=="rGBM"] - MMP.9[class=="iGBM"]) %>%
  dplyr::rowwise() %>%
  # dplyr::mutate(Time.to.reccurence..days. = ttr_categorizer(as.numeric(Time.to.reccurence..days.))) %>%
  # dplyr::mutate(Time.to.reccurence..days. = ttr_categorizer(as.numeric(Time.to.reccurence..days.), 207.5)) %>%
  dplyr::select(Number.x, Time.to.reccurence..days., ASAH1_delta, SYNEMIN_delta, GPNMB_delta, MMP.9_delta) %>%
  unique() -> ttr_delta

# ttr_delta$Time.to.reccurence..days. <- as.numeric(as.factor(ttr_delta$Time.to.reccurence..days.))
lm_ttr <- lm(Time.to.reccurence..days. ~ ASAH1_delta + SYNEMIN_delta + GPNMB_delta + MMP.9_delta, data=ttr_delta )
summary(lm_ttr)
anova(lm_ttr)


power_calc <- pwr.f2.test(u = 4, f2 = sqrt(summary(lm_ttr)$adj.r.squared), sig.level = 0.05, power = 0.8)
print(sprintf("The estimated sample size needed for 80%% power is: %s", round(power_calc$v) + 5))



ttr_categorizer <- function(data) {
  if (data >= 103 & data < 190){
    return("early")
  } else if (data >= 190 & data < 243){
    return("middle")
  } else if (data >= 243 & data <= 521){
    return("late")
  }
}

train_df_Sub %>%
  dplyr::group_by(Number.x, Time.to.reccurence..days.) %>%
  dplyr::mutate(ASAH1_delta = ASAH1[class=="rGBM"] - ASAH1[class=="iGBM"],
                SYNEMIN_delta = SYNEMIN[class=="rGBM"] - SYNEMIN[class=="iGBM"],
                GPNMB_delta = GPNMB[class=="rGBM"] - GPNMB[class=="iGBM"],
                MMP.9_delta = MMP.9[class=="rGBM"] - MMP.9[class=="iGBM"]) %>%
  dplyr::rowwise() %>%
  dplyr::mutate(Time.to.reccurence..days. = ttr_categorizer(as.numeric(Time.to.reccurence..days.))) %>%
  # dplyr::mutate(Time.to.reccurence..days. = ttr_categorizer(as.numeric(Time.to.reccurence..days.), 207.5)) %>%
  dplyr::select(Number.x, Time.to.reccurence..days., ASAH1_delta, SYNEMIN_delta, GPNMB_delta, MMP.9_delta) %>%
  unique() -> ttr_delta

tmp_pdata <- ttr_delta %>% tidyr::pivot_longer(cols = c(ASAH1_delta, SYNEMIN_delta, GPNMB_delta, MMP.9_delta), names_to="gene", values_to="log2_conc")
tmp_pdata$Time.to.reccurence..days. <- factor(tmp_pdata$Time.to.reccurence..days., levels=c("early", "middle", "late"))
# ggplot(data = tmp_pdata, aes(x=Time.to.reccurence..days., y=log2_conc, col=Time.to.reccurence..days.)) +
#   geom_point() +
#   labs(x="", y="log2( Concentration )") +
#   facet_wrap(~gene, scales = "free_y") +
#   theme_classic() +
#   theme(plot.title = element_text(hjust = 0.5, size=17),
#         axis.title = element_text(size = 17),
#         axis.text = element_text(size=16),
#         strip.text = element_text(size=16),
#         legend.title = element_text(size=16),
#         legend.text = element_text(size=15),
#         legend.position = "bottom", legend.box="vertical",) +
#   guides(fill=guide_legend(title="GBM\nOccurence", ncol=1),
#          col=guide_legend(title="Time to Recurrence"))

# Pre-surgery vs Post-surgery ----
## Build gg dataframe for plotting ----
data_long %>%
  dplyr::filter(is.na(manifestation_type) | patient_state==0) %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(`Collection Type` = plyr::mapvalues(collection_type, from=c("ctrl", "E1", "E2"), to=c("Ctrl", "pre-\nsurgery", "post-\nsurgery"))) %>%
  # dplyr::mutate(`Collection Type` = ifelse(is.na(`Collection Type`), "Ctrl", `Collection Type`)) %>%
  dplyr::mutate(collection_type2 = factor(`Collection Type`, levels = c("Ctrl", "pre-\nsurgery", "post-\nsurgery"))) -> pdata2

## Perform Statistical-Test ----
comparisons <- matrix(c("Ctrl", "Ctrl", "pre-\nsurgery", "pre-\nsurgery", "post-\nsurgery", "post-\nsurgery"), ncol=2)
methods <- c("perm.test", "perm.test", "paired.perm.test")
# methods <- c("perm.test", "perm.test", "perm.test")

stats_frame <- parallel::mclapply(seq_len(nrow(comparisons)), function(iterator){
  comp <- comparisons[iterator, ]
  method <- methods[iterator]
  data_sub <- filter_data(pdata2, comp, method, "collection_type2")
  
  uni_genes <- data_sub %>% dplyr::pull(gene) %>% unique()
  
  gene_stat_frame <- parallel::mclapply(split(data_sub, data_sub$gene), function(.d){
    if (method=="paired.perm.test"){
      .d %>%
        dplyr::select(-gene) %>%
        tidyr::pivot_wider(id_cols = "Number.x", names_from = "collection_type2", values_from = "log2_conc") -> .d_wide
    } else {
      .d -> .d_wide
    }
    message(sprintf("INFO: Computing %s for %s for comparison %s", method, unique(.d$gene), paste(comp, collapse=", ") ))
    do_perm_test(data = .d_wide, gene = .d$gene, response_var = "log2_conc", group1 = comp[1], group2 = comp[2], method = method, grouping_column = "collection_type2")
  }, mc.cores = 4) %>%
    rbindlist() %>%
    unique()
}, mc.cores=3) %>% rbindlist()

stat_test_coord <- pdata2 %>%
  group_by(gene) %>%
  t_test(log2_conc ~ collection_type2) %>%
  add_xy_position() %>%
  dplyr::select(gene, .y., group1, group2, n1, n2, df, y.position, groups, xmin, xmax)

stat_test <- stats_frame %>%
  dplyr::group_by(group1, group2) %>%
  adjust_pvalue(method = "fdr") %>%
  add_significance(p.col="p", output.col = "p.signif",
                   cutpoints = c(0, 1e-04, 0.001, 0.01, 0.05, 1),
                   symbols = c("****", "***", "**", "*", "ns")) %>%
  merge(stat_test_coord, by=c("gene", ".y.", "group1", "group2")) %>%
  dplyr::select(-p.adj) %>%
  dplyr::mutate(p=round(p, 6))

stat_test %>%
  mutate(y.position = c(7.42688, 7.42688, 7.42688, 8.60988, 8.60988, 8.60988, 10.82388, 10.82388, 10.82388, 13.57988, 13.57988, 13.57988)) -> stat_test

stat_test_pre_v_post_surgery <- stat_test

## Build the plot ----
ggpaired(data=pdata2, id = "Number.x", x = "collection_type2", y = "log2_conc", fill = "collection_type2", palette = "jco", line.color = "grey", line.size = 0.4) +
  scale_fill_manual(values = RColorBrewer::brewer.pal(length(unique(pdata2$`collection_type2`)), "Dark2"), labels = c("Ctrl", "pre-surgery", "post-surgery")) +
  # stat_compare_means(paired = TRUE) +
  # ggnewscale::new_scale_color() +
  # geom_hline(data=mean_ctrl, aes(yintercept=mean_conc, col="ctrl"), linetype="dashed") +
  # geom_rect(data=mean_ctrl, aes(xmin=-Inf, xmax=Inf, ymin=mean_conc-sd_conc, ymax=mean_conc+sd_conc), fill="red", alpha=0.1) +
  # scale_color_manual(label="Mean Conc.", values="red") +
  # ggtitle("Pre-Surgery vs Post-Surgery") +
  labs(x="", y="log2( Concentration )") +
  facet_wrap(~gene, scales = "free_y") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="vertical",) +
  guides(fill=guide_legend(title="EDTA Blood\nCollection Type", ncol=3),
         col=guide_legend(title="Healthy Control Baseline")) -> p2

p2 +
  stat_pvalue_manual(stat_test, label = "{p.signif} p = {p}", step.group.by = "gene", step.increase = 0.15, hide.ns = T, size=4, bracket.size=1) +
  geom_point(data=data.frame(Manifestation=c("pre-\nsurgery", "pre-\nsurgery", "pre-\nsurgery", "pre-\nsurgery"), gene=c("ASAH1", "SYNEMIN", "GPNMB", "MMP-9" ), y=c(10, 15, 9, 12)),
             aes(x=Manifestation, y=y, group=gene), alpha=0) -> p2

supp_figure_pre_v_post <- p2

## Build the table ----
stat_test %>%
  dplyr::mutate(method = plyr::mapvalues(method, from = c("perm.test", "paired.perm.test"), to = c("Permutation Test", "Paired Permutation Test"))) %>%
  dplyr::select(Gene=gene, group1, group2, `Statistical Test`=method, p) %>%
  dplyr::mutate(Comparison = sprintf("%s vs %s", group1, group2)) %>%
  tidyr::pivot_wider(id_cols = c("Comparison", "Statistical Test"), names_from = c(Gene), values_from = "p") %>%
  gt() %>%
  tab_header(title = md("**Statistical Comparisons Between Population Groups per Genes**")) %>%
  cols_label(
    Comparison = md("**Comparison**"),
    `Statistical Test` = md("**Statistical Test**"),
    ASAH1 = md("**ASAH1**"),
    GPNMB = md("**GPNMB**"),
    `MMP-9` = md("**MMP-9**"),
    SYNEMIN = md("**SYNEMIN**"),
  ) %>% 
  tab_footnote(
    footnote = md("Permutation test using t-test statisics for non-paired samples, or paired permutation test using pair t-test statistics for paired samples."),
    locations = cells_column_labels("Statistical Test")
  ) %>% 
  tab_footnote(
    footnote = md("Values in cells represent the non-adjusted p-value from the corresponding statiscal test."),
    locations = cells_column_labels(c("ASAH1", "SYNEMIN", "GPNMB", "MMP-9"))
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(ASAH1),
      rows = ASAH1 < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(GPNMB),
      rows = GPNMB < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(`MMP-9`),
      rows = `MMP-9` < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(SYNEMIN),
      rows = SYNEMIN < 0.05
    )
  ) -> stat_pre_post_surgery_comp_table

# Build a master table ----
rbind(stat_test_igm_v_rgm %>%
        dplyr::mutate(method = plyr::mapvalues(method, from = c("perm.test", "paired.perm.test"), to = c("Permutation Test", "Paired Permutation Test"))) %>%
        dplyr::select(Gene=gene, group1, group2, `Statistical Test`=method, p) %>%
        dplyr::mutate(Comparison = sprintf("%s vs %s", group1, group2)) %>%
        tidyr::pivot_wider(id_cols = c("Comparison", "Statistical Test"), names_from = c(Gene), values_from = "p"),
      stat_test_pre_v_post_surgery %>%
        dplyr::mutate(method = plyr::mapvalues(method, from = c("perm.test", "paired.perm.test"), to = c("Permutation Test", "Paired Permutation Test"))) %>%
        dplyr::select(Gene=gene, group1, group2, `Statistical Test`=method, p) %>%
        dplyr::mutate(Comparison = sprintf("%s vs %s", group1, group2)) %>%
        tidyr::pivot_wider(id_cols = c("Comparison", "Statistical Test"), names_from = c(Gene), values_from = "p")) %>%
  gt() %>%
  tab_row_group(
    label = md("**Control vs Initial GBM vs Recurrent GBM Plasma Samples**"),
    rows = 1:3
  ) %>%
  tab_row_group(
    label = md("**Control vs Pre-Surgery vs Post-Surgery Plasma Samples**"),
    rows = 4:6
  ) %>%
  tab_header(title = md("**Statistical Comparisons Between Population Groups per Genes**")) %>%
  cols_label(
    Comparison = md("**Comparison**"),
    `Statistical Test` = md("**Statistical Test**"),
    ASAH1 = md("**ASAH1**"),
    GPNMB = md("**GPNMB**"),
    `MMP-9` = md("**MMP-9**"),
    SYNEMIN = md("**SYNEMIN**"),
  ) %>% 
  tab_footnote(
    footnote = md("Permutation test using t-test statisics for non-paired samples, or paired permutation test using pair t-test statistics for paired samples."),
    locations = cells_column_labels("Statistical Test")
  ) %>% 
  tab_footnote(
    footnote = md("Values in cells represent the non-adjusted p-value from the corresponding statistical test."),
    locations = cells_column_labels(c("ASAH1", "SYNEMIN", "GPNMB", "MMP-9"))
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(ASAH1),
      rows = ASAH1 < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(GPNMB),
      rows = GPNMB < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(`MMP-9`),
      rows = `MMP-9` < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(SYNEMIN),
      rows = SYNEMIN < 0.05
    )
  ) -> stat_comp_table


# Generate plot of rGBM to iGBM Ratios ----

data_long %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::add_count() %>%
  dplyr::filter(n==2) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>%
      dplyr::group_by(gene, Number.x, patient_state, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`) %>%
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) -> data_long_igm_rgm_ratios

data_long_igm_rgm_ratios %>%
  dplyr::filter(sample_comparison!="ctrl") %>%
  dplyr::filter(sample_comparison!="pre-surgery/post-surgery") %>%
  dplyr::group_by(gene, sample_comparison) %>%
  dplyr::mutate(myaxis = paste0(sample_comparison, "\n", "n=", n())) %>%
  ggplot() +
  # geom_violin(aes(x=as.factor(sample_comparison), y=log2(ratio), fill=as.factor(sample_comparison)), width=0.9) +
  geom_boxplot(aes(x=as.factor(gene), y=log2(ratio), fill=as.factor(gene)), width=0.3, alpha=0.9) +
  scale_fill_manual(values = RColorBrewer::brewer.pal(4, "Dark2")) +
  # ggnewscale::new_scale_color() +
  # geom_hline(data=mean_ctrl, aes(yintercept=mean_conc, col="ctrl"), linetype="dashed") +
  # geom_hline(data=mean_ctrl, aes(yintercept=mean_conc+sd_conc, col="ctrl"), linetype="solid", alpha=0.4) +
  # geom_hline(data=mean_ctrl, aes(yintercept=mean_conc-sd_conc, col="ctrl"), linetype="solid", alpha=0.4) +
  # geom_rect(data=mean_ctrl, aes(xmin=-Inf, xmax=Inf, ymin=mean_conc-sd_conc, ymax=mean_conc+sd_conc), fill="red", alpha=0.1) +
  # scale_color_manual(label="Mean Conc.", values="red") +
  geom_jitter(aes(x=as.factor(gene), y=log2(ratio), col=as.factor(Number.x)), alpha=0.7) +
  # ggtitle("iGBM / rGBM Ratios") +
  labs(x="", y=expression(paste("log2 ", bgroup("(", frac('initial GBM', 'recurrent GBM'), ")"), " Ratio"))) +
  # facet_wrap(~gene) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="vertical") +
  guides(fill="none", col="none") -> p3

supp_figure_ratios <- p3

data_long %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::add_count() %>%
  dplyr::filter(n==2) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>%
      dplyr::group_by(gene, Number.x, patient_state, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`) %>%
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Second manifestation"]
      .d_second <- .d[manifestation_type=="Initial manifestation"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "rGBM/iGBM", "pre-surgery/post-surgery"))) -> data_long_rgm_igm_ratios

data_long_rgm_igm_ratios %>%
  dplyr::filter(sample_comparison!="ctrl") %>%
  dplyr::filter(sample_comparison!="pre-surgery/post-surgery") %>%
  dplyr::group_by(gene, sample_comparison) %>%
  dplyr::mutate(myaxis = paste0(sample_comparison, "\n", "n=", n())) %>%
  ggplot() +
  # geom_violin(aes(x=as.factor(sample_comparison), y=log2(ratio), fill=as.factor(sample_comparison)), width=0.9) +
  geom_boxplot(aes(x=as.factor(gene), y=log2(ratio), fill=as.factor(gene)), width=0.3, alpha=0.9) +
  scale_fill_manual(values = RColorBrewer::brewer.pal(4, "Dark2")) +
  # ggnewscale::new_scale_color() +
  # geom_hline(data=mean_ctrl, aes(yintercept=mean_conc, col="ctrl"), linetype="dashed") +
  # geom_hline(data=mean_ctrl, aes(yintercept=mean_conc+sd_conc, col="ctrl"), linetype="solid", alpha=0.4) +
  # geom_hline(data=mean_ctrl, aes(yintercept=mean_conc-sd_conc, col="ctrl"), linetype="solid", alpha=0.4) +
  # geom_rect(data=mean_ctrl, aes(xmin=-Inf, xmax=Inf, ymin=mean_conc-sd_conc, ymax=mean_conc+sd_conc), fill="red", alpha=0.1) +
  # scale_color_manual(label="Mean Conc.", values="red") +
  geom_jitter(aes(x=as.factor(gene), y=log2(ratio), col=as.factor(Number.x)), alpha=0.7) +
  # ggtitle("rGBM / iGBM Ratios") +
  labs(x="", y=expression(paste("log2 ", bgroup("(", frac('recurrent GBM', 'initial GBM'), ")"), " Ratio"))) +
  # facet_wrap(~gene) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="vertical") +
  guides(fill="none", col="none") -> p3

supp_figure_ratios2 <- p3

# Master figure ----
# subfigure_6 <- ggarrange(p1, p2, p3, ncol = 3)
subfigure_6 <- p1
# subfigure_6
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_a.png", plot = subfigure_6, width = 21, height = 11, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_a.svg", plot = subfigure_6, width = 21, height = 11, dpi = 200)

subfigure_6_org <- p1_org

p1_avg_ttr
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_a_avg.png", plot = p1_avg_ttr, width = 21, height = 11, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_a_avg.svg", plot = p1_avg_ttr, width = 21, height = 11, dpi = 200)
```

<!-- # Dimensionality Reduction -->

```{r echo=FALSE, fig.height=11, fig.width=21, message=FALSE, warning=FALSE}
# Dimensionality Reduction ----
## Data wrnagling ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type %in% c("Initial manifestation", "Second manifestation") | sample_comparison=="ctrl") %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(0,1), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(class = ifelse(manifestation_type=="Initial manifestation", "iGBM", class)) %>%
  dplyr::mutate(class = ifelse(manifestation_type=="Second manifestation", "rGBM", class)) %>%
  dplyr::mutate(class = ifelse(is.na(manifestation_type), "Ctrl", class)) -> pdata

pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction ----
data_wide %>%
  dplyr::mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide

## sPLS-DA ----
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)

MyResult.splsda <- mixOmics::splsda(train_df[, -1], as.factor(train_df[, 1]), keepX = c(4,4)) 

## Predict boundary planes of separation ----
background <- background.predict(MyResult.splsda, comp.predicted=2, dist = "max.dist") 

## Prepare data frame for plotting ----
plsda_data <- cbind( patient_id=na.omit(data_wide)  %>% dplyr::pull(Number.x), class=as.factor(train_df[, 1]), as.data.frame(MyResult.splsda$variates$X))
plsda_data$patient_id[plsda_data$class=="Ctrl"] <- "Ctrl"
plsda_data$patient_id <- as.factor(plsda_data$patient_id)
plsda_data %>%
  dplyr::arrange(class) -> plsda_data
plsda_data <- as.data.table(plsda_data)

## Get average trajectories of iGBM to rGBM in low dimension ----
plsda_data[patient_id!="Ctrl" ] %>%
  dplyr::group_by(class) %>%
  dplyr::add_count() %>%
  dplyr::group_by(class, n) %>%
  dplyr::summarise(avg_comp1=sum(comp1)/n,
                   avg_comp2=sum(comp2)/n) %>%
  dplyr::ungroup() %>%
  unique() %>%
  dplyr::mutate(patient_id="avg_all") %>%
  as.data.table() -> average_comp

slope <- (average_comp$avg_comp2[2] - average_comp$avg_comp2[1])/(average_comp$avg_comp1[2] - average_comp$avg_comp1[1])
b = -0.25

average_comp[class=='rGBM']$avg_comp2 = slope*-1.5 + b
average_comp[class=='rGBM']$avg_comp1 = -1.5
average_comp[class=='iGBM']$avg_comp2 = slope*3 + b
average_comp[class=='iGBM']$avg_comp1 = 3

## Prepare decision boundaries dataframe fo plotting ----
for(i in 1:length(background))
{
  if(!is.null(background[[i]]))
    background[[i]]=data.frame(id=i,col=names(background)[i], background[[i]])
}

background = do.call(rbind,background)

## Build plot ----
### Average Trajectories
p <- ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c( "#C2C2C2", "#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#C2C2C2", "#388ECC", "#F68B33")) +
  ggnewscale::new_scale_color() +
  geom_path(data=average_comp, aes(x=avg_comp1, y=avg_comp2, group = patient_id, col="Avg. iGBM\nto rGBM\nTrajectory"),
            arrow = arrow(type = "closed",
                          length=unit(0.1, "inches")), size=2, alpha=0.8) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(shape="none",
         col_org=guide_legend(title="g", ncol=1),
         col=guide_legend(title="Trajectory", ncol=1),
         col_new=guide_legend(title="g", ncol=1))

splsda_avg_traj_fig <- p

# ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b.png", plot = splsda_avg_traj_fig, width = 11, height = 8.5, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b.svg", plot = splsda_avg_traj_fig, width = 11, height = 8.5, dpi = 200)

### Individual trajectories
p <- ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c( "#C2C2C2", "#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#C2C2C2", "#388ECC", "#F68B33")) +
  ggnewscale::new_scale_color() +
  geom_path(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, group = patient_id, col=patient_id),
            arrow = arrow(type = "closed",
                          length=unit(0.1, "inches")), size=2, alpha=0.8) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(shape="none",
         col_org=guide_legend(title="g", ncol=1),
         col=guide_legend(title="Patient ID\nTrajectory", ncol=2),
         col_new=guide_legend(title="g", ncol=1))

splsda_ind_traj_fig <- p

# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig2_subfigure_a.png", plot = splsda_ind_traj_fig, width = 11, height = 8.5, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig2_subfigure_a.svg", plot = splsda_ind_traj_fig, width = 11, height = 8.5, dpi = 200)

```

<!-- # Ratios -->

```{r echo=FALSE, fig.height=11, fig.width=21, message=FALSE, warning=FALSE}
## Data wrangling ----
### iGBM over rGBM ratio
data_long %>% 
  dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBMoverrGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBMoverrGBM" ) %>%
  dplyr::mutate(log2_conc = log2(ratio)) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("iGBMoverrGBM"))) -> pdata

### rGBM over iGBM
data_long %>% 
  dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
  dplyr::filter(!is.na(manifestation_type) | patient_state!=0) %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_second$conc / .d_initial$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "rGBMoveriGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="rGBMoveriGBM" ) %>%
  dplyr::filter(grepl(";", agg_lab_id)) %>%
  dplyr::mutate(log2_conc = log2(ratio)) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("rGBMoveriGBM"))) -> pdata2

### Prepare final table with ratios
pdata %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="iGBM") -> pdata

pdata2 %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="rGBM") -> pdata2


pdata <- rbindlist(list(pdata, pdata2))

pdata <- merge(pdata, meta_dt, by="Number.x")

### Pivot Table to wide
pdata %>%
  dplyr::select(Number.x, class, ASAH1, SYNEMIN, GPNMB, `MMP-9`, `Time-to-reccurence (days)`, `Latency (days)`, `Type of resection`) -> data_wide

# Prepare final table for analysis
train_df <- data_wide
train_df <- na.omit(train_df)  
train_df <- train_df[ !is.infinite(train_df$ASAH1) , ]
# Make Valid Column Names 
colnames(train_df) <- make.names(colnames(train_df))
train_df$class <- factor(train_df$class, levels=c("iGBM", "rGBM"))
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.table(train_df)
train_df$group=0

## Plot Protein Ratios Against Each Other ----
p_pa_pb_ratio_scatter1 <- ggplot(train_df, aes(x=ASAH1, y=GPNMB, col=class, shape=as.factor(Number.x))) +
  geom_point(size=3) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  labs(x="ASAH1 Ratio", y="GPNMB Ratio") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="horizontal",) +
  guides(shape=guide_legend(title="Patient ID", ncol=5),
         colour_new=guide_legend(title="Class", ncol=1, byrow = T),
         col=guide_legend(title="Class", ncol=2, byrow = T))

p_pa_pb_ratio_scatter2 <- ggplot(train_df, aes(x=ASAH1, y=MMP.9, col=class, shape=as.factor(Number.x))) +
  geom_point(size=3) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  labs(x="ASAH1 Ratio", y="MMP-9 Ratio") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="horizontal",) +
  guides(shape=guide_legend(title="Patient ID", ncol=5),
         colour_new=guide_legend(title="Class", ncol=1, byrow = T),
         col=guide_legend(title="Class", ncol=2, byrow = T))

p_pa_pb_ratio_scatter3 <- ggplot(train_df, aes(x=ASAH1, y=SYNEMIN, col=class, shape=as.factor(Number.x))) +
  geom_point(size=3) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  labs(x="ASAH1 Ratio", y="SYNEMIN Ratio") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="horizontal",) +
  guides(shape=guide_legend(title="Patient ID", ncol=5),
         colour_new=guide_legend(title="Class", ncol=1, byrow = T),
         col=guide_legend(title="Class", ncol=2, byrow = T))

p_pa_pb_ratio_scatter4 <- ggplot(train_df, aes(x=MMP.9, y=SYNEMIN, col=class, shape=as.factor(Number.x))) +
  geom_point(size=3) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  labs(x="MMP-9 Ratio", y="SYNEMIN Ratio") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="horizontal",) +
  guides(shape=guide_legend(title="Patient ID", ncol=5),
         colour_new=guide_legend(title="Class", ncol=1, byrow = T),
         col=guide_legend(title="Class", ncol=2, byrow = T))

p_pa_pb_ratio_scatter5 <- ggplot(train_df, aes(x=GPNMB, y=SYNEMIN, col=class, shape=as.factor(Number.x))) +
  geom_point(size=3) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  labs(x="GPNMB Ratio", y="SYNEMIN Ratio") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="horizontal",) +
  guides(shape=guide_legend(title="Patient ID", ncol=5),
         colour_new=guide_legend(title="Class", ncol=1, byrow = T),
         col=guide_legend(title="Class", ncol=2, byrow = T))

p_pa_pb_ratio_scatter6 <- ggplot(train_df, aes(x=GPNMB, y=MMP.9, col=class, shape=as.factor(Number.x))) +
  geom_point(size=3) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  labs(x="GPNMB Ratio", y="MMP-9 Ratio") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="horizontal",) +
  guides(shape=guide_legend(title="Patient ID", ncol=5),
         colour_new=guide_legend(title="Class", ncol=1, byrow = T),
         col=guide_legend(title="Class", ncol=2, byrow = T))

p_pa_pb_ratio_scatter <- ggpubr::ggarrange(p_pa_pb_ratio_scatter1, p_pa_pb_ratio_scatter2, p_pa_pb_ratio_scatter3, 
                                           p_pa_pb_ratio_scatter4, p_pa_pb_ratio_scatter5, p_pa_pb_ratio_scatter6,
                                           ncol=3, nrow=2, common.legend = TRUE, legend="bottom")

# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure3_subfigure_a.png", plot = p_pa_pb_ratio_scatter, width = 15, height = 9, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure3_subfigure_a.svg", plot = p_pa_pb_ratio_scatter, width = 15, height = 9, dpi = 200)

## sPLS-DA of Ratios ----

org_class <- train_df$class
train_df$class <- sample(train_df$class)

MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2, -7, -8, -9, -10)], train_df$class, keepX = c(4,4)) 
MyResult.splsda <- mixOmics::splsda(train_df[, c(-1)], train_df$class, keepX = c(4,4))
background <- background.predict(MyResult.splsda, comp.predicted=2, dist = "max.dist") 

plsda_data <- cbind( patient_id=na.omit(data_wide %>% dplyr::filter(Number.x!=32))  %>% dplyr::pull(Number.x), class=train_df$class, as.data.frame(MyResult.splsda$variates$X))
plsda_data$patient_id <- as.factor(plsda_data$patient_id)
plsda_data %>%
  dplyr::arrange(class) -> plsda_data
plsda_data <- as.data.table(plsda_data)

plsda_data %>%
  dplyr::group_by(class) %>%
  dplyr::add_count() %>%
  dplyr::group_by(class, n) %>%
  dplyr::summarise(avg_comp1=sum(comp1)/n,
                   avg_comp2=sum(comp2)/n) %>%
  dplyr::ungroup() %>%
  unique() %>%
  dplyr::mutate(patient_id="avg_all") %>%
  as.data.table() -> average_comp

slope <- (average_comp$avg_comp2[2] - average_comp$avg_comp2[1])/(average_comp$avg_comp1[2] - average_comp$avg_comp1[1])
b = -0.25

for(i in 1:length(background))
{
  if(!is.null(background[[i]]))
    background[[i]]=data.frame(id=i,col=names(background)[i],
                               background[[i]])
}

background = do.call(rbind,background)

p <- ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c("#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  # ggnewscale::new_scale_color() +
  # ggrepel::geom_label_repel(data=merge(plsda_data %>% dplyr::mutate(patient_id = as.numeric(as.character(patient_id))), unique(data_long[manifestation_type!='' & patient_state==1 & `Time-to-reccurence (days)`!='', .(Number.x, `Time-to-reccurence (days)`)]), by.x='patient_id', by.y="Number.x", all.x=T), aes(x=comp1, y=comp2, col=`Time-to-reccurence (days)`, label=paste(patient_id, `Time-to-reccurence (days)`, sep=": "))) +
  # ggnewscale::new_scale_color() +
  # geom_path(data=average_comp, aes(x=avg_comp1, y=avg_comp2, group = patient_id, col="Avg. iGBM\nto rGBM\nTrajectory"),
  #           arrow = arrow(type = "closed", ends='both',
  #                         length=unit(0.1, "inches")), size=2, alpha=0.8) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(shape="none",
         colour_new_new=guide_legend(title="Class", ncol=1, byrow = T),
         colour_new=guide_legend(title="Class", ncol=1, byrow = T),
         col=guide_legend(title="Class", ncol=1, byrow = T))
  # guides(shape="none",
  #        colour_new_new=guide_legend(title="Class", ncol=1, byrow = T),
  #        colour_new='none',
  #        col=guide_legend(title="Trajectory", ncol=1))

splsda_ratio_avg_traj_fig <- p

background %>%
  dplyr::filter(Var2 < 1.59058 & Var2 > -1.56044 & Var1 > -2.5462 & Var1 < 2.36565) -> background_line

lm_eq <- lm(Var2 ~ Var1, data=background_line)
coef(lm_eq)

plsda_data %>%
  dplyr::rowwise() %>%
  dplyr::mutate(new_x = (comp1 + -1/coef(lm_eq)[2]*comp2  - (-1/coef(lm_eq)[2]*(-coef(lm_eq)[1])))/(1+(-1/coef(lm_eq)[2])^2),
                new_y = (-1/coef(lm_eq)[2]*comp1 + ((-1/coef(lm_eq)[2])^2)*comp2  + (-coef(lm_eq)[1]))/(1+(-1/coef(lm_eq)[2])^2)) -> plsda_data_proj

ggplot() +
  # geom_point(data=background_line, aes(x=Var1, y=Var2)) +
  geom_abline(slope = coef(lm_eq)[2], intercept = coef(lm_eq)[1]) +
  geom_abline(slope = -1/coef(lm_eq)[2], intercept = -coef(lm_eq)[1]) +
  geom_point(data=plsda_data_proj, aes(x=new_x, y=new_y, col=class)) 

ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c("#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3, alpha=0.45) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  # ggnewscale::new_scale_color() +
  # ggrepel::geom_label_repel(data=merge(plsda_data %>% dplyr::mutate(patient_id = as.numeric(as.character(patient_id))), unique(data_long[manifestation_type!='' & patient_state==1 & `Time-to-reccurence (days)`!='', .(Number.x, `Time-to-reccurence (days)`)]), by.x='patient_id', by.y="Number.x", all.x=T), aes(x=comp1, y=comp2, col=`Time-to-reccurence (days)`, label=`Time-to-reccurence (days)`)) +
  ggnewscale::new_scale_color() +
  ggnewscale::new_scale('shape') +
  # geom_path(data=average_comp, aes(x=avg_comp1, y=avg_comp2, group = patient_id, col="Avg. iGBM\nto rGBM\nTrajectory"),
  #           arrow = arrow(type = "closed", ends='both',
  #                         length=unit(0.1, "inches")), size=2, alpha=0.8) +
  geom_abline(slope = -1/coef(lm_eq)[2], intercept = -coef(lm_eq)[1]) +
  geom_point(data=plsda_data_proj, aes(x=new_x, y=new_y, col=class, shape=class), size=3)  +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(shape="none",
         shape_new="none",
         colour_new="none",
         col=guide_legend(title="Class", ncol=1))

ggplot() +
  geom_point(data=merge(plsda_data_proj %>% dplyr::filter(class=='rGBM') %>% dplyr::mutate(patient_id = as.numeric(as.character(patient_id))), unique(data_long[manifestation_type!='' & patient_state==1 & `Time-to-reccurence (days)`!='', .(Number.x, `Time-to-reccurence (days)`)]), by.x='patient_id', by.y="Number.x", all.x=T), aes(x=class, y=-new_x, col=`Time-to-reccurence (days)`)) +
  ggrepel::geom_label_repel(data=merge(plsda_data_proj %>% dplyr::filter(class=='rGBM') %>% dplyr::mutate(patient_id = as.numeric(as.character(patient_id))), unique(data_long[manifestation_type!='' & patient_state==1 & `Time-to-reccurence (days)`!='', .(Number.x, `Time-to-reccurence (days)`)]), by.x='patient_id', by.y="Number.x", all.x=T), aes(x=class, y=-new_x, col=`Time-to-reccurence (days)`, label=`Time-to-reccurence (days)`)) +
  labs(y="Projected X-variate 1", x=NULL) +
  scale_x_discrete(labels=c(expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(col='none')



# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure3_subfigure_b.png", plot = splsda_ratio_avg_traj_fig, width = 11, height = 8.5, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure3_subfigure_b.svg", plot = splsda_ratio_avg_traj_fig, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b.png", plot = splsda_ratio_avg_traj_fig, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b.svg", plot = splsda_ratio_avg_traj_fig, width = 11, height = 8.5, dpi = 200)

p <- ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c( "#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  ggnewscale::new_scale_color() +
  geom_path(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, group = patient_id, col=patient_id),
            arrow = arrow(type = "closed", ends='both',
                          length=unit(0.1, "inches")), size=1.5, alpha=0.8) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(shape="none",
         colour_new=guide_legend(title="Class", ncol=1, byrow = T),         
         col=guide_legend(title="Patient ID\nTrajectory", ncol=2))

splsda_ratio_ind_traj_fig <- p

# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure3_subfigure_c.png", plot = splsda_ratio_ind_traj_fig, width = 11, height = 8.5, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure3_subfigure_c.svg", plot = splsda_ratio_ind_traj_fig, width = 11, height = 8.5, dpi = 200)

## Time to Recurrence using Ratios ----
train_df_Sub <- train_df[class=="rGBM"] %>%
  dplyr::arrange(Time.to.reccurence..days.)

train_df_Sub %>%
  tidyr::pivot_longer(cols = c("ASAH1", "SYNEMIN", "GPNMB", "MMP.9"), names_to = "gene", values_to = "ratio") %>%
  dplyr::group_by(gene, Time.to.reccurence..days.) %>%
  # dplyr::summarise(ratio=mean(ratio)) %>%
  dplyr::mutate(Time.to.reccurence..days. = as.numeric(Time.to.reccurence..days.)) %>%
  dplyr::ungroup() -> train_df_Sub_long

p_ratio_ttr <- ggplot(train_df_Sub_long) +
  geom_point(aes(x=as.numeric(Time.to.reccurence..days.), y=ratio, group=gene, col=as.factor(Number.x)), size=3) +
  ggnewscale::new_scale("color") +
  # geom_line(aes(x=as.factor(Time.to.reccurence..days.), y=ratio, group=gene, col=gene)) +
  geom_smooth(aes(x=as.numeric(Time.to.reccurence..days.), y=ratio, group=gene, col=gene), method = "lm", alpha=0.2) +
  stat_cor(method = "pearson", aes(x=as.numeric(Time.to.reccurence..days.), y=ratio, group=gene)) +
  labs(x="Time to Recurrence (Days)", y=expression(paste("log2 ", bgroup("(", frac('recurrent GBM', 'initial GBM'), ")"), " Ratio"))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical") +
  guides(col=guide_legend(title="Gene"), 
         colour_new=guide_legend(title="Patient ID", ncol = 2)) +
  facet_wrap(~gene)

# train_df_Sub <- train_df %>%
#   dplyr::select(Number.x, Time.to.reccurence..days., class, ASAH1, SYNEMIN, GPNMB, MMP.9) %>%
#   tidyr::pivot_wider(id_cols = c("Number.x", "Time.to.reccurence..days."), names_from = c("class"), values_from = c("ASAH1", "SYNEMIN", "GPNMB", "MMP.9")) %>%
#   dplyr::arrange(Time.to.reccurence..days.)
#   train_df_Sub <- na.omit(train_df_Sub)
# 
# train_df_Sub$Number.x <- as.factor(train_df_Sub$Number.x)
# train_df_Sub$Time.to.reccurence..days. <- as.numeric(train_df_Sub$Time.to.reccurence..days.)
# lm_ttr <- lmerTest::lmer(Time.to.reccurence..days. ~ ASAH1_iGBM + ASAH1_rGBM + SYNEMIN_iGBM + SYNEMIN_rGBM + GPNMB_iGBM + GPNMB_rGBM + MMP.9_iGBM + MMP.9_rGBM + (1 | Type.of.resection), data=train_df_Sub )
# summary(lm_ttr)
# anova(lm_ttr)
# 
# lm_ttr <- lm(Time.to.reccurence..days. ~ ASAH1_iGBM + ASAH1_rGBM + SYNEMIN_iGBM + SYNEMIN_rGBM + GPNMB_iGBM + GPNMB_rGBM + MMP.9_iGBM + MMP.9_rGBM, data=train_df_Sub )
# summary(lm_ttr)
# car::Anova(lm_ttr, type="III")

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_e.png", plot = p_ratio_ttr, width = 15, height = 9, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_e.svg", plot = p_ratio_ttr, width = 15, height = 9, dpi = 200)

```

# Subsetting and hold out patient for Validating Ratios

I still think there is value in usiing the ratios, so I performed the sPLS-DA on the ratio data again, but instead of using all the ratios and inverse ratios from all patients, I do the following:

1. Randomly sample a single patient to use as a test patient, to make predictions on
2. Randomly sample n random patients from remaining patients and compute iGBM/rGBM ratios
3. For the rest of the patients I compute rGBM/iGBM ratios

This results in the following data used in the sPLS-DA model

**Training Data**

| Patient ID | Class | ASAH1      | SYNMEIN    | GPNMB      | MMP-9      |
|------------|-------|------------|------------|------------|------------|
| 6          | iGBM  | -4.3219281 | 2.6629410  | 0.3162515  | 3.8886097  |
| 10         | rGBM  | 2.2103599  | -3.8779037 | 2.5628539  | -3.1538943 |
| 16         | rGBM  | 1.6956066  | 0.8506559  | 0.1659252  | -1.9814540 |
| 23         | rGBM  | -0.0257631 | -3.6121574 | 1.6151451  | -1.7166787 |
| 26         | iGBM  | 1.7744403  | 0.4268049  | 4.2786146  | -1.5475206 |
| 29         | rGBM  | 3.3645724  | 0.4959143  | -0.6652013 | 0.5074684  |
| 31         | iGBM  | -1.6599246 | -2.8570919 | -1.9497226 | -1.3115642 |
| 34         | iGBM  | -0.1768778 | 0.7865737  | 2.1875247  | 1.1604031  |

**Test Data**

| Patient ID | Class | ASAH1      | SYNMEIN    | GPNMB      | MMP-9      |
|------------|-------|------------|------------|------------|------------|
| 35          | rGBM  | -0.02599521 | -4.115453  | -2.892362  | -1.679008  |

This way we do not duplicate patient IDs (i.e. having iGBM/rGBM and rGBM/iGBM for the same patient) and we have a single patient we can make a prediction on to see where in the PLS-DA dimension they would land.

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=11, fig.width=21, include=FALSE}
data_long2 <- data_long

## Data wrangling ----
# Get ratio data
message("Get Ratio Data")
ratio_data <- get_ratio_data(data_long2)
message("Get Test Train Data")
train_test <- get_train_test_for_loocv(ratio_data, test_patient_id=NULL, unique_random_train_patient_per_class=T)
train_df <- train_test$train_df
test_df <- train_test$test_df
test_patients <- test_df$Number.x
train_df <- rbindlist(list(train_df, test_df))

## sPLS-DA of Ratios ----
message("Perform sPLS-DA")
MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2, -7, -8, -9, -10)], train_df$class, keepX = c(4,4)) 
MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2)], train_df$class, keepX = c(4,4)) 

pred_test <- predict(MyResult.splsda, test_df[, c(-1, -2, -7, -8, -9, -10)], test_df$class)
pred_test <- predict(MyResult.splsda, test_df[, c(-1, -2)], test_df$class)
message("Perform background predictions")
background <- background.predict(MyResult.splsda, comp.predicted=2, dist = "max.dist", xlim = c(min(c(MyResult.splsda$variates$X, pred_test$variates)), max(c(MyResult.splsda$variates$X, pred_test$variates))), ylim = c(min(c(MyResult.splsda$variates$X, pred_test$variates)), max(c(MyResult.splsda$variates$X, pred_test$variates))))
background <- background.predict(MyResult.splsda, comp.predicted=2, dist = "max.dist", xlim = c(min(c(MyResult.splsda$variates$X)), max(c(MyResult.splsda$variates$X))), ylim = c(min(c(MyResult.splsda$variates$X)), max(c(MyResult.splsda$variates$X))))
message("Prepare data for plotting")
plsda_data <- cbind( patient_id=train_df  %>% dplyr::pull(Number.x), class=train_df$class, as.data.frame(MyResult.splsda$variates$X))
plsda_data$patient_id <- as.factor(plsda_data$patient_id)
plsda_data %>%
  dplyr::arrange(class) -> plsda_data
plsda_data <- as.data.table(plsda_data)

plsda_data %>%
  dplyr::group_by(class) %>%
  dplyr::add_count() %>%
  dplyr::group_by(class, n) %>%
  dplyr::summarise(avg_comp1=sum(comp1)/n,
                   avg_comp2=sum(comp2)/n) %>%
  dplyr::ungroup() %>%
  unique() %>%
  dplyr::mutate(patient_id="avg_all") %>%
  as.data.table() -> average_comp

slope <- (average_comp$avg_comp2[2] - average_comp$avg_comp2[1])/(average_comp$avg_comp1[2] - average_comp$avg_comp1[1])
b = -0.25

for(i in 1:length(background))
{
  if(!is.null(background[[i]]))
    background[[i]]=data.frame(id=i,col=names(background)[i],
                               background[[i]])
}

background = do.call(rbind,background)

message("Gnerate sPLS-DA Plot")
p <- ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c("#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  # ggnewscale::new_scale_color() +
  # geom_path(data=average_comp, aes(x=avg_comp1, y=avg_comp2, group = patient_id, col="Avg. iGBM\nto rGBM\nTrajectory"),
            # arrow = arrow(type = "closed", ends='both',
                          # length=unit(0.1, "inches")), size=2, alpha=0.8) +
  # ggnewscale::new_scale_color() +
  # geom_point(data=data.frame(x=pred_test$variates[,1], y=pred_test$variates[,2]), aes(x=x, y=y, col="black"), size=3, shape=3) +
  # scale_color_manual(values=c( "black"), labels=c(expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))) +
  ggnewscale::new_scale_color() +
  ggrepel::geom_label_repel(data=merge(plsda_data %>% dplyr::mutate(patient_id = as.numeric(as.character(patient_id))), unique(data_long[manifestation_type!='' & patient_state==1 & `Time-to-reccurence (days)`!='', .(Number.x, `Time-to-reccurence (days)`)]), by.x='patient_id', by.y="Number.x", all.x=T), aes(x=comp1, y=comp2, col=`Time-to-reccurence (days)`, label=`Time-to-reccurence (days)`)) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
   guides(shape="none",
         colour_new_new=guide_legend(title="Class", ncol=1, byrow = T),
         colour_new=guide_legend(title="Class", ncol=1, byrow = T),
         col='none')
  # guides(shape="none",
  #        colour_new_new=guide_legend(title="Class", ncol=1, byrow = T),
  #        colour_new=guide_legend(title="Trajectory", ncol=1, byrow = T),
  #        col=guide_legend(title="Prediction on\nHeld Out Patients", ncol=1))

splsda_better_ratio_avg_traj_fig <- p
splsda_better_ratio_avg_traj_fig
ggsave("../../figures/elisa_plasma_proteomics/elisa_splada_ratio_random.png", plot = splsda_better_ratio_avg_traj_fig, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_splada_ratio_random.svg", plot = splsda_better_ratio_avg_traj_fig, width = 11, height = 8.5, dpi = 200)

```

# Comparison of Using All patients or Subsetting for Unique Patients per Class

Lets compare to see if it makes a difference whether we use all patients for both iGBM/rGBM and rGBM/iGBM classes vs only using a unique set of patients for iGBM/rGBM and rGBM/iGBM (i.e. no duplicate patient in each class)

1. I perform a leave one out cross validation, leaving one patient out and using the rest of the data for training. 
2. The left out patient gets using for assessing model performance accuracy. 
3. The mean accuracy of each left out sample is computed
4. Steps 1-3 is repeated 100 times to assess the variation in mean accuracy of the LOOCV
5. 1-4 is performed for both cases of using all patients and using unique subset patients

```{r echo=FALSE, fig.height=8.5, fig.width=11, message=FALSE, warning=FALSE}
# Get ratio data
ratio_data <- get_ratio_data(data_long2)
ratio_data <- data_wide
colnames(ratio_data) <- make.names(colnames(ratio_data))
ratio_data$class <- factor(ratio_data$class, levels=c("iGBM", "rGBM"))
ratio_data <- ratio_data[Number.x!=32]
# Get unique set of patient ids
patient_ids <- unique(ratio_data$Number.x)

plan(multisession(workers = 8))

# Compute mean accuracy when using unique patients per class for 100 repeats
mean_acc_rep_unique <- future.apply::future_replicate(100, splsda_loocv_perf(ratio_data, patient_ids, unique_random_train_patient_per_class=TRUE))
# hist(mean_acc_rep_unique)
message("Unique Patient Class Split: 95% Confidence Interval:")
print(c(quantile(mean_acc_rep_unique, .025), quantile(mean_acc_rep_unique, .975)))

# Compute mean accuracy when using all patients per class for 100 repeats
mean_acc_rep_all <- future.apply::future_replicate(100, splsda_loocv_perf(ratio_data, patient_ids, unique_random_train_patient_per_class=FALSE))
# hist(mean_acc_rep_all)
message("All Patients per Class: 95% Confidence Interval:")
print(c(quantile(mean_acc_rep_all, .025), quantile(mean_acc_rep_all, .975)))

plan(sequential)

df_plot <- rbind(data.frame(method="Unique Patient Class Split", mean_acc=mean_acc_rep_unique), data.frame(method="All Patients per Class", mean_acc=mean_acc_rep_all))
df_plot$method <- factor(df_plot$method, levels = c("All Patients per Class", "Unique Patient Class Split"))

ggplot(df_plot, aes(x=mean_acc, fill=method)) +
  geom_histogram(binwidth = 0.05, colour = "black") +
  scale_x_continuous(breaks = seq(0, 1, by = 0.2)) +
  scale_fill_brewer(palette = "Dark2") +
  stat_bin(binwidth = 0.05, aes(label=..count..), vjust=-0.5, geom = "text") +
  facet_wrap(~method) +
  labs(x="Mean Accuracy (LOOCV)", y="Count") +
  ggtitle("Mean Accuracy of LOOCV for 100 Repeats") +
  theme_classic() + 
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15)) 
```


<!-- # ROC Prediction Analysis -->

```{r fig.height=8, fig.width=9, message=FALSE, warning=FALSE, include=FALSE}
# iGBM vs Ctrl protein conc ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Initial manifestation" | sample_comparison=="ctrl") %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(0,1), to=c("ctrl", "GBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, `MMP-9`, SYNEMIN, GPNMB) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
pred_obj_1.1 <- run_predictions(train_df)

## ROC Figure 
g <- ggplot(pred_obj_1.1$roc_data, aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.1 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.1[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.1[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.1[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.1[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.1 <- g

# rGBM vs Ctrl protein conc ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | sample_comparison=="ctrl") %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(0,1), to=c("ctrl", "GBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, `MMP-9`, SYNEMIN, GPNMB) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
pred_obj_1.2 <- run_predictions(train_df)

## ROC Figure 
g <- ggplot(pred_obj_1.2$roc_data, aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.2 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.2[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.2[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.2[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.2[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.2 <- g

# iGBM vs rGBM protein conc----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | manifestation_type=="Initial manifestation") %>%
  dplyr::mutate(class = plyr::mapvalues(manifestation_type, from=c("Initial manifestation","Second manifestation"), to=c("iGBM", "rGBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, `MMP-9`, SYNEMIN, GPNMB) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
pred_obj_1.5 <- run_predictions(train_df, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.5$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.5 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.5[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.5[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.5[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.5[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.5 <- g

pred_obj_1.5$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.5_roc_data

# iGBM vs rGBM ratio ----

## Median Centre data before computing ratios

data_long[patient_state==1] %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::add_count() %>%
  dplyr::filter(n==2) %>% 
  dplyr::ungroup() -> tmp_select

tmp_select %>%
  dplyr::select(gene, Number.x, conc, class=manifestation_type) %>%
  dplyr::mutate(conc = log2(conc+1)) -> tmp
tmp %>%
  tidyr::pivot_wider(id_cols = c(Number.x, class), names_from = gene, values_from = conc ) -> tmp
colnames(tmp) <- make.names(colnames(tmp))
tmp <- na.omit(tmp)

tmp_df_proteins <- median_centre_data(tmp[, c(2:6)])
tmp_median_centred <- cbind(tmp[, 1], tmp_df_proteins) 

tmp_median_centred %>%
  dplyr::rename(manifestation_type=class, `MMP-9`=MMP.9) %>%
  tidyr::pivot_longer(cols = c(ASAH1, `MMP-9`, SYNEMIN, GPNMB), names_to = "gene", values_to = "conc") -> tmp_median_centred

data_long_tmp <- merge(tmp_select[, c("Number.x", "gene", "patient_state", "LAB_ID", "collection_type", "manifestation_type")], tmp_median_centred, by=c("Number.x", "manifestation_type", "gene"), all.y=T)

ggplot(data_long_tmp, aes(x=as.factor(Number.x), y=conc, col=Number.x)) +
    geom_boxplot()

data_long_tmp %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_initial$conc - .d_second$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBMoverrGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBMoverrGBM" ) %>%
  dplyr::mutate(log2_conc = ratio) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("iGBMoverrGBM"))) -> pdata

data_long_tmp %>% 
  # dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
  dplyr::filter(!is.na(manifestation_type) | patient_state!=0) %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_second$conc - .d_initial$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "rGBMoveriGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="rGBMoveriGBM" ) %>%
  dplyr::filter(grepl(";", agg_lab_id)) %>%
  dplyr::mutate(log2_conc = ratio) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("rGBMoveriGBM"))) -> pdata2

pdata %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="iGBM") -> pdata

pdata2 %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="rGBM") -> pdata2


pdata <- rbindlist(list(pdata, pdata2))

## Pivot Table
pdata %>%
  dplyr::select(Number.x, class, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide

## Prediction
train_df <- data_wide
train_df <- data_wide[Number.x!=32]
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, )
train_df <- train_df[ !is.infinite(train_df$ASAH1) , ]
## Make Valid Column Names 
colnames(train_df) <- make.names(colnames(train_df))
train_df$class <- factor(train_df$class, levels=c("iGBM", "rGBM"))
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
pred_obj_1.3 <- run_predictions(train_df, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.3$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) +
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.3 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.3[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.3[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.3[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.3[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.3 <- g

pred_obj_1.3$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.3_roc_data

## Merged Plot
pred_obj_1.3$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.3_roc_data

# iGBM vs rGBM with clinical data ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | manifestation_type=="Initial manifestation") %>%
  dplyr::mutate(class = plyr::mapvalues(manifestation_type, from=c("Initial manifestation","Second manifestation"), to=c("iGBM", "rGBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, `MMP-9`, SYNEMIN, GPNMB,  Sex, `Age at surgery (years)`, `Tumor localization`, `Type of resection`) -> data_wide
# data_wide %>%
  # dplyr::mutate(`Time-to-reccurence (days)` = ifelse(class=="iGBM", 0, `Time-to-reccurence (days)`)) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df_proteins <- median_centre_data(train_df[, c(1:5)])
train_df <- cbind(train_df_proteins, train_df[, c(6:9)])
pred_obj_1.4 <- run_predictions(train_df, pred_col_prob = "rGBM")

# coef(pred_obj_1.5$model$finalModel, pred_obj_1.5$model$bestTune$lambda)
# coef(pred_obj_1.41$model$finalModel, pred_obj_1.41$model$bestTune$lambda)
# coef(pred_obj_1.4$model$finalModel, pred_obj_1.4$model$bestTune$lambda)


## ROC Figure 
g <- ggplot(pred_obj_1.4$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.4 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.4[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.4[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.4[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.4[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.4 <- g

pred_obj_1.4$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.4_roc_data

# iGBM vs rGBM with clinical data Only ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | manifestation_type=="Initial manifestation") %>%
  dplyr::mutate(class = plyr::mapvalues(manifestation_type, from=c("Initial manifestation","Second manifestation"), to=c("iGBM", "rGBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n,  Sex, `Age at surgery (years)`, `Tumor localization`, `Type of resection`) -> data_wide
# data_wide %>%
  # dplyr::mutate(`Time-to-reccurence (days)` = ifelse(class=="iGBM", 0, `Time-to-reccurence (days)`)) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
# train_df_proteins <- median_centre_data(train_df[, c(1:5)])
# train_df <- cbind(train_df_proteins, train_df[, c(6:9)])
pred_obj_1.41 <- run_predictions(train_df, pred_col_prob = "rGBM")

# coef(pred_obj_1.5$model$finalModel, pred_obj_1.5$model$bestTune$lambda)
# coef(pred_obj_1.4$model$finalModel, pred_obj_1.4$model$bestTune$lambda)


## ROC Figure 
g <- ggplot(pred_obj_1.41$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.41 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.41[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.41$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.41[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.41$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.41[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.41$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.41[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.41$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.41 <- g

pred_obj_1.41$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.41_roc_data

# iGBM vs rGBM ratio  with clinical data ----
# data_long %>% 
#   dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
#   dplyr::group_by(gene, Number.x) %>%
#   dplyr::group_map(function(.d, ...){
#     .d <- as.data.table(.d)
#     .d %>% 
#       dplyr::group_by(gene, Number.x, patient_state, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`) %>% 
#       dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
#                        agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
#     # print(.d)
#     if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
#       .d_initial <- .d[manifestation_type=="Initial manifestation"]
#       .d_second <- .d[manifestation_type=="Second manifestation"]
#       tmp$ratio <- .d_initial$conc / .d_second$conc
#     } else if (nrow(.d)==2) {
#       .d_initial <- .d[collection_type=="E1"]
#       .d_second <- .d[collection_type=="E2"]
#       tmp$ratio <- .d_initial$conc / .d_second$conc
#     } else {
#       tmp$ratio <- .d$conc
#     }
#     return(tmp)
#   }, .keep = TRUE) %>%
#   rbindlist() %>%
#   dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBMoverrGBM", "pre-surgery/post-surgery"))) %>%
#   dplyr::filter(sample_comparison=="iGBMoverrGBM" ) %>%
#   dplyr::mutate(log2_conc = log2(ratio)) %>%
#   # dplyr::group_by(gene, Number.x) %>%
#   # dplyr::add_count() %>%
#   # dplyr::filter(n==2) %>%
#   dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("iGBMoverrGBM"))) -> pdata
# 
# data_long %>% 
#   dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
#   dplyr::filter(!is.na(manifestation_type) | patient_state!=0) %>%
#   dplyr::group_by(gene, Number.x) %>%
#   dplyr::group_map(function(.d, ...){
#     .d <- as.data.table(.d)
#     .d %>% 
#       dplyr::group_by(gene, Number.x, patient_state, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`) %>% 
#       dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
#                        agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
#     # print(.d)
#     if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
#       .d_initial <- .d[manifestation_type=="Initial manifestation"]
#       .d_second <- .d[manifestation_type=="Second manifestation"]
#       tmp$ratio <- .d_second$conc / .d_initial$conc
#     } else if (nrow(.d)==2) {
#       .d_initial <- .d[collection_type=="E1"]
#       .d_second <- .d[collection_type=="E2"]
#       tmp$ratio <- .d_initial$conc / .d_second$conc
#     } else {
#       tmp$ratio <- .d$conc
#     }
#     return(tmp)
#   }, .keep = TRUE) %>%
#   rbindlist() %>%
#   dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "rGBMoveriGBM", "pre-surgery/post-surgery"))) %>%
#   dplyr::filter(sample_comparison=="rGBMoveriGBM" ) %>%
#   dplyr::filter(grepl(";", agg_lab_id)) %>%
#   dplyr::mutate(log2_conc = log2(ratio)) %>%
#   # dplyr::group_by(gene, Number.x) %>%
#   # dplyr::add_count() %>%
#   # dplyr::filter(n==2) %>%
#   dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("rGBMoveriGBM"))) -> pdata2
# 
# pdata %>%
#   dplyr::select(Number.x, gene, log2_conc) %>%
#   tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
#   dplyr::mutate(class="iGBM") -> pdata
# 
# pdata2 %>%
#   dplyr::select(Number.x, gene, log2_conc) %>%
#   tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
#   dplyr::mutate(class="rGBM") -> pdata2
# 
# 
# pdata <- rbindlist(list(pdata, pdata2))
# 
# pdata <- merge(pdata, patient_meta_data[, .(Number.x, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `Time-to-reccurence (days)`)], by="Number.x", all.x=T)
# 
# ## Pivot Table
# pdata %>%
#   dplyr::select(Number.x, class, ASAH1, SYNEMIN, GPNMB, `MMP-9`, Sex, `Tumor localization`, Sex, `Age at surgery (years)`, `Type of resection`, `Time-to-reccurence (days)`) -> data_wide
# # data_wide %>%
# #   dplyr::mutate(`Time-to-reccurence (days)` = ifelse(class=="iGBM", 0, `Time-to-reccurence (days)`)) -> data_wide
# 
# ## Prediction
# train_df <- data_wide
# train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, )
# train_df <- train_df[ !is.infinite(train_df$ASAH1) , ]
# ## Make Valid Column Names 
# colnames(train_df) <- make.names(colnames(train_df))
# train_df$class <- factor(train_df$class, levels=c("iGBM", "rGBM"))
# colnames(train_df) <- make.names(colnames(train_df))
# train_df <- as.data.frame(train_df)
# pred_obj_1.6 <- run_predictions(train_df, pred_col_prob = "rGBM")
# 
# ## ROC Figure 
# g <- ggplot(pred_obj_1.6$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) +
#   geom_roc(n.cuts=0) + 
#   scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
#   coord_equal() +
#   style_roc(x="1 - Specificity", y="Sensitivity")
# 
# auc_tbl_1.6 <- as.data.table(calc_auc(g))
# 
# g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.6[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.6$master_lift$liftModelVar)), "Dark2")[1])
# g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.6[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.6$master_lift$liftModelVar)), "Dark2")[2])
# g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.6[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.6$master_lift$liftModelVar)), "Dark2")[3])
# g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.6[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.6$master_lift$liftModelVar)), "Dark2")[4])
# subfigure_1.6 <- g
# 
# pred_obj_1.6$roc_data %>%
#   dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
#   dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
#   dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.6_roc_data
# 
# ## Merged Plot
# pred_obj_1.6$roc_data %>%
#   dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
#   dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
#   dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.6_roc_data

## Merged Plot ----

# Full
master_roc_data <- rbindlist(list(`Ctrl vs iGBM`=pred_obj_1.1$roc_data, `Ctrl vs rGBM`=pred_obj_1.2$roc_data, `iGBM vs rGBM protein`=tmp_pred_obj_1.5_roc_data, `iGBM vs rGBM protein + clinical`=tmp_pred_obj_1.4_roc_data, `iGBM vs rGBM protein ratios`=tmp_pred_obj_1.3_roc_data), idcol = "Comparison")

master_auc_data <- rbindlist(list(`iGBM vs rGBM protein + clinical`=auc_tbl_1.41, `Ctrl vs iGBM`=auc_tbl_1.1, `Ctrl vs rGBM`=auc_tbl_1.2, `iGBM vs rGBM protein`=auc_tbl_1.5, `iGBM vs rGBM protein + clinical`=auc_tbl_1.4, `iGBM vs rGBM protein ratios`=auc_tbl_1.3), idcol = "Comparison")

# Only iGMB and rGBM
master_roc_data <- rbindlist(list(`iGBM vs rGBM clinical`=tmp_pred_obj_1.41_roc_data, `iGBM vs rGBM protein`=tmp_pred_obj_1.5_roc_data, `iGBM vs rGBM protein + clinical`=tmp_pred_obj_1.4_roc_data, `iGBM vs rGBM protein ratios`=tmp_pred_obj_1.3_roc_data), idcol = "Comparison")

master_auc_data <- rbindlist(list(`iGBM vs rGBM clinical`=auc_tbl_1.41, `iGBM vs rGBM protein`=auc_tbl_1.5, `iGBM vs rGBM protein + clinical`=auc_tbl_1.4, `iGBM vs rGBM protein ratios`=auc_tbl_1.3), idcol = "Comparison")


master_auc_data[Model %in% c("Elastic Logit")] %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  tidyr::pivot_wider(id_cols = "Comparison", names_from = Model, values_from = AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(core = list(fg_params=list(col=(RColorBrewer::brewer.pal(5, "Dark2")))), base_size = 13)
auc_grob <- tableGrob(auc_annotation_tbl, rows=NULL, theme = table_theme)

g <- ggplot(master_roc_data[Model %in% c("Elastic Logit")], aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Comparison), linetype='solid') +
  geom_roc(n.cuts=0, size=1.5) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(5, "Dark2")) +
  # scale_linetype_manual(values=c("solid", "dotted"), labels=c("Elastic Logit")) + 
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity") +
  annotation_custom(auc_grob, xmin=0.25, xmax=0.95, ymin=0.10, ymax=0.30) +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13)) +
  guides(col="none")

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_d.png", plot = g, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_d.svg", plot = g, width = 11, height = 8.5, dpi = 200)
subfigure_1 <- g

## Separate ROCs for controls

master_roc_data <- rbindlist(list(`Ctrl vs iGBM`=pred_obj_1.1$roc_data, `Ctrl vs rGBM`=pred_obj_1.2$roc_data), idcol = "Comparison")

master_auc_data <- rbindlist(list(`Ctrl vs iGBM`=auc_tbl_1.1, `Ctrl vs rGBM`=auc_tbl_1.2), idcol = "Comparison")

master_auc_data[Model %in% c("Elastic Logit")] %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  tidyr::pivot_wider(id_cols = "Comparison", names_from = Model, values_from = AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(core = list(fg_params=list(col=(RColorBrewer::brewer.pal(5, "Dark2")))))
auc_grob <- tableGrob(auc_annotation_tbl, rows=NULL, theme = table_theme)

g <- ggplot(master_roc_data[Model %in% c("Elastic Logit")], aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Comparison), linetype='solid') +
  geom_roc(n.cuts=0, size=1.5) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(5, "Dark2")) +
  # scale_linetype_manual(values=c("solid", "dotted"), labels=c("Elastic Logit")) + 
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity") +
  annotation_custom(auc_grob, xmin=0.23, xmax=0.90, ymin=0.10, ymax=0.38) +
  theme(axis.title = element_text(size=14),
        axis.text = element_text(size=13),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13))

# ggsave("../../figures/elisa_plasma_proteomics/elisa_roc_controls.png", plot = g, width = 11, height = 8.5, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_roc_controls.svg", plot = g, width = 11, height = 8.5, dpi = 200)
roc_controls <- g
```

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Generate table of prediction metrics ----
pred_col_prob="GBM"

pred_mods_dt <- rbindlist(list(`Ctrl vs iGBM`=pred_obj_1.1$model$pred, 
                               `Ctrl vs rGBM`=pred_obj_1.2$model$pred, 
                               `iGBM vs rGBM protein`=pred_obj_1.5$model$pred %>% dplyr::rename(ctrl=iGBM, GBM=rGBM) %>% dplyr::mutate(pred=plyr::mapvalues(pred, c("iGBM", "rGBM"), c("ctrl", "GBM"))) %>% dplyr::mutate(obs=plyr::mapvalues(obs, c("iGBM", "rGBM"), c("ctrl", "GBM"))), 
                               `iGBM vs rGBM protein + clinical`=pred_obj_1.4$model$pred %>% dplyr::rename(ctrl=iGBM, GBM=rGBM) %>% dplyr::mutate(pred=plyr::mapvalues(pred, c("iGBM", "rGBM"), c("ctrl", "GBM"))) %>% dplyr::mutate(obs=plyr::mapvalues(obs, c("iGBM", "rGBM"), c("ctrl", "GBM"))), 
                               `iGBM vs rGBM protein ratios`=pred_obj_1.3$model$pred %>% dplyr::rename(ctrl=iGBM, GBM=rGBM) %>% dplyr::mutate(pred=plyr::mapvalues(pred, c("iGBM", "rGBM"), c("ctrl", "GBM"))) %>% dplyr::mutate(obs=plyr::mapvalues(obs, c("iGBM", "rGBM"), c("ctrl", "GBM")))), idcol = "Comparison")

conf_mats <-lapply(split(pred_mods_dt[, .(obs, pred)], f=pred_mods_dt$Comparison), function(.d){
  caret::confusionMatrix(.d$obs, .d$pred, positive=pred_col_prob)
})
metric_table <- rbindlist(lapply(conf_mats, function(x){data.frame(metric=names(x$byClass), values=x$byClass)}), idcol = "Comparison")

metric_table %>%
  dplyr::group_by(Comparison, metric) %>% 
  dplyr::filter(metric %in% c("Balanced Accuracy", "Sensitivity", "Specificity", "Pos Pred Value", "Neg Pred Value")) %>%
  dplyr::mutate(values = sprintf("%s%%", round(values*100, 2))) %>%
  dplyr::select(Comparison, metric, values) -> metric_table

metric_table %>%
  tidyr::pivot_wider(id_cols="Comparison", names_from = metric, values_from = values) -> performance_dt

performance_dt %>% 
  dplyr::arrange(Comparison) %>%
  ungroup() %>%
  gt() %>%
  gt::tab_header(md("**Logistic Regression with Elastic Net Regularization Performance Metrics**")) %>%
  cols_label(.list = list(Comparison=md("**Comparison**"), `Balanced Accuracy`=md("**Balanced Accuracy**"), Sensitivity=md("**Sensitivity**"), Specificity=md("**Specificity**"), `Pos Pred Value`=md("**Pos Pred Value**"), `Neg Pred Value`=md("**Neg Pred Value**"))) -> logit_net_comp_perf_met_table
```

<!-- # Prediction for Ctrl vs iGBM -->
```{r fig.height=8, fig.width=9, message=FALSE, warning=FALSE, include=FALSE}
# iGBM vs Ctrl protein conc ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Initial manifestation" | sample_comparison=="ctrl") %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(0,1), to=c("ctrl", "GBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
pred_obj_2.1 <- run_predictions(train_df)

## ROC Figure 
g <- ggplot(pred_obj_2.1$roc_data, aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_2.1$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_2 <- as.data.table(calc_auc(g))

auc_tbl_2 %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  dplyr::select(AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(base_size = 14, core = list(fg_params=list(col=(RColorBrewer::brewer.pal(4, "Dark2")))))
auc_grob <- tableGrob(auc_annotation_tbl, cols = NULL, rows=NULL, theme = table_theme)

g +
  annotation_custom(auc_grob, xmin=0.70, xmax=0.90, ymin=0.10, ymax=0.30) +
  theme(axis.title = element_text(size=14),
        axis.text = element_text(size=13),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13)) -> g
g
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure1_subfigure_a.png", plot = g, width = 9, height = 8, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure1_subfigure_a.svg", plot = g, width = 9, height = 8, dpi = 200)

subfigure_2 <- g

```

<!-- # Prediction for iGBM vs rGBM  -->
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=9, include=FALSE}
# iGBM vs rGBM protein conc ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | sample_comparison=="ctrl") %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(0,1), to=c("ctrl", "GBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
pred_obj_3 <- run_predictions(train_df)

## ROC Figure 
g <- ggplot(pred_obj_3$roc_data, aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_3$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_3 <- as.data.table(calc_auc(g))

auc_tbl_3 %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  dplyr::select(AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(base_size = 14, core = list(fg_params=list(col=(RColorBrewer::brewer.pal(4, "Dark2")))))
auc_grob <- tableGrob(auc_annotation_tbl, cols = NULL, rows=NULL, theme = table_theme)

g +
  annotation_custom(auc_grob, xmin=0.70, xmax=0.90, ymin=0.10, ymax=0.30) +
  theme(axis.title = element_text(size=14),
        axis.text = element_text(size=13),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13)) -> g
g
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure1_subfigure_b.png", plot = g, width = 9, height = 8, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure1_subfigure_b.svg", plot = g, width = 9, height = 8, dpi = 200)

subfigure_3 <- g
```

<!-- # Prediction for iGBM vs rGBM ratio -->
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=9, include=FALSE}
# iGBM vs rGBM ratio ----
data_long %>% 
  dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBMoverrGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBMoverrGBM" ) %>%
  dplyr::mutate(log2_conc = log2(ratio)) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("iGBMoverrGBM"))) -> pdata

data_long %>% 
  dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
  dplyr::filter(!is.na(manifestation_type) | patient_state!=0) %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_second$conc / .d_initial$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "rGBMoveriGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="rGBMoveriGBM" ) %>%
  dplyr::filter(grepl(";", agg_lab_id)) %>%
  dplyr::mutate(log2_conc = log2(ratio)) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("rGBMoveriGBM"))) -> pdata2

pdata %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="iGBM") -> pdata

pdata2 %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="rGBM") -> pdata2


pdata <- rbindlist(list(pdata, pdata2))

## Pivot Table
pdata %>%
  dplyr::select(Number.x, class, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, )
train_df <- train_df[ !is.infinite(train_df$ASAH1) , ]
## Make Valid Column Names 
colnames(train_df) <- make.names(colnames(train_df))
train_df$class <- factor(train_df$class, levels=c("iGBM", "rGBM"))
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
pred_obj_4 <- run_predictions(train_df, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_4$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) +
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_4$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_4 <- as.data.table(calc_auc(g))

auc_tbl_4 %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  dplyr::select(AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(base_size = 14, core = list(fg_params=list(col=(RColorBrewer::brewer.pal(4, "Dark2")))))
auc_grob <- tableGrob(auc_annotation_tbl, cols = NULL, rows=NULL, theme = table_theme)

g +
  annotation_custom(auc_grob, xmin=0.70, xmax=0.90, ymin=0.10, ymax=0.30) +
  theme(axis.title = element_text(size=14),
        axis.text = element_text(size=13),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13)) -> g
g
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure1_subfigure_d.png", plot = g, width = 9, height = 8, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure1_subfigure_d.svg", plot = g, width = 9, height = 8, dpi = 200)

subfigure_4 <- g
```

<!-- # Protein Measurements and Clinical Data -->
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=9, include=FALSE}
data_long %>%
  dplyr::filter(!is.na(manifestation_type) | patient_state==0) %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(Manifestation = plyr::mapvalues(manifestation_type, from=c("Initial manifestation", "Second manifestation"), to=c("iGBM", "rGBM"))) %>%
  dplyr::mutate(Manifestation = ifelse(is.na(Manifestation), "Ctrl", Manifestation)) %>%
  dplyr::mutate(Manifestation = factor(Manifestation, levels = c("Ctrl", "iGBM", "rGBM"))) %>%
  dplyr::filter(Manifestation!="Ctrl") %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, log2_conc, Manifestation) %>%
  tidyr::pivot_wider(id_cols = c("Number.x", "Manifestation",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Manifestation'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide #'Latency (days)'
  
# data_wide$`Age at surgery (years)` <- factor(data_wide$`Age at surgery (years)`, levels = unique(data_wide$`Age at surgery (years)`))
data_wide$`Tumor localization` <- factor(data_wide$`Tumor localization`, levels = unique(data_wide$`Tumor localization`))
data_wide$`Tumor localization` <- as.numeric(data_wide$`Tumor localization`)
data_wide$`Type of resection` <- factor(data_wide$`Type of resection`, levels = unique(data_wide$`Type of resection`))
data_wide$`Type of resection`  <- as.numeric(data_wide$`Type of resection` )
data_wide$`MGMT promotor methylation` <- factor(data_wide$`MGMT promotor methylation`, levels = unique(data_wide$`MGMT promotor methylation`))
data_wide$`MGMT promotor methylation` <- as.numeric(data_wide$`MGMT promotor methylation`)
data_wide$`p53 accumulation` <- factor(data_wide$`p53 accumulation`, levels = unique(data_wide$`p53 accumulation`))
data_wide$`p53 accumulation` <- as.numeric(data_wide$`p53 accumulation` )
data_wide$`Ki67-Li` <- factor(data_wide$`Ki67-Li`, levels = unique(data_wide$`Ki67-Li`))
data_wide$`Ki67-Li` <- as.numeric(data_wide$`Ki67-Li` )
data_wide$`EGFR vIII` <- factor(data_wide$`EGFR vIII`, levels = unique(data_wide$`EGFR vIII`))
data_wide$`EGFR vIII` <- as.numeric(data_wide$`EGFR vIII` )
data_wide$Treatment <- factor(data_wide$Treatment, levels = unique(data_wide$Treatment))
data_wide$Treatment <- as.numeric(data_wide$Treatment)
# data_wide$`Time-to-reccurence (days)` <- factor(data_wide$`Time-to-reccurence (days)`, levels = unique(data_wide$`Time-to-reccurence (days)`))
# data_wide$`Latency (days)` <- factor(data_wide$`Latency (days)`, levels = unique(data_wide$`Latency (days)`))

# Encode Category Sex
data_wide$Sex <- plyr::mapvalues(data_wide$Sex, from = c("w", "m"), to = c(0, 1))
data_wide$Sex <- as.factor(data_wide$Sex)
data_wide$Sex <- as.numeric(data_wide$Sex)

data_wide  %>%
  dplyr::select(Number.x, -`Age at surgery (years)`, -Sex, `Tumor localization`, -`Type of resection`, -`MGMT promotor methylation`, -`p53 accumulation`, -`Ki67-Li`, -`EGFR vIII`, -Treatment, -`Time-to-reccurence (days)`, Manifestation, ASAH1, SYNEMIN,GPNMB,`MMP-9`) -> data_wide



# Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x)
# Make Valid Column Names 
colnames(train_df) <- make.names(colnames(train_df))
train_df %>% dplyr::rename(class=Manifestation) -> train_df
train_df$class <- factor(train_df$class, levels=c("iGBM", "rGBM"))
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
pred_obj_5 <- run_predictions(train_df, pred_col_prob = "rGBM")

# ROC
g <- ggplot(pred_obj_5$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) +
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_5$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_5 <- as.data.table(calc_auc(g))

auc_tbl_5 %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  dplyr::select(AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(base_size = 14, core = list(fg_params=list(col=(RColorBrewer::brewer.pal(4, "Dark2")))))
auc_grob <- tableGrob(auc_annotation_tbl, cols = NULL, rows=NULL, theme = table_theme)

g +
  annotation_custom(auc_grob, xmin=0.70, xmax=0.90, ymin=0.10, ymax=0.30) +
  theme(axis.title = element_text(size=14),
        axis.text = element_text(size=13),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13)) -> g
g
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure1_subfigure_c.png", plot = g, width = 9, height = 8, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure1_subfigure_c.svg", plot = g, width = 9, height = 8, dpi = 200)

subfigure_5 <- g
```

<!-- # Combination Proteins -->
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=9, include=FALSE}
# iGBM vs rGBM protein conc----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | manifestation_type=="Initial manifestation") %>%
  dplyr::mutate(class = plyr::mapvalues(manifestation_type, from=c("Initial manifestation","Second manifestation"), to=c("iGBM", "rGBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, `MMP-9`, SYNEMIN, GPNMB) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
train_df <- train_df %>% dplyr::select(class, ASAH1)
train_df$dummy_var <- 0
pred_obj_1.51 <- run_predictions(train_df, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.51$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.51$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.51 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.51[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.51$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.51[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.51$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.51[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.51$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.51[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.51$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.51 <- g

pred_obj_1.51$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.51_roc_data

## 2 Proteins
## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
train_df <- train_df %>% dplyr::select(class, ASAH1, SYNEMIN)
pred_obj_1.52 <- run_predictions(train_df, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.52$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.52$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.52 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.52[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.52$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.52[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.52$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.52[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.52$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.52[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.52$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.52 <- g

pred_obj_1.52$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.52_roc_data

## 3 Protein
## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
train_df <- train_df %>% dplyr::select(class, ASAH1, SYNEMIN, GPNMB)
pred_obj_1.53 <- run_predictions(train_df, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.53$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.53$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.53 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.53[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.53$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.53[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.53$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.53[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.53$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.53[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.53$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.53 <- g

pred_obj_1.53$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.53_roc_data

## 4 Protein
## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
train_df <- train_df %>% dplyr::select(class, ASAH1, SYNEMIN, GPNMB, MMP.9)
pred_obj_1.54 <- run_predictions(train_df, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.54$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.54$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.54 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.54[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.54$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.54[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.54$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.54[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.54$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.54[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.54$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.54 <- g

pred_obj_1.54$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.54_roc_data

master_Table <- rbindlist(list(`1 Protein`=tmp_pred_obj_1.51_roc_data, `1-2 Proteins`=tmp_pred_obj_1.52_roc_data, `1-3 Proteins`=tmp_pred_obj_1.53_roc_data, `1-4 Proteins`=tmp_pred_obj_1.54_roc_data), idcol = 'Number of Proteins')


master_auc_data <- rbindlist(list(`1 Protein`=auc_tbl_1.51, `1-2 Proteins`=auc_tbl_1.52, `1-3 Proteins`=auc_tbl_1.53, `1-4 Proteins`=auc_tbl_1.54), idcol = "Number of Proteins")


master_auc_data[Model %in% c("Elastic Logit")] %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  tidyr::pivot_wider(id_cols = "Number of Proteins", names_from = Model, values_from = AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(core = list(fg_params=list(col=(RColorBrewer::brewer.pal(5, "Dark2")))), base_size = 13)
auc_grob <- tableGrob(auc_annotation_tbl, rows=NULL, theme = table_theme)

g <- ggplot(master_Table[Model %in% c("Elastic Logit")], aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=`Number of Proteins`), linetype='solid') +
  geom_roc(n.cuts=0, size=1.5) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(5, "Dark2")) +
  # scale_linetype_manual(values=c("solid", "dotted"), labels=c("Elastic Logit")) + 
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity") +
  annotation_custom(auc_grob, xmin=0.25, xmax=0.95, ymin=0.10, ymax=0.30) +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13)) +
  guides(col="none")
g
# iGBM vs rGBM ratio ----

## Median Centre data before computing ratios

data_long[patient_state==1] %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::add_count() %>%
  dplyr::filter(n==2) %>% 
  dplyr::ungroup() -> tmp_select

tmp_select %>%
  dplyr::select(gene, Number.x, conc, class=manifestation_type) %>%
  dplyr::mutate(conc = log2(conc+1)) -> tmp
tmp %>%
  tidyr::pivot_wider(id_cols = c(Number.x, class), names_from = gene, values_from = conc ) -> tmp
colnames(tmp) <- make.names(colnames(tmp))
tmp <- na.omit(tmp)

tmp_df_proteins <- median_centre_data(tmp[, c(2:6)])
tmp_median_centred <- cbind(tmp[, 1], tmp_df_proteins) 

tmp_median_centred %>%
  dplyr::rename(manifestation_type=class, `MMP-9`=MMP.9) %>%
  tidyr::pivot_longer(cols = c(ASAH1, `MMP-9`, SYNEMIN, GPNMB), names_to = "gene", values_to = "conc") -> tmp_median_centred

data_long_tmp <- merge(tmp_select[, c("Number.x", "gene", "patient_state", "LAB_ID", "collection_type", "manifestation_type")], tmp_median_centred, by=c("Number.x", "manifestation_type", "gene"), all.y=T)

ggplot(data_long_tmp, aes(x=as.factor(Number.x), y=conc, col=Number.x)) +
    geom_boxplot()

data_long_tmp %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_initial$conc - .d_second$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBMoverrGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBMoverrGBM" ) %>%
  dplyr::mutate(log2_conc = ratio) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("iGBMoverrGBM"))) -> pdata

data_long_tmp %>% 
  # dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
  dplyr::filter(!is.na(manifestation_type) | patient_state!=0) %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_second$conc - .d_initial$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "rGBMoveriGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="rGBMoveriGBM" ) %>%
  dplyr::filter(grepl(";", agg_lab_id)) %>%
  dplyr::mutate(log2_conc = ratio) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("rGBMoveriGBM"))) -> pdata2

pdata %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="iGBM") -> pdata

pdata2 %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="rGBM") -> pdata2


pdata <- rbindlist(list(pdata, pdata2))

## Pivot Table
pdata %>%
  dplyr::select(Number.x, class, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide

## Prediction
train_df <- data_wide
train_df <- data_wide[Number.x!=32]
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, )
train_df <- train_df[ !is.infinite(train_df$ASAH1) , ]
## Make Valid Column Names 
colnames(train_df) <- make.names(colnames(train_df))
train_df$class <- factor(train_df$class, levels=c("iGBM", "rGBM"))
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)

## 1 Protein ----
train_df_sub <- train_df %>% dplyr::select(class, ASAH1)
train_df_sub$dummy_var <- 0
pred_obj_1.31 <- run_predictions(train_df_sub, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.31$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) +
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.31$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.31 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.31[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.31$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.31[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.31$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.31[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.31$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.31[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.31$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.31 <- g

pred_obj_1.31$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.31_roc_data

## 2 Protein ----
train_df_sub <- train_df %>% dplyr::select(class, ASAH1, SYNEMIN)
pred_obj_1.32 <- run_predictions(train_df_sub, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.32$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) +
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.32$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.32 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.32[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.32$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.32[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.32$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.32[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.32$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.32[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.32$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.32 <- g

pred_obj_1.32$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.32_roc_data

## 3 Protein ----
train_df_sub <- train_df %>% dplyr::select(class, ASAH1, SYNEMIN, GPNMB)
pred_obj_1.33 <- run_predictions(train_df_sub, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.33$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) +
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.33$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.33 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.33[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.33$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.33[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.33$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.33[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.33$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.33[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.33$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.33 <- g

pred_obj_1.33$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.33_roc_data

## 4 Protein ----
train_df_sub <- train_df %>% dplyr::select(class, ASAH1, SYNEMIN, GPNMB, MMP.9)
pred_obj_1.34 <- run_predictions(train_df_sub, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.34$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) +
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.34$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.34 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.34[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.34$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.34[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.34$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.34[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.34$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.34[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.34$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.34 <- g

pred_obj_1.34$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.34_roc_data

## Merged Plot
master_Table <- rbindlist(list(`1 Protein`=tmp_pred_obj_1.31_roc_data, `1-2 Proteins`=tmp_pred_obj_1.32_roc_data, `1-3 Proteins`=tmp_pred_obj_1.33_roc_data, `1-4 Proteins`=tmp_pred_obj_1.34_roc_data), idcol = 'Number of Proteins')


master_auc_data <- rbindlist(list(`1 Protein`=auc_tbl_1.31, `1-2 Proteins`=auc_tbl_1.32, `1-3 Proteins`=auc_tbl_1.33, `1-4 Proteins`=auc_tbl_1.34), idcol = "Number of Proteins")


master_auc_data[Model %in% c("Elastic Logit")] %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  tidyr::pivot_wider(id_cols = "Number of Proteins", names_from = Model, values_from = AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(core = list(fg_params=list(col=(RColorBrewer::brewer.pal(5, "Dark2")))), base_size = 13)
auc_grob <- tableGrob(auc_annotation_tbl, rows=NULL, theme = table_theme)

g <- ggplot(master_Table[Model %in% c("Elastic Logit")], aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=`Number of Proteins`), linetype='solid') +
  geom_roc(n.cuts=0, size=1.5) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(5, "Dark2")) +
  # scale_linetype_manual(values=c("solid", "dotted"), labels=c("Elastic Logit")) + 
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity") +
  annotation_custom(auc_grob, xmin=0.25, xmax=0.95, ymin=0.10, ymax=0.30) +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13)) +
  guides(col="none")
g

```


<!-- # Latency Data -->
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=9, include=FALSE}
latency_data_file <- "../../data/elisa_plasma_proteomics/ELISA_LATENCY_KM.csv"
latency_data <- fread(latency_data_file)
# Clean names
latency_data <- rstatix::make_clean_names(latency_data)
latency_data %>%
  dplyr::mutate(ASAH1.ng.ml = log2(ASAH1.ng.ml+1),
                SYNM.ng.ml = log2(SYNM.ng.ml+1),
                GPNMB.ng.ml = log2(GPNMB.ng.ml+1),
                MMP.9.ng.ml = log2(MMP.9.ng.ml+1)) %>%
  dplyr::mutate(ASAH1 = ifelse(ASAH1.ng.ml>=mean(ASAH1.ng.ml), "high", "low"),
                SYNM = ifelse(SYNM.ng.ml>=mean(SYNM.ng.ml), "high", "low"),
                GPNMB = ifelse(GPNMB.ng.ml>=mean(GPNMB.ng.ml), "high", "low"),
                MMP9 = ifelse(MMP.9.ng.ml>=mean(MMP.9.ng.ml, na.rm=T), "high", "low")) -> latency_data

fit <- survfit(Surv(Latency..days., status) ~ ASAH1, data = latency_data)
print(fit)
n_data <- table(latency_data[!is.na(latency_data$Latency..days.)]$ASAH1)
# Change color, linetype by strata, risk.table color by strata
surv_p1 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1700, 0.95), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high\n(n=%s)", n_data[1]), sprintf("ASAH1=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Latency Time (Days)")

surv_p1 <- surv_p1$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm')) 

fit <- survfit(Surv(Latency..days., status) ~ SYNM, data = latency_data)
print(fit)
n_data <- table(latency_data[!is.na(latency_data$Latency..days.)]$SYNM)
# Change color, linetype by strata, risk.table color by strata
surv_p2 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1700, 0.95), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("SYNM=high\n(n=%s)", n_data[1]), sprintf("SYNM=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Latency Time (Days)")

surv_p2 <- surv_p2$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'))

fit <- survfit(Surv(Latency..days., status) ~ GPNMB, data = latency_data)
print(fit)
n_data <- table(latency_data[!is.na(latency_data$Latency..days.)]$GPNMB)
# Change color, linetype by strata, risk.table color by strata
surv_p3 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1700, 0.95), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("GPNMB=high\n(n=%s)", n_data[1]), sprintf("GPNMB=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Latency Time (Days)")

surv_p3 <- surv_p3$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'))

fit <- survfit(Surv(Latency..days., status) ~ MMP9, data = latency_data)
print(fit)
n_data <- table(latency_data[!is.na(latency_data$Latency..days.)]$MMP9)
# Change color, linetype by strata, risk.table color by strata
surv_p4 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1700, 0.95), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("MMP9=high\n(n=%s)", n_data[1]), sprintf("MMP9=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Latency Time (Days)")

surv_p4 <- surv_p4$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'))

fit <- survfit(Surv(Latency..days., status) ~ ASAH1 + SYNM, data = latency_data)
print(fit)
# Change color, linetype by strata, risk.table color by strata
surv_p5 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1700, 0.95), conf.int = F,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high, SYNM=high (n=%s)", fit$n.risk[1]), sprintf("ASAH1=high, SYNM=low (n=%s)", fit$n.risk[fit$n.risk[1]+1]), sprintf("ASAH1=low, SYNM=high (n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]), sprintf("ASAH1=low, SYNM=low (n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]+fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]]))
          ) +
  labs(x="Latency Time (Days)")  +
  guides(col=guide_legend(nrow=4,byrow=TRUE))

surv_p5 <- surv_p5$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")

surv_p5_with_ci <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1700, 0.95), conf.int = T,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high, SYNM=high (n=%s)", fit$n.risk[1]), sprintf("ASAH1=high, SYNM=low (n=%s)", fit$n.risk[fit$n.risk[1]+1]), sprintf("ASAH1=low, SYNM=high (n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]), sprintf("ASAH1=low, SYNM=low (n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]+fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]]))
          ) +
  labs(x="Latency Time (Days)")  +
  guides(col=guide_legend(nrow=2,byrow=TRUE))

surv_p5_with_ci <- surv_p5_with_ci$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")

survival_plot_grid <- ggarrange(surv_p1, surv_p2, surv_p3, surv_p4, ncol = 4)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c.png", plot = survival_plot_grid, width = 18, height = 7, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c.svg", plot = survival_plot_grid, width = 18, height = 7, dpi = 200)

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c1.png", plot = surv_p1, width = 11, height = 7, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c1.svg", plot = surv_p1, width = 11, height = 7, dpi = 200)

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c2.png", plot = surv_p2, width = 11, height = 7, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c2.svg", plot = surv_p2, width = 11, height = 7, dpi = 200)

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c3.png", plot = surv_p3, width = 11, height = 7, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c3.svg", plot = surv_p3, width = 11, height = 7, dpi = 200)

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c4.png", plot = surv_p4, width = 11, height = 7, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_c4.svg", plot = surv_p4, width = 11, height = 7, dpi = 200)

# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_.png", plot = surv_p5$plot, width = 11, height = 7, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_.svg", plot = surv_p5$plot, width = 11, height = 7, dpi = 200)
# 
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_with_ci.png", plot = surv_p5_with_ci$plot, width = 11, height = 7, dpi = 200)
# ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_with_ci.svg", plot = surv_p5_with_ci$plot, width = 11, height = 7, dpi = 200)

# survival model with time to recurrence; combined group of all 4 proteins
ttr_surv_data <- merge(latency_data, patient_meta_data[, .(LAB_ID, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `Time-to-reccurence (days)`)], by.x="lab_id", by.y="LAB_ID", all.x=T)
ttr_surv_data$`Time-to-reccurence (days)`[ ttr_surv_data$`Time-to-reccurence (days)`=="/" ] = ""
ttr_surv_data$`Time-to-reccurence (days)` <- as.numeric(ttr_surv_data$`Time-to-reccurence (days)`)
ttr_surv_data <- rstatix::make_clean_names(ttr_surv_data)


fit <- survfit(Surv(Time.to.reccurence..days., status) ~ ASAH1, data = ttr_surv_data)
print(fit)
n_data <- table(ttr_surv_data[!is.na(ttr_surv_data$Time.to.reccurence..days.)]$ASAH1)
# Change color, linetype by strata, risk.table color by strata
ttr_surv_p1 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high\n(n=%s)", n_data[1]), sprintf("ASAH1=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability") 

ttr_surv_p1 <- ttr_surv_p1$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")

fit <- survfit(Surv(Time.to.reccurence..days., status) ~ SYNM, data = ttr_surv_data %>% dplyr::select(-MMP.9.ng.ml))
print(fit)
n_data <- table(ttr_surv_data[!is.na(ttr_surv_data$Time.to.reccurence..days.)]$SYNM)
# Change color, linetype by strata, risk.table color by strata
ttr_surv_p2 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("SYNM=high\n(n=%s)", n_data[1]), sprintf("SYNM=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")

ttr_surv_p2 <- ttr_surv_p2$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")

fit <- survfit(Surv(Time.to.reccurence..days., status) ~ GPNMB, data = ttr_surv_data)
print(fit)
n_data <- table(ttr_surv_data[!is.na(ttr_surv_data$Time.to.reccurence..days.)]$GPNMB)
# Change color, linetype by strata, risk.table color by strata
ttr_surv_p3 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("GPNMB=high\n(n=%s)", n_data[1]), sprintf("GPNMB=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")

ttr_surv_p3 <- ttr_surv_p3$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")

fit <- survfit(Surv(Time.to.reccurence..days., status) ~ MMP9, data = ttr_surv_data)
print(fit)
n_data <- table(ttr_surv_data[!is.na(ttr_surv_data$Time.to.reccurence..days.)]$MMP9)
# Change color, linetype by strata, risk.table color by strata
ttr_surv_p4 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("MMP9=high\n(n=%s)", n_data[1]), sprintf("MMP9=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")

ttr_surv_p4 <- ttr_surv_p4$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")

fit <- survfit(Surv(Time.to.reccurence..days., status) ~ ASAH1 + SYNM, data = ttr_surv_data)
print(fit)
# Change color, linetype by strata, risk.table color by strata
ttr_surv_p5 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = F,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high, SYNM=high (n=%s)", fit$n.risk[1]), sprintf("ASAH1=high, SYNM=low (n=%s)", fit$n.risk[fit$n.risk[1]+1]), sprintf("ASAH1=low, SYNM=high (n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]), sprintf("ASAH1=low, SYNM=low (n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]+fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")  +
  guides(col=guide_legend(nrow=1,byrow=TRUE))

ttr_surv_p5 <- ttr_surv_p5$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")

ttr_surv_p5_with_ci <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = T,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high, SYNM=high\n(n=%s)", fit$n.risk[1]), sprintf("ASAH1=high, SYNM=low\n(n=%s)", fit$n.risk[fit$n.risk[1]+1]), sprintf("ASAH1=low, SYNM=high\n(n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]), sprintf("ASAH1=low, SYNM=low\n(n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]+fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")  +
  guides(col=guide_legend(nrow=2,byrow=TRUE))

ttr_surv_p5_with_ci <- ttr_surv_p5_with_ci$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")


## All 4 ----
fit <- survfit(Surv(Time.to.reccurence..days., status) ~ ASAH1 + SYNM + GPNMB + MMP9, data = ttr_surv_data)
print(fit)
# Change color, linetype by strata, risk.table color by strata
ttr_surv_p5 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = F,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          legend.title = "",
          legend.labs = c(
            sprintf("ASAH1=high, SYNM=high, GPNMB=high, MMP9=high (n=1)"), 
            sprintf("ASAH1=high, SYNM=low, GPNMB=high, MMP9=low (n=1)"),  
            sprintf("ASAH1=low, SYNM=high, GPNMB=low , MMP9=high (n=2)"), 
            sprintf("ASAH1=high, SYNM=high, GPNMB=low , MMP9=low (n=1)"),
            sprintf("ASAH1=high, SYNM=low , GPNMB=high, MMP9=low (n=1)"),
            sprintf("ASAH1=high, SYNM=low , GPNMB=low , MMP9=high (n=1)"),
            sprintf("ASAH1=high, SYNM=low , GPNMB=low , MMP9=low (n=2)"),
            sprintf("ASAH1=low, SYNM=high, GPNMB=high, MMP9=high (n=1)"),
            sprintf("ASAH1=low, SYNM=high, GPNMB=high, MMP9=low (n=1)"),
            sprintf("ASAH1=low, SYNM=high, GPNMB=low , MMP9=high (n=2)"),
            sprintf("ASAH1=low, SYNM=low , GPNMB=high, MMP9=low (n=3)"),
            sprintf("ASAH1=low, SYNM=low , GPNMB=low , MMP9=high (n=1)"),
            sprintf("ASAH1=low, SYNM=low , GPNMB=low , MMP9=low (n=2)")
            )
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")  +
  guides(col=guide_legend(nrow=4,byrow=TRUE))

ttr_surv_p5 <- ttr_surv_p5$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")

ttr_surv_p5_with_ci <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = T,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high, SYNM=high\n(n=%s)", fit$n.risk[1]), sprintf("ASAH1=high, SYNM=low\n(n=%s)", fit$n.risk[fit$n.risk[1]+1]), sprintf("ASAH1=low, SYNM=high\n(n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]), sprintf("ASAH1=low, SYNM=low\n(n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]+fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")  +
  guides(col=guide_legend(nrow=2,byrow=TRUE))

ttr_surv_p5_with_ci <- ttr_surv_p5_with_ci$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = "bottom")

## COXPH ----
fit <- coxph(Surv(Time.to.reccurence..days., status) ~ sex + ASAH1.ng.ml + SYNM.ng.ml + GPNMB.ng.ml + MMP.9.ng.ml, data = ttr_surv_data)
print(summary(fit))

library(finalfit)

dependent_var <- "Surv(Time.to.reccurence..days., status)"
explanatory_vars <- c("ASAH1.ng.ml", "SYNM.ng.ml", "GPNMB.ng.ml", "MMP.9.ng.ml")

ttr_surv_data %>% 
    coxphmulti(dependent_var, explanatory_vars) %>% 
    cox.zph() %>% 
    {zph_result <<- .} %>% 
    plot(var=3)

zph_result

ttr_surv_data %>%
  finalfit(dependent_var, explanatory_vars) %>%
  gt()
  

ttr_surv_data %>%
  hr_plot(dependent_var, explanatory_vars, dependent_label="Time to Recurrence")

ttr_surv_data %>%
  surv_plot(dependent_var, explanatory_vars, xlab="Time (days)", pval=TRUE, legend="none")


ttr_surv_data %>%
  ff_plot(dependent_var, explanatory_vars, dependent_label="Time to Recurrence")
```

\blandscape

# Master Figure
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=18, fig.width=21}
# mataster_figure <- ggarrange(ggarrange(subfigure_6, 
#                                        splsda_ratio_avg_traj_fig, ncol=2, labels = c("A", "B")),
#                              ggarrange(survival_plot_grid, ncol=1, labels = c("C")), 
#                              ggarrange(subfigure_1, 
#                                        p_ratio_ttr, ncol=2, labels = c("D", "E")),
#                              ncol = 1, nrow=3)

mataster_figure <- ggarrange(ggarrange(subfigure_6, 
                                       ggarrange(ttr_surv_p1, ttr_surv_p2, ncol=2), ncol=2, labels = c("A", "B")),
                             ggarrange(subfigure_1, 
                                       splsda_better_ratio_avg_traj_fig, ncol=2, labels = c("C", "D")),
                             ncol = 1, nrow=2)
mataster_figure
ggsave("../../figures/elisa_plasma_proteomics/elisa_master_figure.png", plot = mataster_figure, width = 25, height = 21, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_master_figure.svg", plot = mataster_figure, width = 25, height = 21, dpi = 200)

```

<!-- **Figure 3: A Promising Plasma Proteomics Signature for Stratification and Prediction of Recurrent Glioblastoma Multiforme.** Concentration of four proteins identified from TMT-MS tissue proteomics, were measured in patients' plasma samples using Enzyme-Linked Immunosorbent Assay (ELISA). (**a**) The log2 transformed protein concentration measurements demonstrates the difference in log2 concentration distributions between control, initial GBM and paired recurrent GBM populations. Note, a patient was omitted due to being an outlier, see Supp Figure X for full set of patients. (**b**) Reducing the data of the four proteins ratios demonstrates an average trajectory and separation of $\frac{initial \quad GBM}{recurrent \quad GBM}$ vs $\frac{recurrent \quad GBM}{initial \quad GBM}$. (**c**)  Kaplan-Meyer plots showing demonstrating high concentration of ASAH1 and high concentration of SYNEMIN contributing to a higher probability of longer survival. (**d**) The receiver operating characteristic (ROC) shows the potential of using plasma proteomics for separating each population, driven mostly from information contained in the proteins. (**e**) The potential of recurrent GBM / initial GBM protein ratios for predicting time to recurrence. Note, each point represents a single patient, so we may not be able to directly state a specific ratio can be used to infer to time to recurrence. However, we can still state given a specific patients ratio, we can predict that patients time to recurrence.  -->
**Figure 3: A Promising Plasma Proteomics Signature for Stratification and Prediction of Recurrent Glioblastoma Multiforme.** Concentration of four proteins identified from TMT-MS tissue proteomics, were measured in patients' plasma samples using Enzyme-Linked Immunosorbent Assay (ELISA). (**a**) The log2 transformed protein concentration measurements demonstrates the difference in log2 concentration distributions between control, initial GBM and paired recurrent GBM populations. Note, a patient was omitted due to being an outlier, see Supp Figure X for full set of patients. (**b**)  Kaplan-Meyer plots showing demonstrating high concentration of ASAH1 and high concentration of SYNEMIN contributing to a higher probability of tumor recurrence occurring at a later date. (**c**) The receiver operating characteristic (ROC) shows the potential of using plasma proteomics for separating each population, driven mostly from information contained in the proteins. (**d**) Reducing the data of the four proteins ratios demonstrates an average trajectory and separation of $\frac{initial \quad GBM}{recurrent \quad GBM}$ vs $\frac{recurrent \quad GBM}{initial \quad GBM}$. One patient was randomly left out as a test patient (black + point), 4 samples were randomly sampled to represent $\frac{initial \quad GBM}{recurrent \quad GBM}$ (blue points) and remaining 4 samples were used to represent $\frac{recurrent \quad GBM}{initial \quad GBM}$.

\elandscape

\pagebreak

# Supplemental Figures and Tables

```{r message=FALSE, warning=FALSE, include=FALSE}
fig_counter <- 1
tab_counter <- 1
```

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=8.5, fig.width=11}

supp_figure_0 <- ggarrange(ggarrange(subfigure_6_org, supp_figure_pre_v_post, ncol=1, nrow=2, labels=c("A", "B")), ggarrange(supp_figure_ratios, supp_figure_ratios2, nrow=2, ncol=1), ncol=2, nrow=1, labels=c("", "C"))
supp_figure_0
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.png", fig_counter), plot = supp_figure_0, width = 18, height = 16, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.svg", fig_counter), plot = supp_figure_0, width = 18, height = 16, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.png", fig_counter), plot = subfigure_6_org, width = 11, height = 8.5, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.svg", fig_counter), plot = subfigure_6_org, width = 11, height = 8.5, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.png", fig_counter), plot = supp_figure_pre_v_post, width = 11, height = 8.5, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.svg", fig_counter), plot = supp_figure_pre_v_post, width = 11, height = 8.5, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_c.png", fig_counter), plot =  ggarrange(supp_figure_ratios, supp_figure_ratios2, nrow=2, ncol=1), width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_c.svg", fig_counter), plot =  ggarrange(supp_figure_ratios, supp_figure_ratios2, nrow=2, ncol=1), width = 9, height = 8, dpi = 200)
```
**Supplemental Figure** `r knitr::asis_output(md(sprintf("**%s**", fig_counter)))` `r fig_counter <- fig_counter + 1`**: Distribution of Pre- vs Post-Surgery Plasma Proteomics and Ratios of Initial and Recurrent GBM Plasma Proteomics.**  (**a**) The log2 transformed protein concentration measurements demonstrates the difference in log2 concentration distributions between control, initial GBM and paired recurrent GBM population for all patients. (**b**) The log2 transformed protein concentration measurements demonstrates the difference in log2 concentration distributions between control, pre-surgery and post-surgery populations.  (**c**) Demonstrates the log transformed $\frac{initial \quad GBM}{recurrent \quad GBM}$ ratio (top) as well as the inverse ratio (bottom) for the potential use of identifying a patients state to recurrence using these ratios.

\pagebreak

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=18, fig.width=21}
supp_figure_1 <- ggarrange(ggarrange(roc_controls, subfigure_2, subfigure_3, ncol=3, nrow=1, labels=c("A", "B", "C")), ggarrange(subfigure_5, subfigure_4, labels=c("D", "E")), ncol=1, nrow=2)

supp_figure_1
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.png", fig_counter), plot = supp_figure_1, width = 18, height = 18, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.svg", fig_counter), plot = supp_figure_1, width = 18, height = 18, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.png", fig_counter), plot = roc_controls, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.svg", fig_counter), plot = roc_controls, width = 9, height = 8, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.png", fig_counter), plot = subfigure_2, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.svg", fig_counter), plot = subfigure_2, width = 9, height = 8, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_c.png", fig_counter), plot = subfigure_3, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_c.svg", fig_counter), plot = subfigure_3, width = 9, height = 8, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_d.png", fig_counter), plot = subfigure_5, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_d.svg", fig_counter), plot = subfigure_5, width = 9, height = 8, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_e.png", fig_counter), plot = subfigure_4, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_e.svg", fig_counter), plot = subfigure_4, width = 9, height = 8, dpi = 200)
```
**Supplemental Figure `r fig_counter` `r fig_counter <- fig_counter + 1`: Variation in Predictive Performance using Different Models.** Separation on Controls and testing parametric (logistic regression, logistic regression with elastic net regularization), non-parametric (k-nearest neighbours) and ensemble (random forest) supervised learning algorithms on different prediction tasks. (**a**)  The receiver operating characteristic (ROC) shows the potential of using plasma proteomics for separating controls (health patients) from patients with GBM using an elastic logistic regression. (**b**) Most models have AUCs greater than 60% when predicting control vs initial GBM. (**c**) Most models have comparable AUCs, except for k-nearest neighbours when predicting for initial GBM vs recurrent GBM. (**d**) Including clinical data for *Tumor localization*, *Sex*, *Age at surgery (years)*, *Type of resection* reduces predictive performance for all models. (**e**) All models, except random forest, have AUCs above 80% when predicting initial GBM vs recurrent GBM using protein ratios.

\pagebreak

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=11, fig.width=8.5, dpi = 200}
supp_figure_2 <- ggarrange(splsda_avg_traj_fig, splsda_ind_traj_fig, ncol=1, nrow=2, labels=c("A", "B"))

supp_figure_2
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.png", fig_counter), plot = supp_figure_2, width = 11, height = 8.5, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.svg", fig_counter), plot = supp_figure_2, width = 11, height = 8.5, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.png", fig_counter), plot = splsda_avg_traj_fig, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.svg", fig_counter), plot = splsda_avg_traj_fig, width = 9, height = 8, dpi = 200)

# ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.png", fig_counter), plot = subfigure_2, width = 9, height = 8, dpi = 200)
# ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.svg", fig_counter), plot = subfigure_2, width = 9, height = 8, dpi = 200)

```
**Supplemental Figure `r fig_counter` `r fig_counter <- fig_counter + 1`: Individual Patient Trajectory in Low Dimensional Space.** (**a**) A sparse partial least squares discriminant analysis shows separation between the different population groups, as well as an overall trajectory of a initial GBM patient going towards a recurrent GBM patient in a low dimensional space. (**b**) Individual trajectories demonstrate the individual variance of patients path to recurrence in a low-dimensional space.

\pagebreak

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=18, fig.width=21, dpi=200}
supp_figure_3 <- ggarrange( p_pa_pb_ratio_scatter, splsda_ratio_ind_traj_fig, ncol=1, nrow=2, labels=c("A", "B"), heights = c(1.5,1))

supp_figure_3
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.png", fig_counter), plot = supp_figure_3, width = 18, height = 18, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.svg", fig_counter), plot = supp_figure_3, width = 18, height = 18, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.png", fig_counter), plot = p_pa_pb_ratio_scatter, width = 11, height = 8.5, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.svg", fig_counter), plot = p_pa_pb_ratio_scatter, width = 11, height = 8.5, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.png", fig_counter), plot = splsda_ratio_ind_traj_fig, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.svg", fig_counter), plot = splsda_ratio_ind_traj_fig, width = 9, height = 8, dpi = 200)
```
**Supplemental Figure `r fig_counter` `r fig_counter <- fig_counter + 1`: Information Contained in Plasma Proteomic Ratios for Initial GBM vs Recurrent GBM Stratification.** (**a**) Pairwise protein ratios shows promise of individual protein-pair ratios being able to stratify initial vs recurrent plasma GBM samples. (**b**) Individual trajectories demonstrate the individual variance of patients path to recurrence in a low-dimensional space using ratios.

\pagebreak

\blandscape

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=8.5, fig.width=11, dpi = 200}

ttr_survival_plot_grid <- ggarrange(surv_p1, surv_p2, surv_p3, surv_p4, ncol = 4, labels="A")
bottom_plots <- ggarrange(ggarrange(ttr_surv_p3, ttr_surv_p4, ncol=2), ggarrange(surv_p5, ttr_surv_p5, ncol=2), labels=c("B", "C"))

supp_figure_4 <- ggarrange(ttr_survival_plot_grid, bottom_plots, nrow=2, ncol=1)
supp_figure_4

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.png", fig_counter), plot = supp_figure_4, width = 25, height = 15, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s.svg", fig_counter), plot = supp_figure_4, width = 25, height = 15, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.png", fig_counter), plot = ttr_survival_plot_grid, width = 11, height = 8.5, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a.svg", fig_counter), plot = ttr_survival_plot_grid, width = 11, height = 8.5, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a_1.png", fig_counter), plot = ttr_surv_p1, width = 11, height = 8.5, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a_1.svg", fig_counter), plot = ttr_surv_p1, width = 11, height = 8.5, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a_2.png", fig_counter), plot = ttr_surv_p2, width = 11, height = 8.5, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a_2.svg", fig_counter), plot = ttr_surv_p2, width = 11, height = 8.5, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a_3.png", fig_counter), plot = ttr_surv_p3, width = 11, height = 8.5, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a_3.svg", fig_counter), plot = ttr_surv_p3, width = 11, height = 8.5, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a_4.png", fig_counter), plot = ttr_surv_p4, width = 11, height = 8.5, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_a_4.svg", fig_counter), plot = ttr_surv_p4, width = 11, height = 8.5, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.png", fig_counter), plot = surv_p5, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_b.svg", fig_counter), plot = surv_p5, width = 9, height = 8, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_c.png", fig_counter), plot = ttr_surv_p5, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_c.svg", fig_counter), plot = ttr_surv_p5, width = 9, height = 8, dpi = 200)

ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_c_with_ci.png", fig_counter), plot = ttr_surv_p5, width = 9, height = 8, dpi = 200)
ggsave(sprintf("../../figures/elisa_plasma_proteomics/supp_figure_%s_subfigure_c_with_ci.svg", fig_counter), plot = ttr_surv_p5, width = 9, height = 8, dpi = 200)
```
**Supplemental Figure `r fig_counter` `r fig_counter <- fig_counter + 1`: Survival and Time to Recurrence Analysis.** (**a**) Kaplan-Meyer plots showing demonstrating high concentration of ASAH1 and high concentration of SYNEMIN contributing to a higher probability of tumor recurrence occurring at a later date. (**b**) Kaplan-Meyer plots demonstrating the combination of both ASAH1 and SYNEMIN and their contribution of survival probability and time to recurrence.

\elandscape

\pagebreak

\blandscape

```{r echo=FALSE, message=FALSE, warning=FALSE}
logit_net_comp_perf_met_table %>%
  gtsave(sprintf("../../tables/elisa_plasma_proteomics/supp_table_%s.rtf", tab_counter))
logit_net_comp_perf_met_table %>%
  gtsave(sprintf("../../tables/elisa_plasma_proteomics/supp_table_%s.rtf", tab_counter))

logit_net_comp_perf_met_table
```

**Supplemental Table `r as.roman(tab_counter)` `r tab_counter <- tab_counter + 1`: Logistic Regression with Elastic Net Regularization Performance Metrics** 

```{r echo=FALSE, message=FALSE, warning=FALSE}
stat_comp_table %>%
  gtsave(sprintf("../../tables/elisa_plasma_proteomics/supp_table_%s.rtf", tab_counter))
stat_comp_table %>%
  gtsave(sprintf("../../tables/elisa_plasma_proteomics/supp_table_%s.tex", tab_counter))

stat_comp_table
```

**Supplemental Table `r as.roman(tab_counter)` `r tab_counter <- tab_counter + 1`: Statistical Comparisons Between Population Groups per Genes** 

\elandscape

\pagebreak

# Methods

## Statistical Test for Comparison of Log2 Distributions

Log2 transformed distributions of protein concentrations were compared pairwise for control, initial GBM, and recurrent GBM samples for each protein, ASAH1, GNMPB, MMP-9 and SYNEMIN. Statistical p-values were computed using a permutation test, that utilizes t-test statistics but performs random permutations of the data for computing a p-value. Note, for paired sample data, initial GBM vs recurrent GBM, a paired permutation test was performed instead. Visualization and statistical analyses were performed in R (v4.1.2) and RStudio (v2021.09.2+382 ), permutation tests were performed using broman (v0.80).

## Dimensionaly Reduction

To visualize the samples plasma proteomics on a different dimension, we applied a sparse Partial Least Squares Discriminant Analysis (sPLS-DA), which maximizes the covariance between the explanatory and response variable(s). This was performed on the log2 transformed protein concentrations, as well as the log2 transformed ratios ($\frac{initial \quad GBM}{recurrent \quad GBM}$ vs $\frac{recurrent \quad GBM}{initial \quad GBM}$) of each protein. Individual trajectories are displayed as vector arrows connecting initial GBM patient samples to their corresponding recurrent GBM patient samples, in the sPLS-DA space using the first two components that contain the most amount of explained variance. An average trajectory direction can be computed by taking the average of the individual trajectory vectors, to visualize an overall trajectory of initial GBM state to recurrent GBM state. To visualize decision boundaries for class separation (control, initial GBM and recurrent GBM samples), we applied a maximum distance approach. [1] Visualization and statistical analyses were performed in R (v4.1.2) and RStudio (v2021.09.2+382), sPLS-DA analysis was performed using mixOmics (v6.18.1).

## Prediction Analysis

Prediction tasks were performed using parametric (logistic regression, logistic regression with elastic net regularization), non-parametric (k-nearest neighbours) and ensemble (random forest) supervised learning algorithms. General predictive tasks were performed on: controls vs initial GBM samples, controls vs recurrent GBM samples, initial GBM samples vs recurrent GBM samples. Data used in prediction tasks are log2 transformed protein concentrations, log2 transformed protein concentrations supplemented with clinical data (*Tumor localization*, *Sex*, *Age at surgery (years)*, *Type of resection*), and log2 transformed ratios of $\frac{initial \quad GBM}{recurrent \quad GBM}$ vs $\frac{recurrent \quad GBM}{initial \quad GBM}$ protein measurements. Model training and prediction performances are performed on the entire data set using leave-one-out cross-validation due to the small sample size. Visualization and statistical analyses were performed in R (v4.1.2) and RStudio (v2021.09.2+382), prediction analyses were performed using caret (v6.0-93).  

# References

1. Rohart, F, B Gautier, A Singh, and K-A Lê Cao. 2017. “MixOmics: An R Package for ‘Omics Feature Selection and Multiple Data Integration.” PLoS Computational Biology 13 (11). Cold Spring Harbor Labs Journals.