---
title: "ASAH1 rGBM Manuscript ELISA Figure"
author: "Justin Sing, University of Toronto"
date: "`r format(Sys.time(), '%a %B %d %X %Z %Y')`"
header-includes:
- \usepackage{pdflscape}
- \newcommand{\blandscape}{\begin{landscape}}
- \newcommand{\elandscape}{\end{landscape}}
- \usepackage{fancyhdr}
- \usepackage[fit, breakall]{truncate}
  \usepackage{tcolorbox}
  \newtcolorbox{myquote}{colback=red!5!white, colframe=red!75!black}
  \renewenvironment{quote}{\begin{myquote}}{\end{myquote}}
output:
  pdf_document: 
    keep_tex: no
    latex_engine: xelatex
    toc: yes
    toc_depth: 4
  html_notebook: default
editor_options:
  chunk_output_type: console
---

\pagestyle{fancy}
<!-- \fancyhead[LO,CE]{ASAH1 rGBM Manuscript} -->
\fancyfoot[CO,CE]{Justin Sing, University of Toronto}
\fancyfoot[LE,RO]{\thepage}

<!-- # Required Libraries -->
```{r message=FALSE, warning=FALSE, include=FALSE}
#' @title Try load package, install from correct package manager if not installed
#' @param package (character) string name of package to load
#' @param s (character) source to install package from it not installed
installIfNeeded <- function(package, s = "CRAN") {
  # s: "CRAN" or "BIO"
  if (s == "CRAN") {
    if (! requireNamespace(package, quietly=TRUE)) {
      install.packages(package)
      library(package, character.only = TRUE, quietly = TRUE)
    } else {
      library(package, character.only = TRUE, quietly = TRUE)
    }
  } else if (s == "BIO") {
    if (! requireNamespace("BiocManager", quietly=TRUE)) {
      install.packages("BiocManager")
    }
    if (! requireNamespace(package, quietly=TRUE)) {
      BiocManager::install(package)
      library(package, character.only = TRUE, quietly = TRUE)
    } else {
      library(package, character.only = TRUE, quietly = TRUE)
    }
  } else {
    stop(sprintf("Unknown source \"%s\".", s))
  }
}

# Data I/O and Wrangling
installIfNeeded("data.table")
installIfNeeded("dplyr")
# Visualization
installIfNeeded("ggplot2")
installIfNeeded("grid")
installIfNeeded("gridExtra")
installIfNeeded("ggpubr")
# Table Visualization
installIfNeeded("gt")
# Statistics
installIfNeeded("rstatix")
installIfNeeded("broman")
installIfNeeded("pwr")
# Survival Analysis
installIfNeeded("survival")
installIfNeeded("survminer")
# Machine Learning 
installIfNeeded("caret")
installIfNeeded("plotROC")
installIfNeeded("mixOmics", "BIO")
# parallilization
installIfNeeded("future")
```

<!-- # Helper Functions -->
```{r message=FALSE, warning=FALSE, include=FALSE}
#' median_centre_data
#'
#' @param train_df  (data.frame) A dataframe that contains labels and features to run through prediction models. Has to have the following structure:
#' \item{class}{(character) sample/observation label. i.e. ctrl, iGBM, or rGBM sample.}
#' \item{feature_1}{(numeric} a specific feature measurement such as protein ASAH1)}
#' \item{feature_n}{(numeric) another specific feature.}
#' @param pred_col_prob (character) A character corresponding to the class of probabilities you want to extract for. i.e. "GBM"
#' @param multiclass (logical) Run predictions using multiclass modelling and prediction if more than 2 classes.
#' 
#' @return a data.frame with columns sample_id_col, class, and the measures normalized by median
#' 
#' This function takes in a train data.frame and returns the data after normalizing the measures 
#' by subtracting the median of each sample from each value and then adding the median of the whole data.
#' The input data.frame is also pivoted for easier analysis.
median_centre_data <- function(train_df){
  values_to = "measure_col"
  measure_col="measure_col"
  train_df %>%
  tibble::rowid_to_column(var = "sample_id_col") %>%
  tidyr::pivot_longer(cols = c(ASAH1, MMP.9, SYNEMIN, GPNMB), names_to = "feature_id_col", values_to = "measure_col") ->
  train_long

  normalized_df = train_long %>% group_by(sample_id_col) %>% 
      mutate(median_run = median(measure_col, na.rm = TRUE)) %>% 
      ungroup()
  old_measure_col = paste("preNorm", measure_col, sep = "_")
  normalized_df = normalized_df %>% mutate(median_global = median(!!(sym(measure_col)), 
    na.rm = TRUE), `:=`(!!(old_measure_col), !!(sym(measure_col)))) %>% 
    mutate(diff_norm = median_global - median_run) %>% mutate(`:=`(!!(sym(measure_col)), 
    !!(sym(measure_col)) + diff_norm))
    
  normalized_df = switch("default", all = normalized_df, 
      default = normalized_df, minimal = normalized_df %>% 
        dplyr::select(c(sample_id_col, feature_id_col, 
          measure_col, old_measure_col)))
  
  # ggplot(normalized_df, aes(x=as.factor(sample_id_col), y=measure_col, col=sample_id_col)) +
  #   geom_boxplot()
  
  normalized_df %>%
    dplyr::select(sample_id_col, class, feature_id_col, measure_col) %>%
    tidyr::pivot_wider(id_cols = c("sample_id_col", "class"), names_from = "feature_id_col", values_from = "measure_col") %>%
    tibble::column_to_rownames("sample_id_col") -> train_df
  
  return(train_df)
}


median_centre_recurrent_data <- function(data_long){
  data_long[patient_state==1] %>%
    dplyr::group_by(gene, Number.x) %>%
    dplyr::add_count() %>%
    dplyr::filter(n==2) %>% 
    dplyr::ungroup() -> tmp_select
  
  tmp_select %>%
    dplyr::select(gene, Number.x, conc, class=manifestation_type) %>%
    dplyr::mutate(conc = log2(conc+1)) -> tmp
  tmp %>%
    tidyr::pivot_wider(id_cols = c(Number.x, class), names_from = gene, values_from = conc ) -> tmp
  colnames(tmp) <- make.names(colnames(tmp))
  tmp <- na.omit(tmp)
  
  tmp_df_proteins <- median_centre_data(tmp[, c(2:6)])
  tmp_median_centred <- cbind(tmp[, 1], tmp_df_proteins) 
  
  tmp_median_centred %>%
    dplyr::rename(manifestation_type=class, `MMP-9`=MMP.9) %>%
    tidyr::pivot_longer(cols = c(ASAH1, `MMP-9`, SYNEMIN, GPNMB), names_to = "gene", values_to = "conc") -> tmp_median_centred
  
  data_long_tmp <- merge(tmp_select[, c("Number.x", "gene", "patient_state", "LAB_ID", "collection_type", "manifestation_type")], tmp_median_centred, by=c("Number.x", "manifestation_type", "gene"), all.y=T)
  return(data_long_tmp)
}

#' @title Master Trainer and Prediction
#' @description Top Function to perform predictions for testing logistic regression, logistic regression + elastic regularization, random forest and k-Nearest Neightbours
#' 
#' @param train_df (data.frame) A dataframe that contains labels and features to run through prediction models. Has to have the following structure:
#' \item{class}{(character) sample/observation label. i.e. ctrl, iGBM, or rGBM sample.}
#' \item{feature_1}{(numeric} a specific feature measurement such as protein ASAH1)}
#' \item{feature_n}{(numeric) another specific feature.}
#' @param pred_col_prob (character) A character corresponding to the class of probabilities you want to extract for. i.e. "GBM"
#' @param multiclass (logical) Run predictions using multiclass modelling and prediction if more than 2 classes.
#' 
#' @return
#' 
#' returns a list object that contains trained model objects, lift data and roc data.
#' 
#' \item(model){(glmnet) Logistic Regression with Elastic Net Regularization Model }
#' \item(model_rf){(rf) Random Forest Model}
#' \item(model_logit){(glm) Logistic Regression Model}
#' \item(model_boostedlogit){(knn) k-Nearest Neighbours Models}
#' \item(master_lift){(data.table) lift data for all models}
#' \item(roc_data){(data.table) roc data for all models}
run_predictions <- function(train_df, pred_col_prob="GBM", multiclass=FALSE){
  
  ##*****************************************************
  ##  Model Training  ----
  
  if (!multiclass){
    # specify the cross-validation method
    ctrl <- trainControl(method = "LOOCV", search = "grid", classProbs = T, savePredictions = "final", summaryFunction = twoClassSummary, verboseIter = T)
    
    # fit models and use LOOCV to evaluate performance
    model <- train( class ~ ., data = train_df, na.action = na.omit, method = "glmnet", family="binomial", trControl = ctrl, metric = "ROC")
    model_rf <- train( class ~ ., data = train_df, na.action = na.omit, method = "rf", verbose=1, trControl = ctrl, metric = "ROC")
    model_logit <- train( class ~ . , data = train_df, na.action = na.omit, method = "glm", family="binomial", trControl = ctrl, metric = "ROC")
    model_boostedlogit <- train( class ~ . , data = train_df, na.action = na.omit, method = "knn", trControl = ctrl, metric = "ROC")
    
  } else{
    # specify the cross-validation method
    ctrl <- trainControl(method = "LOOCV", number = 10, search = "grid", classProbs = T, savePredictions = "final", verboseIter = T)
    
    # fit models and use LOOCV to evaluate performance
    model <- train( class ~ . , data = train_df, na.action = na.omit, method = "glmnet", family="multinomial", trControl = ctrl, metric = "ROC")
    model_rf <- train( class ~ ., data = train_df, na.action = na.omit, method = "rf", verbose=1, trControl = ctrl, metric = "ROC")
    model_logit <- train( class ~ . , data = train_df, na.action = na.omit, method = "multinom", family="binomial", trControl = ctrl, metric = "ROC")
    model_boostedlogit <- train( class ~ . , data = train_df, na.action = na.omit, method = "knn", trControl = ctrl, metric = "ROC")
  }
  
  ##*****************************************************
  ##  Generate Lift Plot Data ----
  
  if(!multiclass)  {
    for_lift = data.frame(Class = model$pred[model$pred$alpha==model$bestTune$alpha & model$pred$lambda==model$bestTune$lambda ,]$obs,  
                          glmnet = model$pred[model$pred$alpha==model$bestTune$alpha & model$pred$lambda==model$bestTune$lambda , pred_col_prob])
    for_lift_rf = data.frame(Class = model_rf$pred[model_rf$pred$mtry==model_rf$bestTune$mtry,]$obs,  
                             rf = model_rf$pred[model_rf$pred$mtry==model_rf$bestTune$mtry ,pred_col_prob])
    if (!multiclass){
      for_lift_logit = data.frame(Class = model_logit$pred$obs,  
                                  logit = model_logit$pred[,pred_col_prob])
    } else {
      for_lift_logit = data.frame(Class = model_logit$pred[model_logit$pred$decay==model_logit$bestTune$decay,]$obs,  
                                  logit = model_logit$pred[model_logit$pred$decay==model_logit$bestTune$decay,pred_col_prob])
    }
    for_lift_boostedlogit = data.frame(Class = model_boostedlogit$pred[model_boostedlogit$pred$k==model_boostedlogit$bestTune$k,]$obs,  
                                       knn = model_boostedlogit$pred[model_boostedlogit$pred$k==model_boostedlogit$bestTune$k ,pred_col_prob])
    
    lift_obj = lift(Class ~ glmnet, data = for_lift, class = pred_col_prob)
    lift_obj_rf = lift(Class ~ rf, data = for_lift_rf, class = pred_col_prob)
    lift_obj_logit = lift(Class ~ logit, data = for_lift_logit, class = pred_col_prob)
    lift_obj_boostedlogit = lift(Class ~ knn, data = for_lift_boostedlogit, class = pred_col_prob)
    
    master_lift <- rbindlist(list(lift_obj$data, lift_obj_rf$data, lift_obj_logit$data, lift_obj_boostedlogit$data))
    master_lift$liftModelVar <- factor(master_lift$liftModelVar, levels = c("logit", "glmnet", "knn", "rf"))
    
  } else {
    master_lift <- data.frame()
  }
  
  ##*****************************************************
  ## Generate ROC Prediction Data ----
  
  if (!multiclass){
    roc_data <- rbindlist(list(Logit=model_logit$pred, 
                               `Elastic Logit`=model$pred[model$pred$alpha==model$bestTune$alpha & model$pred$lambda==model$bestTune$lambda, ],
                               KNN=model_boostedlogit$pred[model_boostedlogit$pred$k==model_boostedlogit$bestTune$k, ], 
                               RF=model_rf$pred[model_rf$pred$mtry==model_rf$bestTune$mtry, ]), 
                          idcol = "Model", use.names = TRUE, fill = TRUE)
    roc_data$Model <- factor(roc_data$Model, levels=c("Logit", "Elastic Logit", "KNN", "RF"))
  } else {
    roc_data <- rbindlist(list(Logit=model_logit$pred[model_logit$pred$decay==model_logit$bestTune$decay, ], 
                               `Elastic Logit`=model$pred[model$pred$alpha==model$bestTune$alpha & model$pred$lambda==model$bestTune$lambda, ], 
                               KNN=model_boostedlogit$pred[model_boostedlogit$pred$k==model_boostedlogit$bestTune$k, ], 
                               RF=model_rf$pred[model_rf$pred$mtry==model_rf$bestTune$mtry, ]), 
                          idcol = "Model", use.names = TRUE, fill = TRUE)
    roc_data$Model <- factor(roc_data$Model, levels=c("Logit", "Elastic Logit", "KNN", "RF"))
  }
  
  ##*****************************************************
  ##  Return a list object with each model object, ----
  ##  lift data and roc data
  
  return(list(model=model, model_rf=model_rf, model_logit=model_logit, model_boostedlogit=model_boostedlogit, master_lift=master_lift, roc_data=roc_data))
}

#' @title Permutation Test Statistics
#' @description Perform permutation test to compare two groups
#' 
#' @param data (data.frame) if paired, dataframe must have comparison groups in two different columns with the associated log2 conc, else comparison groups must be in a single column with the grouping name. 
#' @param gene (char) character of gene being tested
#' @param response_var (char) column containing response data
#' @param group1 (char) character of group 1. i.e. control
#' @param group2 (char) character of group 2. i.e. treatment
#' @param method (char) character of either 'paired.perm.test' or 'perm.test'
#' @param grouping_column (Char) character of column containing group labels
#' 
#' @return returns a data.frame of data and test p-value
do_perm_test <- function(data, gene, response_var, group1, group2, method, grouping_column){
  data <- as.data.table(data)
  results <- data.frame(gene=gene, .y.=response_var, group1=group1, group2=group2, method=method, p=NA )
  if (method=="paired.perm.test"){
    message(sprintf("INFO: Performing %s", method))
    results$p <- broman::paired.perm.test(data[[group2]] - data[[group1]])
  } else if (method=="perm.test"){
    message(sprintf("INFO: Performing %s", method))
    results$p <- broman::perm.test(data[data[[grouping_column]]==group1][[response_var]], data[data[[grouping_column]]==group2][[response_var]], n.perm = 20000, var.equal = FALSE)
  }
  message(sprintf("INFO: Returning results"))
  return( results )
}

#' @title Subset a data frame based on groups
#' @description Subset a data frame based on groups
#' 
#' @param data (data.frame) if paired, dataframe must have comparison groups in two different columns with the associated log2 conc, else comparison groups must be in a single column with the grouping name. 
#' @param filter_groups (char) vector of character of groups to filter for
#' @param method (char) character of either 'paired.perm.test' or 'perm.test'
#' @param grouping_column (Char) character of column containing groups labels
#' 
#' @return returns a subset of data
filter_data <- function(data, filter_groups, method, grouping_column){
  data %>%
    dplyr::filter(!!rlang::sym(grouping_column) %in% filter_groups) %>%
    dplyr::select(Number.x, !!rlang::sym(grouping_column), gene, log2_conc) -> data_sub
  if (method=="paired.perm.test"){
    data_sub %>% 
      dplyr::group_by(Number.x) %>% 
      dplyr::add_count() %>%
      dplyr::ungroup() %>% 
      dplyr::filter(n==8) %>%
      dplyr::select(-n) -> data_sub
  }
  data_sub
}

#' Calculate the ratio of iGBM to rGBM and rGBM to iGBM for a given data frame
#'
#' @param data_long2 A data frame with columns "gene", "Number.x", "patient_state", "Age at surgery (years)", "Sex", "Tumor localization", "Type of resection", "MGMT promotor methylation", "p53 accumulation", "Ki67-Li", "EGFR vIII", "Treatment", "Time-to-reccurence (days)", "Latency (days)", and "conc".
#' @return A data frame with the calculated ratios.
get_ratio_data <- function(data_long2, already_log_transformed=F){
  ## Data wrangling ----
  ### iGBM over rGBM ratio
  data_long2 %>% 
    dplyr::rowwise() %>%
    dplyr::mutate(conc = ifelse(already_log_transformed==T, conc, conc+1)) %>% # Add one to avoid cases of 0 division
    dplyr::group_by(gene, Number.x) %>%
    dplyr::group_map(function(.d, ...){
      .d <- as.data.table(.d)
      .d %>% 
        dplyr::group_by(gene, Number.x, patient_state) %>% 
        dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                         agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
      # print(.d)
      if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
        .d_initial <- .d[manifestation_type=="Initial manifestation"]
        .d_second <- .d[manifestation_type=="Second manifestation"]
        if (!already_log_transformed){
          tmp$ratio <- .d_initial$conc / .d_second$conc
        } else {
          tmp$ratio <- .d_initial$conc - .d_second$conc
        }
      } else if (nrow(.d)==2) {
        .d_initial <- .d[collection_type=="E1"]
        .d_second <- .d[collection_type=="E2"]
        if (!already_log_transformed){
          tmp$ratio <- .d_initial$conc / .d_second$conc
        } else {
          tmp$ratio <- .d_initial$conc - .d_second$conc
        }
      } else {
        tmp$ratio <- NA #.d$conc
      }
      return(tmp)
    }, .keep = TRUE) %>%
    rbindlist() %>%
    dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBMoverrGBM", "pre-surgery/post-surgery"))) %>%
    dplyr::filter(sample_comparison=="iGBMoverrGBM" ) %>%
    dplyr::rowwise() %>%
    dplyr::mutate(log2_conc = ifelse(already_log_transformed==T, ratio, log2(ratio))) %>%
    # dplyr::group_by(gene, Number.x) %>%
    # dplyr::add_count() %>%
    # dplyr::filter(n==2) %>%
    dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("iGBMoverrGBM"))) -> pdata
  
  ### rGBM over iGBM
  data_long2 %>% 
    dplyr::rowwise() %>%
    dplyr::mutate(conc = ifelse(already_log_transformed==T, conc, conc+1)) %>% # Add one to avoid cases of 0 division
    dplyr::filter(!is.na(manifestation_type) | patient_state!=0) %>%
    dplyr::group_by(gene, Number.x) %>%
    dplyr::group_map(function(.d, ...){
      .d <- as.data.table(.d)
      .d %>% 
        dplyr::group_by(gene, Number.x, patient_state) %>% 
        dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                         agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
      # print(.d)
      if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
        .d_initial <- .d[manifestation_type=="Initial manifestation"]
        .d_second <- .d[manifestation_type=="Second manifestation"]
        if (!already_log_transformed){
          tmp$ratio <- .d_second$conc / .d_initial$conc
        } else {
          tmp$ratio <- .d_second$conc - .d_initial$conc
        }
      } else if (nrow(.d)==2) {
        .d_initial <- .d[collection_type=="E1"]
        .d_second <- .d[collection_type=="E2"]
        if (!already_log_transformed){
          tmp$ratio <- .d_second$conc / .d_initial$conc
        } else {
          tmp$ratio <- .d_second$conc - .d_initial$conc
        }
      } else {
        tmp$ratio <- NA #.d$conc
      }
      return(tmp)
    }, .keep = TRUE) %>%
    rbindlist() %>%
    dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "rGBMoveriGBM", "post-surgery/pre-surgery"))) %>%
    dplyr::filter(sample_comparison=="rGBMoveriGBM" ) %>%
    dplyr::filter(grepl(";", agg_lab_id)) %>%
    dplyr::rowwise() %>%
    dplyr::mutate(log2_conc = ifelse(already_log_transformed==T, ratio, log2(ratio))) %>%
    # dplyr::group_by(gene, Number.x) %>%
    # dplyr::add_count() %>%
    # dplyr::filter(n==2) %>%
    dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("rGBMoveriGBM"))) -> pdata2
  
  ### Prepare final table with ratios
  pdata %>%
    dplyr::select(Number.x, gene, log2_conc) %>%
    tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
    dplyr::mutate(class="iGBM") -> pdata
  
  pdata2 %>%
    dplyr::select(Number.x, gene, log2_conc) %>%
    tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
    dplyr::mutate(class="rGBM") -> pdata2
  
  
  pdata <- rbindlist(list(pdata, pdata2))

  ### Pivot Table to wide
  pdata %>%
    dplyr::select(Number.x, class, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide
  
  ## Remove non-paired patients
  data_wide %>%
    dplyr::group_by(Number.x) %>%
    dplyr::add_count() %>%
    dplyr::filter(n==2) %>%
    dplyr::ungroup() %>%
    dplyr::select(-n) -> data_wide
  
  # Prepare final table for analysis
  train_df <- data_wide
  train_df <- na.omit(train_df)  
  train_df <- train_df[ !is.infinite(train_df$ASAH1) , ]
  # Make Valid Column Names 
  colnames(train_df) <- make.names(colnames(train_df))
  train_df$class <- factor(train_df$class, levels=c("iGBM", "rGBM"))
  colnames(train_df) <- make.names(colnames(train_df))
  train_df <- as.data.table(train_df)
  train_df$group=0
  
  return(train_df)
}

#' Generate train and test datasets for leave-one-patient-out cross-validation
#'
#' This function takes in a dataset and returns a list containing train and test datasets
#' for leave-one-patient-out cross-validation. The test dataset consists of a single patient
#' with the "rGBM" class. If the `test_patient_id` parameter is not `NULL`, the test dataset
#' is subsetted using the provided patient ID. If the `test_patient_id` parameter is `NULL`,
#' a random patient with the "rGBM" class is chosen for the test dataset. The train dataset
#' consists of all other patients. If the `unique_random_train_patient_per_class` parameter is 
#' `TRUE`, the train dataset is further subsetted to include a unique patient for each "iGBM"
#' and "rGBM" class. The `test` parameter can be used to set the seed for the random number 
#' generator when sampling patients for the test and train datasets. This is useful for testing 
#' purposes.
#'
#' @param data A data frame containing the "Number.x" and "class" columns.
#' @param test_patient_id An optional patient ID to use for the test dataset. If not provided,
#' a random patient with the "rGBM" class is chosen.
#' @param unique_random_train_patient_per_class A logical value indicating whether to include
#' a unique patient for each "iGBM" and "rGBM" class in the train dataset.
#' @param test A logical value indicating whether to set the seed for the random number
#' generator for testing purposes.
#' @return A list containing the train and test datasets.
#' @export
#' @examples
#' get_train_test_for_loocv(data)
get_train_test_for_loocv <- function(data, test_patient_id=NULL, unique_random_train_patient_per_class = TRUE, test = FALSE){
  # Set seed for testing
  if (test){
    set.seed(2023)
  }
  
  if (is.null(test_patient_id)){
    # Sample single sample rGBM for Test data
    test_patients <- sample(unique(data$Number.x), 1)
    test_df <- subset(data, Number.x %in% test_patients & class == "rGBM")
  } else {
    # Subset for test data using provided test_patient_id
    test_df <- subset(data, Number.x %in% test_patient_id & class == "rGBM")
    test_patients <- test_patient_id
  }
  
  # Subset data to remove sample in test data
  data <- subset(data, !(Number.x %in% test_patients))
  
  if (unique_random_train_patient_per_class) {
    # Sample a unique patient for each iGBM and rGBM class
    igbm_train_patients <- sample(unique(data$Number.x[data$class == "iGBM"]), length(unique(data$Number.x))/2)
    rgbm_train_patients <- setdiff(data$Number.x, igbm_train_patients)
    
    # Filter data to obtain training data with a unique patient for each iGBM and rGBM class
    train_df <- subset(data, (Number.x %in% igbm_train_patients & class == "iGBM") | 
                         (Number.x %in% rgbm_train_patients & class == "rGBM"))
  } else {
    train_df <- data
  }
  
  return(list(train_df = train_df, test_df = test_df))
}


#' Calculate leave-one-patient-out cross-validation performance for splsda model
#'
#' This function takes in a dataset and a vector of patient IDs and returns a proportion
#' of correct predictions using a leave-one-patient-out cross-validation approach. A splsda
#' model is fit using the train data and used to make predictions on the test data. The
#' `unique_random_train_patient_per_class` parameter can be used to specify whether to 
#' include a unique patient for each "iGBM" and "rGBM" class in the train dataset. The
#' function returns a proportion of correct predictions.
#'
#' @param ratio_data A data frame containing the ratio data and class labels.
#' @param patient_ids A vector of patient IDs to use for the leave-one-patient-out
#' cross-validation.
#' @param unique_random_train_patient_per_class A logical value indicating whether to include
#' a unique patient for each "iGBM" and "rGBM" class in the train dataset.
#' @return A proportion of correct predictions.
#' @examples
#' splsda_loocv_perf(ratio_data, patient_ids)
splsda_loocv_perf <- function(ratio_data, patient_ids, unique_random_train_patient_per_class=TRUE){
  # Vector to store accuracy of LOOCV
  acc_vec <- numeric(length(patient_ids))
  i <- 1
  for (test_patient_id in patient_ids){
    # Split data into train and testing
    train_test <- get_train_test_for_loocv(ratio_data, test_patient_id=test_patient_id, unique_random_train_patient_per_class=unique_random_train_patient_per_class)
    train_df <- train_test$train_df
    test_df <- train_test$test_df
    # Perform splsda
    # MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2, -7, -8, -9, -10)], train_df$class, keepX = c(4,4)) 
    MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2)], train_df$class, keepX = c(4,4)) 
    # Make prediction
    # pred_test <- predict(MyResult.splsda, test_df[, c(-1, -2, -7, -8, -9, -10)], test_df$class)
    pred_test <- predict(MyResult.splsda, test_df[, c(-1, -2)], test_df$class)
    # evaluate the prediction accuracy for the first two components
    predict.comp2 <- pred_test$class$mahalanobis.dist[,2]
    # compute confusion matrix
    conf_mat <- table(factor(predict.comp2, levels = levels(train_df$class)), test_df$class)
    # print(conf_mat)
    # compute accuracy
    acc <- (conf_mat[1,1] + conf_mat[2,2])/(conf_mat[1,1] + conf_mat[2,2] + conf_mat[1,2] + conf_mat[2,1])
    # message(sprintf("INFO: Test Patient %s - predicted class = %s - model accuracy = %s", test_patient_id, predict.comp2, acc))
    # Store accuracy
    acc_vec[i] <- acc
    # update index
    i <- i + 1
  }
  # Compute proportion of correct predictions
  mean(acc_vec==1)
}

splsda <- function(train_df, test_df){
    
    # Perform splsda
    MyResult.splsda <- mixOmics::splsda(train_df, train_df$class, keepX = c(4,4)) 
    # Make prediction
    pred_test <- predict(MyResult.splsda, test_df, test_df$class)
    # evaluate the prediction accuracy for the first two components
    predict.comp2 <- pred_test$class$mahalanobis.dist[,2]
    # compute confusion matrix
    conf_mat <- table(factor(predict.comp2, levels = levels(train_df$class)), test_df$class)
    # print(conf_mat)
    # compute accuracy
    acc <- (conf_mat[1,1] + conf_mat[2,2])/(conf_mat[1,1] + conf_mat[2,2] + conf_mat[1,2] + conf_mat[2,1])
    
    return(list(model=MyResult.splsda, acc=acc))
}
```

<!-- # Data File Paths -->
```{r message=FALSE, warning=FALSE, include=FALSE}
patient_clinical_data <- "../../data/elisa_plasma_proteomics/Patients_information_FINAL.csv"
patient_longitudinal_mapping_file <- "../../data/elisa_plasma_proteomics/Patients_for_ELISA_longitudinal_mapping.csv"
elisa_data_file <- "../../data/elisa_plasma_proteomics/ELISA_data.csv"
```

<!-- # Load Data -->
```{r echo=FALSE, warning=FALSE}
# Get patient meta data ----
clinical_data <- fread(patient_clinical_data)
patient_longitudinal_mapping_data <- fread(patient_longitudinal_mapping_file)
patient_meta_data <- merge(clinical_data, patient_longitudinal_mapping_data, by.x="LAB_ID", by.y = "LAB_ID", all = TRUE)

patient_meta_data %>%
  dplyr::filter(!is.na(manifestation_type)) %>%
  dplyr::select(Number.x, Sex, `Age at surgery (years)`, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, `Mass Spectrometry`, Lipidomics, `RT-qPCR`, IHC, ELISA) %>%
  dplyr::group_by(Number.x) %>%
  dplyr::add_count() %>%
  unique() %>%
  dplyr::ungroup()-> meta_dt

# Load actual ELISA data ----
elisa_data <- fread(elisa_data_file)

# Merge ELISA data with patient meta data ----
data_long <- merge(elisa_data, patient_meta_data, by = "LAB_ID")
```

\pagebreak

<!-- # Distribution of ELISA Measurements -->
```{r echo=FALSE, fig.height=11, fig.width=21, message=FALSE, warning=FALSE}
# iGBM vs rGBM ----
## Build gg dataframe for plotting ----
data_long %>%
  dplyr::filter(!is.na(manifestation_type) | patient_state==0) %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(Manifestation = plyr::mapvalues(manifestation_type, from=c("Initial manifestation", "Second manifestation"), to=c("iGBM", "rGBM"))) %>%
  dplyr::mutate(Manifestation = ifelse(is.na(Manifestation), "Ctrl", Manifestation)) %>%
  dplyr::mutate(Manifestation = factor(Manifestation, levels = c("Ctrl", "iGBM", "rGBM"))) -> pdata
pdata <- as.data.table(pdata)

## Perform Statistical-Test ----

comparisons <- matrix(c("Ctrl", "Ctrl", "iGBM", "iGBM", "rGBM", "rGBM"), ncol=2)
methods <- c("perm.test", "perm.test", "paired.perm.test")

stats_frame <- parallel::mclapply(seq_len(nrow(comparisons)), function(iterator){
  comp <- comparisons[iterator, ]
  method <- methods[iterator]
  data_sub <- filter_data(pdata, comp, method, "Manifestation" )
  
  uni_genes <- data_sub %>% dplyr::pull(gene) %>% unique()
  
  gene_stat_frame <- parallel::mclapply(split(data_sub, data_sub$gene), function(.d){
    if (method=="paired.perm.test"){
      .d %>%
        dplyr::select(-gene) %>%
        tidyr::pivot_wider(id_cols = "Number.x", names_from = "Manifestation", values_from = "log2_conc") -> .d_wide
    } else {
      .d -> .d_wide
    }
    message(sprintf("INFO: Computing %s for %s for comparison %s", method, unique(.d$gene), paste(comp, collapse=", ") ))
    do_perm_test(data = .d_wide, gene = .d$gene, response_var = "log2_conc", group1 = comp[1], group2 = comp[2], method = method, grouping_column = "Manifestation")
  }, mc.cores = 4) %>%
    rbindlist() %>%
    unique()
}, mc.cores=3) %>% rbindlist()

stat_test <- stats_frame %>%
  dplyr::group_by(group1, group2) %>%
  adjust_pvalue(method = "fdr") %>%
  add_significance(p.col="p", output.col = "p.signif",
                   cutpoints = c(0, 1e-04, 0.001, 0.01, 0.05, 1),
                   symbols = c("****", "***", "**", "*", "ns")) %>%
  dplyr::arrange(gene) %>%
  add_xy_position(data=pdata, formula= log2_conc ~ Manifestation) 

stat_test %>%
  mutate(y.position = c(6.64, 6.64, 6.64, 8.70004, 8.70004, 8.70004, 11.49704, 11.49704, 11.49704, 13.62804, 13.62804, 13.62804)) -> stat_test

stat_test_igm_v_rgm <- stat_test

## Build the plot ----

ggpaired(data = pdata, id = "Number.x", x = "Manifestation", y = "log2_conc", fill = "Manifestation", palette = 'grey', line.color = "grey", line.size = 0.4, notch = TRUE) +
  scale_fill_manual(values = RColorBrewer::brewer.pal(length(unique(pdata$Manifestation)), "Dark2"), labels = c("Ctrl", "iGBM", "rGBM")) +
  # geom_jitter(aes(label=Number.x)) +
  # ggtitle("Initial vs Recurrence") +
  labs(x="", y="log2( Concentration )") +
  facet_wrap(~gene, nrow = 2, scales = "free_y") +
  theme_classic() +
  theme(panel.background = element_rect(fill="white"),
        # panel.grid.major.y = element_line(colour = "black"), 
        plot.title = element_text(hjust = 0.5, size=17),
        # axis.line = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="vertical",) +
  guides(fill='none',
         col='none') -> p1_org


dat <- ggplot_build(p1_org)$data[[1]]
dat <- cbind(dat, gene=sort(unique(pdata$gene))[dat$PANEL])

p1_org <- p1_org + geom_segment(data=dat, 
                 aes(x=xmin, xend=xmax, 
                    y=middle, yend=middle), 
                 colour="white", size=1)

p1_org + 
  stat_pvalue_manual(stat_test, label = "{p.signif} p = {p}", step.group.by = "gene", step.increase = 0.2, hide.ns = T, size=4, bracket.size=1) +
  geom_point(data=data.frame(Manifestation=c("iGBM", "iGBM", "iGBM", "iGBM"), gene=c("ASAH1", "SYNEMIN", "GPNMB", "MMP-9" ), y=c(6, 16, 8, 11)),
             aes(x=Manifestation, y=y, group=gene), alpha=0) -> p1_org

# Exclude outlier patient number.x 54 
ggpaired(data = pdata[Number.x!=54], id = "Number.x", x = "Manifestation", y = "log2_conc", fill = "Manifestation", palette = 'grey', line.color = "grey", line.size = 0.4) +
  # stat_compare_means(paired = TRUE, label.x = 1.5, geom = "label") +
  # stat_compare_means(comparisons = my_comparisons, group.by="gene", label.y = c(4, 7, 7), label.x = c(1, 2, 3)) +
  scale_fill_manual(values = RColorBrewer::brewer.pal(length(unique(pdata$Manifestation)), "Dark2"), labels = c("Ctrl", "iGBM", "rGBM")) +
  # geom_jitter(aes(label=Number.x)) +
  # ggtitle("Initial vs Recurrence") +
  labs(x="", y="log2( Concentration )") +
  facet_wrap(~gene, scales = "free_y") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "bottom", legend.box="vertical",) +
  guides(fill="none",
         col='none') -> p1

dat <- ggplot_build(p1)$data[[1]]
dat <- cbind(dat, gene=sort(unique(pdata$gene))[dat$PANEL])

p1 <- p1 + geom_segment(data=dat, 
                 aes(x=xmin, xend=xmax, 
                    y=middle, yend=middle), 
                 colour="white", size=1)

p1 + 
  stat_pvalue_manual(stat_test, label = "{p.signif} p = {p}", step.group.by = "gene", step.increase = 0.2, hide.ns = T, size=4, bracket.size=1) +
  geom_point(data=data.frame(Manifestation=c("iGBM", "iGBM", "iGBM", "iGBM"), gene=c("ASAH1", "SYNEMIN", "GPNMB", "MMP-9" ), y=c(6, 15, 8, 11)),
             aes(x=Manifestation, y=y, group=gene), alpha=0) -> p1

## Build the table ----
stat_test %>%
  dplyr::mutate(method = plyr::mapvalues(method, from = c("perm.test", "paired.perm.test"), to = c("Permutation Test", "Paired Permutation Test"))) %>%
  dplyr::select(Gene=gene, group1, group2, `Statistical Test`=method, p) %>%
  dplyr::mutate(Comparison = sprintf("%s vs %s", group1, group2)) %>%
  tidyr::pivot_wider(id_cols = c("Comparison", "Statistical Test"), names_from = c(Gene), values_from = "p") %>%
  gt() %>%
  tab_header(title = md("**Statistical Comparisons Between Population Groups per Genes**")) %>%
  cols_label(
    Comparison = md("**Comparison**"),
    `Statistical Test` = md("**Statistical Test**"),
    ASAH1 = md("**ASAH1**"),
    GPNMB = md("**GPNMB**"),
    `MMP-9` = md("**MMP-9**"),
    SYNEMIN = md("**SYNEMIN**"),
  ) %>% 
  tab_footnote(
    footnote = md("Permutation test using t-test statisics for non-paired samples, or paired permutation test using pair t-test statistics for paired samples."),
    locations = cells_column_labels("Statistical Test")
  ) %>% 
  tab_footnote(
    footnote = md("Values in cells represent the non-adjusted p-value from the corresponding statiscal test."),
    locations = cells_column_labels(c("ASAH1", "SYNEMIN", "GPNMB", "MMP-9"))
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(ASAH1),
      rows = ASAH1 < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(GPNMB),
      rows = GPNMB < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(`MMP-9`),
      rows = `MMP-9` < 0.05
    )
  ) %>%
  tab_style(
    style = list(
      cell_fill(color = "#F9E3D6"),
      cell_text(style = "italic")
    ),
    locations = cells_body(
      columns = c(SYNEMIN),
      rows = SYNEMIN < 0.05
    )
  ) -> stat_comp_table


## Save figure ----
main_fig_a <- p1
# subfigure_6
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_a.png", plot = main_fig_a, width = 11, height = 9, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_a.svg", plot = main_fig_a, width = 11, height = 9, dpi = 200)

supp_figure_1_a <- p1_org

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_1_a_avg.png", plot = supp_figure_1_a, width = 11, height = 9, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_1_a_avg.svg", plot = supp_figure_1_a, width = 11, height = 9, dpi = 200)
```

<!-- # Latency and TTR Data for KM plots-->
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=9, include=FALSE}
# LAtency data ----
latency_data_file <- "../../data/elisa_plasma_proteomics/ELISA_LATENCY_KM.csv"
latency_data <- fread(latency_data_file)
# Clean names
latency_data <- rstatix::make_clean_names(latency_data)
latency_data %>%
  dplyr::mutate(ASAH1.ng.ml = log2(ASAH1.ng.ml+1),
                SYNM.ng.ml = log2(SYNM.ng.ml+1),
                GPNMB.ng.ml = log2(GPNMB.ng.ml+1),
                MMP.9.ng.ml = log2(MMP.9.ng.ml+1)) %>%
  dplyr::mutate(ASAH1 = ifelse(ASAH1.ng.ml>=mean(ASAH1.ng.ml), "high", "low"),
                SYNM = ifelse(SYNM.ng.ml>=mean(SYNM.ng.ml), "high", "low"),
                GPNMB = ifelse(GPNMB.ng.ml>=mean(GPNMB.ng.ml), "high", "low"),
                MMP9 = ifelse(MMP.9.ng.ml>=mean(MMP.9.ng.ml, na.rm=T), "high", "low")) -> latency_data

### ASAH1 ----
fit <- survfit(Surv(Latency..days., status) ~ ASAH1, data = latency_data)
print(fit)
n_data <- table(latency_data[!is.na(latency_data$Latency..days.)]$ASAH1)
# Change color, linetype by strata, risk.table color by strata
surv_p1 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1600, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high\n(n=%s)", n_data[1]), sprintf("ASAH1=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Latency Time (Days)")

surv_p1 <- surv_p1$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.8, 0.9)) 

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_a_1.png", plot = surv_p1, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_a_1.svg", plot = surv_p1, width = 11, height = 8.5, dpi = 200)

### SYNM ----
fit <- survfit(Surv(Latency..days., status) ~ SYNM, data = latency_data)
print(fit)
n_data <- table(latency_data[!is.na(latency_data$Latency..days.)]$SYNM)
# Change color, linetype by strata, risk.table color by strata
surv_p2 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1600, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("SYNM=high\n(n=%s)", n_data[1]), sprintf("SYNM=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Latency Time (Days)")

surv_p2 <- surv_p2$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.8, 0.9))

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_a_2.png", plot = surv_p2, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_a_2.svg", plot = surv_p2, width = 11, height = 8.5, dpi = 200)

### GPNMB ----
fit <- survfit(Surv(Latency..days., status) ~ GPNMB, data = latency_data)
print(fit)
n_data <- table(latency_data[!is.na(latency_data$Latency..days.)]$GPNMB)
# Change color, linetype by strata, risk.table color by strata
surv_p3 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1600, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("GPNMB=high\n(n=%s)", n_data[1]), sprintf("GPNMB=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Latency Time (Days)")

surv_p3 <- surv_p3$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.8, 0.9))

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_a_3.png", plot = surv_p3, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_a_3.svg", plot = surv_p3, width = 11, height = 8.5, dpi = 200)

### MMP9 ----
fit <- survfit(Surv(Latency..days., status) ~ MMP9, data = latency_data)
print(fit)
n_data <- table(latency_data[!is.na(latency_data$Latency..days.)]$MMP9)
# Change color, linetype by strata, risk.table color by strata
surv_p4 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1600, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("MMP9=high\n(n=%s)", n_data[1]), sprintf("MMP9=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Latency Time (Days)")

surv_p4 <- surv_p4$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.8, 0.9))

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_a_4.png", plot = surv_p4, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_a_4.svg", plot = surv_p4, width = 11, height = 8.5, dpi = 200)

elisa_supp_subfigure_a <- ggarrange(surv_p1 + rremove("ylab") + rremove("xlab"), 
                                    surv_p2 + rremove("ylab") + rremove("xlab"),
                                    surv_p3 + rremove("ylab") + rremove("xlab"),
                                    surv_p4 + rremove("ylab") + rremove("xlab"),
                                    nrow=1, labels = NULL, align = "hv")

elisa_supp_subfigure_a <- annotate_figure(elisa_supp_subfigure_a, left = textGrob("Survival probability", rot = 90, hjust=0.4, vjust = 0.8, gp = gpar(cex = 1.3)),
                    bottom = textGrob("Latency Time (Days)", gp = gpar(cex = 1.3)))

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_a.png", plot = elisa_supp_subfigure_a, width = 25, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_a.svg", plot = elisa_supp_subfigure_a, width = 25, height = 8.5, dpi = 200)


### ASAH1 + SYN ----
fit <- survfit(Surv(Latency..days., status) ~ ASAH1 + SYNM, data = latency_data)
print(fit)
# Change color, linetype by strata, risk.table color by strata
surv_p5 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(1600, 0.75), conf.int = F,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high, SYNM=high (n=%s)", fit$n.risk[1]), sprintf("ASAH1=high, SYNM=low (n=%s)", fit$n.risk[fit$n.risk[1]+1]), sprintf("ASAH1=low, SYNM=high (n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]), sprintf("ASAH1=low, SYNM=low (n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]+fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]]))
          ) +
  labs(x="Latency Time (Days)")  +
  guides(col=guide_legend(nrow=4,byrow=TRUE))

surv_p5 <- surv_p5$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.6, 0.87))

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_c_2.png", plot = surv_p5, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_c_2.svg", plot = surv_p5, width = 11, height = 8.5, dpi = 200)  


# survival model with time to recurrence; combined group of all 4 proteins ----
ttr_surv_data <- merge(latency_data, patient_meta_data[, .(LAB_ID, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `Time-to-reccurence (days)`)], by.x="lab_id", by.y="LAB_ID", all.x=T)
ttr_surv_data$`Time-to-reccurence (days)`[ ttr_surv_data$`Time-to-reccurence (days)`=="/" ] = ""
ttr_surv_data$`Time-to-reccurence (days)` <- as.numeric(ttr_surv_data$`Time-to-reccurence (days)`)
ttr_surv_data <- rstatix::make_clean_names(ttr_surv_data)

### ASAH1 ----
fit <- survfit(Surv(Time.to.reccurence..days., status) ~ ASAH1, data = ttr_surv_data)
print(fit)
n_data <- table(ttr_surv_data[!is.na(ttr_surv_data$Time.to.reccurence..days.)]$ASAH1)

ttr_data_km_data_asah1 <- broom::tidy(fit)

# Change color, linetype by strata, risk.table color by strata
ttr_surv_p1 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high\n(n=%s)", n_data[1]), sprintf("ASAH1=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability") 

ttr_surv_p1 <- ttr_surv_p1$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.8, 0.9))

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b_1.png", plot = ttr_surv_p1, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b_1.svg", plot = ttr_surv_p1, width = 11, height = 8.5, dpi = 200)

### SYNM ----
fit <- survfit(Surv(Time.to.reccurence..days., status) ~ SYNM, data = ttr_surv_data %>% dplyr::select(-MMP.9.ng.ml))
print(fit)
n_data <- table(ttr_surv_data[!is.na(ttr_surv_data$Time.to.reccurence..days.)]$SYNM)

ttr_data_km_data_synm <- broom::tidy(fit)

# Change color, linetype by strata, risk.table color by strata
ttr_surv_p2 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("SYNM=high\n(n=%s)", n_data[1]), sprintf("SYNM=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")

ttr_surv_p2 <- ttr_surv_p2$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.8, 0.9))

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b_2.png", plot = ttr_surv_p2, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b_2.svg", plot = ttr_surv_p2, width = 11, height = 8.5, dpi = 200)

elisa_main_subfigure_b <- ggarrange(ttr_surv_p1 + rremove("ylab") + rremove("xlab"), 
                                    ttr_surv_p2 + rremove("ylab") + rremove("xlab"),
                                    ncol=2, labels = NULL, align = "hv")

elisa_main_subfigure_b <- annotate_figure(elisa_main_subfigure_b, left = textGrob("Time to Recurrence probability", rot = 90, hjust=0.4, vjust = 0.8, gp = gpar(cex = 1.3)),
                    bottom = textGrob("Time to Recurrence (Days)", gp = gpar(cex = 1.3)))

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b.png", plot = elisa_main_subfigure_b, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b.svg", plot = elisa_main_subfigure_b, width = 11, height = 8.5, dpi = 200)

### GPNMB ----
fit <- survfit(Surv(Time.to.reccurence..days., status) ~ GPNMB, data = ttr_surv_data)
print(fit)
n_data <- table(ttr_surv_data[!is.na(ttr_surv_data$Time.to.reccurence..days.)]$GPNMB)

ttr_data_km_data_gpnmb <- broom::tidy(fit)

# Change color, linetype by strata, risk.table color by strata
ttr_surv_p3 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("GPNMB=high\n(n=%s)", n_data[1]), sprintf("GPNMB=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")

ttr_surv_p3 <- ttr_surv_p3$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.8, 0.9))

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_b_1.png", plot = ttr_surv_p3, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_b_1.svg", plot = ttr_surv_p3, width = 11, height = 8.5, dpi = 200)


### MMP9 ----
fit <- survfit(Surv(Time.to.reccurence..days., status) ~ MMP9, data = ttr_surv_data)
print(fit)
n_data <- table(ttr_surv_data[!is.na(ttr_surv_data$Time.to.reccurence..days.)]$MMP9)

ttr_data_km_data_mmp9 <- broom::tidy(fit)

# Change color, linetype by strata, risk.table color by strata
ttr_surv_p4 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(650, 0.85), conf.int = TRUE,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = "Dark2",
          legend.title = "",
          legend.labs = c(sprintf("MMP9=high\n(n=%s)", n_data[1]), sprintf("MMP9=low\n(n=%s)", n_data[2]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")

ttr_surv_p4 <- ttr_surv_p4$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.8, 0.9))


# ttr_data_km_data <- rbindlist(list(ASAH1=ttr_data_km_data_asah1, SYNM=ttr_data_km_data_synm, GPNMB=ttr_data_km_data_gpnmb, MMP9=ttr_data_km_data_mmp9), idcol = "Protein")
# 
# ttr_data_km_data[, strata2 := gsub("^\\w+=(\\w+)$", "\\1", strata)]
# 
# ggplot() +
#   geom_step(data=filter(ttr_data_km_data, Protein %in% c("ASAH1", "SYNM")), aes(x=time, y=estimate, group=strata, col=Protein, linetype=strata2)) +
#   
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_b_2.png", plot = ttr_surv_p4, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_b_2.svg", plot = ttr_surv_p4, width = 11, height = 8.5, dpi = 200)  
  
elisa_supp_subfigure_b <- ggarrange(ttr_surv_p3 + rremove("ylab") + rremove("xlab"), 
                                    ttr_surv_p4 + rremove("ylab") + rremove("xlab"),
                                    ncol=2, labels = NULL, align = "hv")

elisa_supp_subfigure_b <- annotate_figure(elisa_supp_subfigure_b, left = textGrob("Time to Recurrence probability", rot = 90, hjust=0.4, vjust = 0.8, gp = gpar(cex = 1.3)),
                    bottom = textGrob("Time to Recurrence (Days)", gp = gpar(cex = 1.3)))

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_b.png", plot = elisa_supp_subfigure_b, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_b.svg", plot = elisa_supp_subfigure_b, width = 11, height = 8.5, dpi = 200)


### ASAH1 + SYNM ----
fit <- survfit(Surv(Time.to.reccurence..days., status) ~ ASAH1 + SYNM, data = ttr_surv_data)
print(fit)

# Change color, linetype by strata, risk.table color by strata
ttr_surv_p5 <- ggsurvplot(fit,
          pval = TRUE, pval.coord = c(640, 0.75), conf.int = F,
          risk.table = FALSE, # Add risk table
          risk.table.col = "strata", # Change risk table color by groups
          linetype = "strata", # Change line type by groups
          surv.median.line = "hv", # Specify median survival
          ggtheme = theme_bw(), # Change ggplot2 theme
          legend.title = "",
          legend.labs = c(sprintf("ASAH1=high, SYNM=high (n=%s)\n", fit$n.risk[1]), sprintf("ASAH1=high, SYNM=low (n=%s)\n", fit$n.risk[fit$n.risk[1]+1]), sprintf("ASAH1=low, SYNM=high (n=%s)\n", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]), sprintf("ASAH1=low, SYNM=low (n=%s)", fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]+fit$n.risk[fit$n.risk[1]+1+fit$n.risk[fit$n.risk[1]+1]]]))
          ) +
  labs(x="Time to Recurrence (Days)", y="Time to Recurrence probability")  +
  guides(col=guide_legend(ncol=1,byrow=TRUE))

ttr_surv_p5 <- ttr_surv_p5$plot +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.text = element_text(size=12),
        legend.spacing.x = unit(0.25, 'cm'),
        legend.position = c(0.6, 0.87))

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_c_1.png", plot = ttr_surv_p5, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_2_c_1.svg", plot = ttr_surv_p5, width = 11, height = 8.5, dpi = 200)  

elisa_supp_subfigure_c <- ggarrange(ttr_surv_p5 , 
                                    surv_p5,
                                    ncol=2, labels = NULL, align = "hv")

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_c.png", plot = elisa_supp_subfigure_c, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_subfigure_c.svg", plot = elisa_supp_subfigure_c, width = 11, height = 8.5, dpi = 200)

## Final Supp Figure 2 ----
elisa_supp_figure_2 <- ggarrange(elisa_supp_subfigure_a, ggarrange(elisa_supp_subfigure_b, elisa_supp_subfigure_c, ncol = 2, labels=c("B", "C")),
          labels = c("A"), nrow = 2)

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure_2.png", plot = elisa_supp_figure_2, width = 27, height = 18, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure_2.svg", plot = elisa_supp_figure_2, width = 27, height = 18, dpi = 200)

```

<!-- # sPLS-DA for conc data and Ratio data -->
```{r echo=FALSE, fig.height=11, fig.width=21, message=FALSE, warning=FALSE}
# Dimensionality Reduction ----
## Protein Concentration
#### Data wrnagling ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type %in% c("Initial manifestation", "Second manifestation") | sample_comparison=="ctrl") %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(0,1), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(class = ifelse(manifestation_type=="Initial manifestation", "iGBM", class)) %>%
  dplyr::mutate(class = ifelse(manifestation_type=="Second manifestation", "rGBM", class)) %>%
  dplyr::mutate(class = ifelse(is.na(manifestation_type), "Ctrl", class)) -> pdata

pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

#### Only use proteins for prediction ----
data_wide %>%
  dplyr::mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide

#### sPLS-DA ----
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)

MyResult.splsda <- mixOmics::splsda(train_df[, -1], as.factor(train_df[, 1]), keepX = c(4,4)) 

#### Predict boundary planes of separation ----
background <- background.predict(MyResult.splsda, comp.predicted=2, dist = "max.dist") 

#### Prepare data frame for plotting ----
plsda_data <- cbind( patient_id=na.omit(data_wide)  %>% dplyr::pull(Number.x), class=as.factor(train_df[, 1]), as.data.frame(MyResult.splsda$variates$X))
plsda_data$patient_id[plsda_data$class=="Ctrl"] <- "Ctrl"
plsda_data$patient_id <- as.factor(plsda_data$patient_id)
plsda_data %>%
  dplyr::arrange(class) -> plsda_data
plsda_data <- as.data.table(plsda_data)

#### Get average trajectories of iGBM to rGBM in low dimension ----
plsda_data[patient_id!="Ctrl" ] %>%
  dplyr::group_by(class) %>%
  dplyr::add_count() %>%
  dplyr::group_by(class, n) %>%
  dplyr::summarise(avg_comp1=sum(comp1)/n,
                   avg_comp2=sum(comp2)/n) %>%
  dplyr::ungroup() %>%
  unique() %>%
  dplyr::mutate(patient_id="avg_all") %>%
  as.data.table() -> average_comp

slope <- (average_comp$avg_comp2[2] - average_comp$avg_comp2[1])/(average_comp$avg_comp1[2] - average_comp$avg_comp1[1])
b = -0.25

average_comp[class=='rGBM']$avg_comp2 = slope*-1.5 + b
average_comp[class=='rGBM']$avg_comp1 = -1.5
average_comp[class=='iGBM']$avg_comp2 = slope*3 + b
average_comp[class=='iGBM']$avg_comp1 = 3

#### Prepare decision boundaries dataframe fo plotting ----
for(i in 1:length(background))
{
  if(!is.null(background[[i]]))
    background[[i]]=data.frame(id=i,col=names(background)[i], background[[i]])
}

background = do.call(rbind,background)

#### Build plot ----
### Average Trajectories 
p <- ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c( "#C2C2C2", "#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#C2C2C2", "#388ECC", "#F68B33")) +
  ggnewscale::new_scale_color() +
  geom_path(data=average_comp, aes(x=avg_comp1, y=avg_comp2, group = patient_id, col="Avg. iGBM\nto rGBM\nTrajectory"),
            arrow = arrow(type = "closed",
                          length=unit(0.1, "inches")), size=2, alpha=0.8) +
  geom_label(label="Ctrl", 
             aes(x=-2, y=2), fill="#C2C2C2", color="white", size=6) +
  geom_label(label="iGBM", 
             aes(x=4, y=0), fill="#388ECC", color="white", size=6) +
  geom_label(label="rGBM", 
             aes(x=-2.5, y=-1), fill="#F68B33", color="white", size=6) +
  geomtextpath::geom_labelpath(data=average_comp, aes(x=avg_comp1, y=avg_comp2, group = patient_id), label="Avg. iGBM to\nrGBM Trajectory",
            arrow = arrow(type = "closed",
                          length=unit(0.1, "inches")), size=5, alpha=0.8, 
             fill="red", color="white", size=6) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(shape="none",
         color_new="none",
         col_org=guide_legend(title="g", ncol=1),
         col="none",
         col_new=guide_legend(title="g", ncol=1))

splsda_avg_traj_fig <- p

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig1_subfigure_b.png", plot = splsda_avg_traj_fig, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig1_subfigure_b.svg", plot = splsda_avg_traj_fig, width = 11, height = 8.5, dpi = 200)


## Ratios ----
#### Get ratio data ----
data_long_tmp <- median_centre_recurrent_data(data_long)
train_df <- get_ratio_data(data_long_tmp, already_log_transformed = T)
#### sPLS-DA ----
MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2, -7)], train_df$class, keepX = c(4,4))
background <- background.predict(MyResult.splsda, comp.predicted=2, dist = "max.dist") 

plsda_data <- cbind( patient_id=train_df$Number.x, class=train_df$class, as.data.frame(MyResult.splsda$variates$X))
plsda_data$patient_id <- as.factor(plsda_data$patient_id)
plsda_data %>%
  dplyr::arrange(class) -> plsda_data
plsda_data <- as.data.table(plsda_data)

plsda_data %>%
  dplyr::group_by(class) %>%
  dplyr::add_count() %>%
  dplyr::group_by(class, n) %>%
  dplyr::summarise(avg_comp1=sum(comp1)/n,
                   avg_comp2=sum(comp2)/n) %>%
  dplyr::ungroup() %>%
  unique() %>%
  dplyr::mutate(patient_id="avg_all") %>%
  as.data.table() -> average_comp

slope <- (average_comp$avg_comp2[2] - average_comp$avg_comp2[1])/(average_comp$avg_comp1[2] - average_comp$avg_comp1[1])
b = -0.25

for(i in 1:length(background))
{
  if(!is.null(background[[i]]))
    background[[i]]=data.frame(id=i,col=names(background)[i],
                               background[[i]])
}

background = do.call(rbind,background)

p <- ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c("#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  # ggnewscale::new_scale_color() +
  # ggrepel::geom_label_repel(data=merge(plsda_data %>% dplyr::mutate(patient_id = as.numeric(as.character(patient_id))), unique(data_long[manifestation_type!='' & patient_state==1 & `Time-to-reccurence (days)`!='', .(Number.x, `Time-to-reccurence (days)`)]), by.x='patient_id', by.y="Number.x", all.x=T), aes(x=comp1, y=comp2, col=`Time-to-reccurence (days)`, label=paste(patient_id, `Time-to-reccurence (days)`, sep=": "))) +
  # ggnewscale::new_scale_color() +
  # geom_path(data=average_comp, aes(x=avg_comp1, y=avg_comp2, group = patient_id, col="Avg. iGBM\nto rGBM\nTrajectory"),
  #           arrow = arrow(type = "closed", ends='both',
  #                         length=unit(0.1, "inches")), size=2, alpha=0.8) +
  geom_label(label=expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")), 
             aes(x=min(plsda_data[class=="rGBM"]$comp1)+0.5, y=min(plsda_data[class=="rGBM"]$comp2)+0.01), fill="#F68B33", color="white", size=6) +
  geom_label(label=expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")), 
             aes(x=max(plsda_data[class=="iGBM"]$comp1)-0.5, y=max(plsda_data[class=="iGBM"]$comp2)-0.01), fill="#388ECC", color="white", size=6) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(shape="none",
         col="none")
  # guides(shape="none",
  #        colour_new_new=guide_legend(title="Class", ncol=1, byrow = T),
  #        colour_new='none',
  #        col=guide_legend(title="Trajectory", ncol=1))

splsda_ratio_avg_traj_fig <- p

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b.png", plot = splsda_ratio_avg_traj_fig, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_b.svg", plot = splsda_ratio_avg_traj_fig, width = 11, height = 8.5, dpi = 200)

## Ratios unique samples ----
#### Get ratio data ----
data_long_tmp <- median_centre_recurrent_data(data_long)
ratio_data <- get_ratio_data(data_long_tmp, already_log_transformed = T)
train_test <- get_train_test_for_loocv(ratio_data, test_patient_id=NULL, unique_random_train_patient_per_class=T)
train_df <- rbindlist(list(train_test$train_df, train_test$test_df), use.names = T)
#### sPLS-DA ----
MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2, -7)], train_df$class, keepX = c(4,4))
background <- background.predict(MyResult.splsda, comp.predicted=2, dist = "max.dist") 

plsda_data <- cbind( patient_id=train_df$Number.x, class=train_df$class, as.data.frame(MyResult.splsda$variates$X))
plsda_data$patient_id <- as.factor(plsda_data$patient_id)
plsda_data %>%
  dplyr::arrange(class) -> plsda_data
plsda_data <- as.data.table(plsda_data)

plsda_data %>%
  dplyr::group_by(class) %>%
  dplyr::add_count() %>%
  dplyr::group_by(class, n) %>%
  dplyr::summarise(avg_comp1=sum(comp1)/n,
                   avg_comp2=sum(comp2)/n) %>%
  dplyr::ungroup() %>%
  unique() %>%
  dplyr::mutate(patient_id="avg_all") %>%
  as.data.table() -> average_comp

slope <- (average_comp$avg_comp2[2] - average_comp$avg_comp2[1])/(average_comp$avg_comp1[2] - average_comp$avg_comp1[1])
b = -0.25

for(i in 1:length(background))
{
  if(!is.null(background[[i]]))
    background[[i]]=data.frame(id=i,col=names(background)[i],
                               background[[i]])
}

background = do.call(rbind,background)

p <- ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c("#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  ggnewscale::new_scale_color() +
  ggrepel::geom_label_repel(data=merge(plsda_data %>% dplyr::mutate(patient_id = as.numeric(as.character(patient_id))), unique(data_long[manifestation_type!='' & patient_state==1 & `Time-to-reccurence (days)`!='', .(Number.x, `Time-to-reccurence (days)`)]), by.x='patient_id', by.y="Number.x", all.x=T), aes(x=comp1, y=comp2, col=`Time-to-reccurence (days)`, label=paste0("id: ", patient_id))) +
  # ggnewscale::new_scale_color() +
  # geom_path(data=average_comp, aes(x=avg_comp1, y=avg_comp2, group = patient_id, col="Avg. iGBM\nto rGBM\nTrajectory"),
  #           arrow = arrow(type = "closed", ends='both',
  #                         length=unit(0.1, "inches")), size=2, alpha=0.8) +
  geom_label(label=expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")), 
             aes(x=min(background$Var1)+0.8, y=min(background$Var2)+0.2), fill="#F68B33", color="white", size=6) +
  geom_label(label=expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")), 
             aes(x=max(background$Var1)-0.8, y=max(background$Var2)-0.2), fill="#388ECC", color="white", size=6) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(shape="none",
         colour_new='none',
         col="none")
  # guides(shape="none",
  #        colour_new_new=guide_legend(title="Class", ncol=1, byrow = T),
  #        colour_new='none',
  #        col=guide_legend(title="Trajectory", ncol=1))

splsda_ratio_unique_patients <- p

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig1_subfigure_d.png", plot = splsda_ratio_unique_patients, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig1_subfigure_d.svg", plot = splsda_ratio_unique_patients, width = 11, height = 8.5, dpi = 200)

## Shuffled ratio data ----
#### Get ratio data and shuffle labels----
data_long_tmp <- median_centre_recurrent_data(data_long)
train_df <- get_ratio_data(data_long_tmp, already_log_transformed = T)
org_class <- train_df$class
#### I set the seed for the purpose of downstream plotting of the legends inside the plot
set.seed(2018)
train_df$class <- sample(train_df$class)
#### sPLS-DA ----
MyResult.splsda <- mixOmics::splsda(train_df[, c(-1, -2, -7)], train_df$class, keepX = c(4,4))
background <- background.predict(MyResult.splsda, comp.predicted=2, dist = "max.dist") 

plsda_data <- cbind( patient_id=train_df$Number.x, class=train_df$class, as.data.frame(MyResult.splsda$variates$X))
plsda_data$patient_id <- as.factor(plsda_data$patient_id)
plsda_data %>%
  dplyr::arrange(class) -> plsda_data
plsda_data <- as.data.table(plsda_data)

plsda_data %>%
  dplyr::group_by(class) %>%
  dplyr::add_count() %>%
  dplyr::group_by(class, n) %>%
  dplyr::summarise(avg_comp1=sum(comp1)/n,
                   avg_comp2=sum(comp2)/n) %>%
  dplyr::ungroup() %>%
  unique() %>%
  dplyr::mutate(patient_id="avg_all") %>%
  as.data.table() -> average_comp

slope <- (average_comp$avg_comp2[2] - average_comp$avg_comp2[1])/(average_comp$avg_comp1[2] - average_comp$avg_comp1[1])
b = -0.25

for(i in 1:length(background))
{
  if(!is.null(background[[i]]))
    background[[i]]=data.frame(id=i,col=names(background)[i],
                               background[[i]])
}

background = do.call(rbind,background)

p <- ggplot() +
  geom_polygon(data = background,aes(x=Var1, y=Var2,
                                     fill = col), inherit.aes = FALSE, show.legend=FALSE, alpha=0.2) +
  scale_fill_manual(values=c("#388ECC", "#F68B33")) +
  # geom_point(data=plsda_data[patient_id=="Ctrl"], aes(x=comp1, y=comp2, shape=class), col="black", size=3) +
  # geom_point(data=plsda_data[patient_id!="Ctrl"], aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  geom_point(data=plsda_data, aes(x=comp1, y=comp2, col=class, shape=class), size=3) +
  scale_shape_manual(values=c(15, 16, 17)) +
  scale_color_manual(values=c( "#388ECC", "#F68B33"), labels=c(expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")),
                                                               expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")))
  ) +
  # ggnewscale::new_scale_color() +
  # ggrepel::geom_label_repel(data=merge(plsda_data %>% dplyr::mutate(patient_id = as.numeric(as.character(patient_id))), unique(data_long[manifestation_type!='' & patient_state==1 & `Time-to-reccurence (days)`!='', .(Number.x, `Time-to-reccurence (days)`)]), by.x='patient_id', by.y="Number.x", all.x=T), aes(x=comp1, y=comp2, col=`Time-to-reccurence (days)`, label=paste0("id: ", patient_id))) +
  # ggnewscale::new_scale_color() +
  # geom_path(data=average_comp, aes(x=avg_comp1, y=avg_comp2, group = patient_id, col="Avg. iGBM\nto rGBM\nTrajectory"),
  #           arrow = arrow(type = "closed", ends='both',
  #                         length=unit(0.1, "inches")), size=2, alpha=0.8) +
  geom_label(label=expression(paste(frac('recurrent GBM', 'initial GBM'), " Ratio")), 
             aes(x=min(background$Var1)+0.8, y=min(background$Var2)+0.2), fill="#F68B33", color="white", size=6) +
  geom_label(label=expression(paste(frac('initial GBM', 'recurrent GBM'), " Ratio")), 
             aes(x=max(background$Var1)-0.8, y=max(background$Var2)-0.2), fill="#388ECC", color="white", size=6) +
  labs(x=sprintf("X-variate 1: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[1]*100)), y =sprintf("X-variate 2: %s%% explained variance", round(MyResult.splsda$prop_expl_var$X[2]*100))) +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, size=17),
        axis.title = element_text(size = 17),
        axis.text = element_text(size=16),
        strip.text = element_text(size=16),
        legend.title = element_text(size=16),
        legend.text = element_text(size=15),
        legend.position = "right", legend.box="vertical",) +
  guides(shape="none",
         colour_new='none',
         col="none")
  # guides(shape="none",
  #        colour_new_new=guide_legend(title="Class", ncol=1, byrow = T),
  #        colour_new='none',
  #        col=guide_legend(title="Trajectory", ncol=1))

splsda_ratio_shuffled_patients <- p

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig1_subfigure_e.png", plot = splsda_ratio_shuffled_patients, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig1_subfigure_e.svg", plot = splsda_ratio_shuffled_patients, width = 11, height = 8.5, dpi = 200)

```

<!-- # ROC prediction analyses -->
```{r fig.height=8, fig.width=9, message=FALSE, warning=FALSE, include=FALSE}
# iGBM vs Ctrl protein conc ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Initial manifestation" | sample_comparison=="ctrl") %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(0,1), to=c("ctrl", "GBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, `MMP-9`, SYNEMIN, GPNMB) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
pred_obj_1.1 <- run_predictions(train_df)

## ROC Figure 
g <- ggplot(pred_obj_1.1$roc_data, aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.1 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.1[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.1[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.1[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.1[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.1$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.1 <- g

# rGBM vs Ctrl protein conc ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | sample_comparison=="ctrl") %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(0,1), to=c("ctrl", "GBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, `MMP-9`, SYNEMIN, GPNMB) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
pred_obj_1.2 <- run_predictions(train_df)

## ROC Figure 
g <- ggplot(pred_obj_1.2$roc_data, aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.2 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.2[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.2[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.2[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.2[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.2$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.2 <- g

# iGBM vs rGBM protein conc----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | manifestation_type=="Initial manifestation") %>%
  dplyr::mutate(class = plyr::mapvalues(manifestation_type, from=c("Initial manifestation","Second manifestation"), to=c("iGBM", "rGBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, `MMP-9`, SYNEMIN, GPNMB) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df <- median_centre_data(train_df)
pred_obj_1.5 <- run_predictions(train_df, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.5$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.5 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.5[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.5[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.5[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.5[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.5$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.5 <- g

pred_obj_1.5$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.5_roc_data

# iGBM vs rGBM ratio ----

## Median Centre data before computing ratios

data_long[patient_state==1] %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::add_count() %>%
  dplyr::filter(n==2) %>% 
  dplyr::ungroup() -> tmp_select

tmp_select %>%
  dplyr::select(gene, Number.x, conc, class=manifestation_type) %>%
  dplyr::mutate(conc = log2(conc+1)) -> tmp
tmp %>%
  tidyr::pivot_wider(id_cols = c(Number.x, class), names_from = gene, values_from = conc ) -> tmp
colnames(tmp) <- make.names(colnames(tmp))
tmp <- na.omit(tmp)

tmp_df_proteins <- median_centre_data(tmp[, c(2:6)])
tmp_median_centred <- cbind(tmp[, 1], tmp_df_proteins) 

tmp_median_centred %>%
  dplyr::rename(manifestation_type=class, `MMP-9`=MMP.9) %>%
  tidyr::pivot_longer(cols = c(ASAH1, `MMP-9`, SYNEMIN, GPNMB), names_to = "gene", values_to = "conc") -> tmp_median_centred

data_long_tmp <- merge(tmp_select[, c("Number.x", "gene", "patient_state", "LAB_ID", "collection_type", "manifestation_type")], tmp_median_centred, by=c("Number.x", "manifestation_type", "gene"), all.y=T)

ggplot(data_long_tmp, aes(x=as.factor(Number.x), y=conc, col=Number.x)) +
    geom_boxplot()

data_long_tmp %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_initial$conc - .d_second$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBMoverrGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBMoverrGBM" ) %>%
  dplyr::mutate(log2_conc = ratio) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("iGBMoverrGBM"))) -> pdata

data_long_tmp %>% 
  # dplyr::mutate(conc = conc+1) %>% # Add one to avoid cases of 0 division
  dplyr::filter(!is.na(manifestation_type) | patient_state!=0) %>%
  dplyr::group_by(gene, Number.x) %>%
  dplyr::group_map(function(.d, ...){
    .d <- as.data.table(.d)
    .d %>% 
      dplyr::group_by(gene, Number.x, patient_state) %>% 
      dplyr::summarise(agg_lab_id = paste(LAB_ID, collapse=";"),
                       agg_collection_type = paste(collection_type, collapse = ";"), .groups = "keep") -> tmp
    # print(.d)
    if ( nrow(.d)==2 & any(!is.na(.d$manifestation_type)) ){
      .d_initial <- .d[manifestation_type=="Initial manifestation"]
      .d_second <- .d[manifestation_type=="Second manifestation"]
      tmp$ratio <- .d_second$conc - .d_initial$conc
    } else if (nrow(.d)==2) {
      .d_initial <- .d[collection_type=="E1"]
      .d_second <- .d[collection_type=="E2"]
      tmp$ratio <- .d_initial$conc / .d_second$conc
    } else {
      tmp$ratio <- .d$conc
    }
    return(tmp)
  }, .keep = TRUE) %>%
  rbindlist() %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "rGBMoveriGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="rGBMoveriGBM" ) %>%
  dplyr::filter(grepl(";", agg_lab_id)) %>%
  dplyr::mutate(log2_conc = ratio) %>%
  # dplyr::group_by(gene, Number.x) %>%
  # dplyr::add_count() %>%
  # dplyr::filter(n==2) %>%
  dplyr::mutate(class = plyr::mapvalues(patient_state, from=c(1), to=c("rGBMoveriGBM"))) -> pdata2

pdata %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="iGBM") -> pdata

pdata2 %>%
  dplyr::select(Number.x, gene, log2_conc) %>%
  tidyr::pivot_wider(id_cols = "Number.x", names_from = "gene", values_from = "log2_conc") %>%
  dplyr::mutate(class="rGBM") -> pdata2


pdata <- rbindlist(list(pdata, pdata2))

## Pivot Table
pdata %>%
  dplyr::select(Number.x, class, ASAH1, SYNEMIN, GPNMB, `MMP-9`) -> data_wide

## Prediction
train_df <- data_wide
train_df <- data_wide[Number.x!=32]
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, )
train_df <- train_df[ !is.infinite(train_df$ASAH1) , ]
## Make Valid Column Names 
colnames(train_df) <- make.names(colnames(train_df))
train_df$class <- factor(train_df$class, levels=c("iGBM", "rGBM"))
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
pred_obj_1.3 <- run_predictions(train_df, pred_col_prob = "rGBM")

## ROC Figure 
g <- ggplot(pred_obj_1.3$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) +
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.3 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.3[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.3[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.3[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.3[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.3$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.3 <- g

pred_obj_1.3$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.3_roc_data

## Merged Plot
pred_obj_1.3$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.3_roc_data

# iGBM vs rGBM with clinical data ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | manifestation_type=="Initial manifestation") %>%
  dplyr::mutate(class = plyr::mapvalues(manifestation_type, from=c("Initial manifestation","Second manifestation"), to=c("iGBM", "rGBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n, ASAH1, `MMP-9`, SYNEMIN, GPNMB,  Sex, `Age at surgery (years)`, `Tumor localization`, `Type of resection`) -> data_wide
# data_wide %>%
  # dplyr::mutate(`Time-to-reccurence (days)` = ifelse(class=="iGBM", 0, `Time-to-reccurence (days)`)) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
train_df_proteins <- median_centre_data(train_df[, c(1:5)])
train_df <- cbind(train_df_proteins, train_df[, c(6:9)])
pred_obj_1.4 <- run_predictions(train_df, pred_col_prob = "rGBM")

# coef(pred_obj_1.5$model$finalModel, pred_obj_1.5$model$bestTune$lambda)
# coef(pred_obj_1.41$model$finalModel, pred_obj_1.41$model$bestTune$lambda)
# coef(pred_obj_1.4$model$finalModel, pred_obj_1.4$model$bestTune$lambda)


## ROC Figure 
g <- ggplot(pred_obj_1.4$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.4 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.4[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.4[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.4[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.4[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.4 <- g

pred_obj_1.4$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.4_roc_data

# iGBM vs rGBM with clinical data Only ----
data_long %>% 
  dplyr::group_by(gene, Number.x) %>%
  dplyr::mutate(sample_comparison = plyr::mapvalues(patient_state, from=c(0, 1, 2), to=c("ctrl", "iGBM/rGBM", "pre-surgery/post-surgery"))) %>%
  dplyr::filter(sample_comparison=="iGBM/rGBM" | sample_comparison=="ctrl") %>%
  dplyr::mutate(log2_conc = log2(conc+1)) %>%
  dplyr::filter(manifestation_type=="Second manifestation" | manifestation_type=="Initial manifestation") %>%
  dplyr::mutate(class = plyr::mapvalues(manifestation_type, from=c("Initial manifestation","Second manifestation"), to=c("iGBM", "rGBM"))) -> pdata

## Pivot Table
pdata %>%
  dplyr::select(Number.x, gene, `Age at surgery (years)`, Sex, `Tumor localization`, `Type of resection`, `MGMT promotor methylation`, `p53 accumulation`, `Ki67-Li`, `EGFR vIII`, Treatment, `Time-to-reccurence (days)`, `Latency (days)`, log2_conc, sample_comparison, class) %>%
  tidyr::pivot_wider(id_cols = c("Number.x",'Age at surgery (years)', 'Sex', 'Tumor localization', 'Type of resection', 'MGMT promotor methylation', 'p53 accumulation', 'Ki67-Li', 'EGFR vIII', 'Treatment', 'Time-to-reccurence (days)', 'Latency (days)', 'sample_comparison',  'class'), names_from = "gene", values_from = "log2_conc" ) %>%
  dplyr::ungroup() -> data_wide

## Only use proteins for prediction
data_wide %>%
  mutate(n = row_number()) %>%
  dplyr::select(Number.x, class, n,  Sex, `Age at surgery (years)`, `Tumor localization`, `Type of resection`) -> data_wide
# data_wide %>%
  # dplyr::mutate(`Time-to-reccurence (days)` = ifelse(class=="iGBM", 0, `Time-to-reccurence (days)`)) -> data_wide

## Prediction
train_df <- data_wide
train_df <- na.omit(train_df)  %>% dplyr::select(-Number.x, -n)
colnames(train_df) <- make.names(colnames(train_df))
train_df <- as.data.frame(train_df)
# train_df_proteins <- median_centre_data(train_df[, c(1:5)])
# train_df <- cbind(train_df_proteins, train_df[, c(6:9)])
pred_obj_1.41 <- run_predictions(train_df, pred_col_prob = "rGBM")

# coef(pred_obj_1.5$model$finalModel, pred_obj_1.5$model$bestTune$lambda)
# coef(pred_obj_1.4$model$finalModel, pred_obj_1.4$model$bestTune$lambda)


## ROC Figure 
g <- ggplot(pred_obj_1.41$roc_data, aes(m=rGBM, d=factor(obs, levels = c("iGBM", "rGBM")), color=Model)) + 
  geom_roc(n.cuts=0) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(length(unique(pred_obj_1.4$master_lift$liftModelVar)), "Dark2"), labels=c("Logit", "Elastic Logit", "KNN", "RF")) +
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity")

auc_tbl_1.41 <- as.data.table(calc_auc(g))

g <- g + annotate("text", x=0.70, y=0.45, label=paste("AUC =", round(auc_tbl_1.41[Model=="Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.41$master_lift$liftModelVar)), "Dark2")[1])
g <- g + annotate("text", x=0.70, y=0.35, label=paste("AUC =", round(auc_tbl_1.41[Model=="Elastic Logit"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.41$master_lift$liftModelVar)), "Dark2")[2])
g <- g + annotate("text", x=0.70, y=0.25, label=paste("AUC =", round(auc_tbl_1.41[Model=="KNN"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.41$master_lift$liftModelVar)), "Dark2")[3])
g <- g + annotate("text", x=0.70, y=0.15, label=paste("AUC =", round(auc_tbl_1.41[Model=="RF"]$AUC, 4)), color=RColorBrewer::brewer.pal(length(unique(pred_obj_1.41$master_lift$liftModelVar)), "Dark2")[4])
subfigure_1.41 <- g

pred_obj_1.41$roc_data %>%
  dplyr::rename(ctrl=iGBM, GBM=rGBM) %>%
  dplyr::mutate(pred=plyr::mapvalues(pred, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) %>%
  dplyr::mutate(obs=plyr::mapvalues(obs, from=c("iGBM", "rGBM"), to=c("ctrl", "GBM"))) -> tmp_pred_obj_1.41_roc_data

## Merged Plot ----

# Full
master_roc_data <- rbindlist(list(`Ctrl vs iGBM`=pred_obj_1.1$roc_data, `Ctrl vs rGBM`=pred_obj_1.2$roc_data, `iGBM vs rGBM protein`=tmp_pred_obj_1.5_roc_data, `iGBM vs rGBM protein + clinical`=tmp_pred_obj_1.4_roc_data, `iGBM vs rGBM protein ratios`=tmp_pred_obj_1.3_roc_data), idcol = "Comparison")

master_auc_data <- rbindlist(list(`iGBM vs rGBM protein + clinical`=auc_tbl_1.41, `Ctrl vs iGBM`=auc_tbl_1.1, `Ctrl vs rGBM`=auc_tbl_1.2, `iGBM vs rGBM protein`=auc_tbl_1.5, `iGBM vs rGBM protein + clinical`=auc_tbl_1.4, `iGBM vs rGBM protein ratios`=auc_tbl_1.3), idcol = "Comparison")

# Only iGMB and rGBM
master_roc_data <- rbindlist(list(`iGBM vs rGBM clinical`=tmp_pred_obj_1.41_roc_data, `iGBM vs rGBM protein`=tmp_pred_obj_1.5_roc_data, `iGBM vs rGBM protein + clinical`=tmp_pred_obj_1.4_roc_data, `iGBM vs rGBM protein ratios`=tmp_pred_obj_1.3_roc_data), idcol = "Comparison")

master_auc_data <- rbindlist(list(`iGBM vs rGBM clinical`=auc_tbl_1.41, `iGBM vs rGBM protein`=auc_tbl_1.5, `iGBM vs rGBM protein + clinical`=auc_tbl_1.4, `iGBM vs rGBM protein ratios`=auc_tbl_1.3), idcol = "Comparison")


master_auc_data[Model %in% c("Elastic Logit")] %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  tidyr::pivot_wider(id_cols = "Comparison", names_from = Model, values_from = AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(core = list(fg_params=list(col=(RColorBrewer::brewer.pal(5, "Dark2")))), base_size = 15)
auc_grob <- tableGrob(auc_annotation_tbl, rows=NULL, theme = table_theme)

g <- ggplot(master_roc_data[Model %in% c("Elastic Logit")], aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Comparison), linetype='solid') +
  geom_roc(n.cuts=0, size=1.5) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(5, "Dark2")) +
  # scale_linetype_manual(values=c("solid", "dotted"), labels=c("Elastic Logit")) + 
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity") +
  annotation_custom(auc_grob, xmin=0.25, xmax=0.985, ymin=0.10, ymax=0.30) +
  theme(axis.title = element_text(size=16),
        axis.text = element_text(size=15),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13)) +
  guides(col="none")

ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_d.png", plot = g, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_subfigure_d.svg", plot = g, width = 11, height = 8.5, dpi = 200)
subfigure_1 <- g

## Separate ROCs for controls

master_roc_data <- rbindlist(list(`Ctrl vs iGBM`=pred_obj_1.1$roc_data, `Ctrl vs rGBM`=pred_obj_1.2$roc_data), idcol = "Comparison")

master_auc_data <- rbindlist(list(`Ctrl vs iGBM`=auc_tbl_1.1, `Ctrl vs rGBM`=auc_tbl_1.2), idcol = "Comparison")

master_auc_data[Model %in% c("Elastic Logit")] %>%
  dplyr::mutate(AUC = sprintf("AUC: %s%%", round(AUC*100, 0))) %>%
  tidyr::pivot_wider(id_cols = "Comparison", names_from = Model, values_from = AUC) -> auc_annotation_tbl

table_theme <- ttheme_minimal(core = list(fg_params=list(col=(RColorBrewer::brewer.pal(5, "Dark2")))), base_size = 15)
auc_grob <- tableGrob(auc_annotation_tbl, rows=NULL, theme = table_theme)

g <- ggplot(master_roc_data[Model %in% c("Elastic Logit")], aes(m=GBM, d=factor(obs, levels = c("ctrl", "GBM")), color=Comparison), linetype='solid') +
  geom_roc(n.cuts=0, size=1.5) + 
  scale_color_manual(values = RColorBrewer::brewer.pal(5, "Dark2")) +
  # scale_linetype_manual(values=c("solid", "dotted"), labels=c("Elastic Logit")) + 
  coord_equal() +
  style_roc(x="1 - Specificity", y="Sensitivity") +
  annotation_custom(auc_grob, xmin=0.23, xmax=0.90, ymin=0.10, ymax=0.38) +
  theme(axis.title = element_text(size=14),
        axis.text = element_text(size=13),
        legend.title = element_text(size=14),
        legend.text = element_text(size=13)) +
  guides(col="none")

ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig1_c.png", plot = g, width = 11, height = 8.5, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_fig_c.svg", plot = g, width = 11, height = 8.5, dpi = 200)
roc_controls <- g
```

# Main Figure

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=18, fig.width=21}

main_figure <- ggarrange(
ggarrange(main_fig_a, elisa_main_subfigure_b, labels=c("A", "B"), ncol = 2),
ggarrange(splsda_ratio_avg_traj_fig, subfigure_1, labels=c("C", "D"), ncol = 2),
nrow=2
)
main_figure
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_figure.png", plot = main_figure, width = 23, height = 18, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_main_figure.svg", plot = main_figure, width = 23, height = 18, dpi = 200)
```

# Supplemental Figure 1
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=18, fig.width=21}
supp_figure_1 <- ggarrange(
ggarrange(supp_figure_1_a, splsda_avg_traj_fig, roc_controls, labels=c("A", "B", "C"), ncol = 3),
ggarrange(splsda_ratio_unique_patients, splsda_ratio_shuffled_patients, labels=c("D", "E"), ncol = 2),
nrow=2
)
supp_figure_1
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure_1.png", plot = supp_figure_1, width = 25, height = 18, dpi = 200)
ggsave("../../figures/elisa_plasma_proteomics/elisa_supp_figure_1.svg", plot = supp_figure_1, width = 25, height = 18, dpi = 200)
```

\blandscape

# Supplemental Figure 2
```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=18, fig.width=21}
elisa_supp_figure_2
```

\elandscape

# Extra COX-PH Model
## Checking Assumptions

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=18, fig.width=21}
# fit <- coxph(Surv(Time.to.reccurence..days., status) ~ sex + ASAH1.ng.ml + SYNM.ng.ml + GPNMB.ng.ml + MMP.9.ng.ml, data = ttr_surv_data)
# print(summary(fit))

library(finalfit)

dependent_var <- "Surv(Time.to.reccurence..days., status)"
explanatory_vars <- c("ASAH1.ng.ml", "SYNM.ng.ml", "GPNMB.ng.ml", "MMP.9.ng.ml")

ttr_surv_data %>% 
    coxphmulti(dependent_var, explanatory_vars) %>% 
    cox.zph() %>% 
    {zph_result <<- .} 

par(mfrow = c(2, 2), mai = c(0.65, 0.8, 0.1, 0.1)) # 2-by-2 grid of plots

plot(zph_result, var=1)
plot(zph_result, var=2)
plot(zph_result, var=3)
plot(zph_result, var=4)

zph_result 
```

Checking assumptions of no time trend of the residuals, it seems like only MMP-9 might violate this assumption. 

\blandscape

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.height=4.5, fig.width=7.5}

ttr_surv_data %>%
  finalfit(dependent_var, explanatory_vars) %>%
  gt()
  

ttr_surv_data %>%
  hr_plot(dependent_var, explanatory_vars, dependent_label="Time to Recurrence")

```

Holding other variables constant, a reduction in ASAH1 reduces the time to recurrence by a factor of 0.55, or 45 percent. This seems to be inline with the KM plot for ASAH1, however, SYNM seems to be a different story..

\elandscape

<!-- # Text -->

<!-- Based on our observations of variations in protein concentrations (ASAH1, SYNEMIN, and MMP-9) between the initial serum measurement and recurrent serum measurements (as shown in Figure xa), as well as the classification of low and high concentrations of ASAH1 and SYNEMIN with respect to the time to recurrence (Figure xb), we further investigated the use of interpersonal longitudinal information from each patient. To this end, we calculated individual patient ratios for each protein, i.e., the initial serum measurement divided by the recurrent serum measurement and vice versa. To represent this information in a lower dimensional space, we used sparse partial least squares discriminant analysis (sPLS-DA) to categorize patients based on their initial serum/recurrent serum ratio against their recurrent serum/initial serum ratio (Figure xc). The resulting separating plane suggests that patients start at the border of the classifier plane and move outward as their state of recurrence changes. This way, repeated serum measurements can be used to locate a patient's position in this lower dimensional space and determine their proximity to the initial post-surgery state or a recurrent state. Additionally, we created an elastic net logistic regression classification model to predict a patient's state, either initial manifestation or recurrent manifestation (Figure xd). The results indicate that the protein measurements provide most of the predictive power, while the addition of clinical parameters reduces the classification performance. However, the best performing classification was achieved when using the ratios (AUC 89%), suggesting that the interpersonal longitudinal information is more informative than the protein concentrations themselves. -->
